{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Sandbox for working with parsers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Setup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-20T01:25:18.987143Z",
     "start_time": "2021-04-20T01:25:14.261979Z"
    }
   },
   "outputs": [],
   "source": [
    "## CX: allows multiple lines of code to print from one code block\n",
    "from IPython.core.interactiveshell import InteractiveShell\n",
    "InteractiveShell.ast_node_interactivity = \"all\"\n",
    "\n",
    "## to get around bugs\n",
    "import nest_asyncio\n",
    "nest_asyncio.apply()\n",
    "\n",
    "import pathlib\n",
    "import pandas as pd\n",
    "import re\n",
    "\n",
    "from collections import defaultdict"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## DisGeNET disease->gene parsing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-03-05T03:44:03.377883Z",
     "start_time": "2021-03-05T03:44:03.374620Z"
    }
   },
   "outputs": [],
   "source": [
    "## put the path you want to use here\n",
    "allgene_pmid_path = pathlib.Path.home().joinpath('Desktop', 'ScrippsJob', 'DisGeNET', 'all_gene_disease_pmid_associations.tsv.gz')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-03-05T03:44:05.042447Z",
     "start_time": "2021-03-05T03:44:05.034698Z"
    }
   },
   "outputs": [],
   "source": [
    "def process_gene(file_path_gene_disease):\n",
    "    df_gene_disease = pd.read_csv(\n",
    "        file_path_gene_disease,\n",
    "        encoding=\"ISO-8859-1\",\n",
    "        sep=\"\\t\",\n",
    "        comment=\"#\",\n",
    "        compression=\"gzip\",\n",
    "    )\n",
    "    rename_gene = {\n",
    "        \"diseaseId\": \"umls\",\n",
    "        \"geneId\": \"gene_id\",\n",
    "        \"geneSymbol\": \"gene_name\",\n",
    "        \"diseaseName\": \"disease_name\",\n",
    "        \"pmid\": \"pubmed\",\n",
    "    }\n",
    "    df_gene_disease = df_gene_disease.where((pd.notnull(df_gene_disease)), None)\n",
    "    # source field could be multiple data sources concatenated by \";\", break them into a list\n",
    "    # df_gene_disease.source = to_list(df_gene_disease.source)\n",
    "    # df_gene_disease.diseaseType = to_list(df_gene_disease.diseaseType)\n",
    "    # df_gene_disease.diseaseSemanticType = to_list(df_gene_disease.diseaseSemanticType)\n",
    "    d = defaultdict(list)\n",
    "    # rename pandas columns\n",
    "    df_gene_disease = df_gene_disease.rename(columns=rename_gene)\n",
    "    # for each gene, group the results based on source, and merge all pubmed IDs together\n",
    "    for grp, subdf in df_gene_disease.groupby([\"umls\", \"source\", \"gene_id\"]):\n",
    "        records = subdf.to_dict(orient=\"records\")\n",
    "        doc = {\"source\": grp[1], \"gene_id\": int(grp[2]), \"pubmed\": set()}\n",
    "        for record in records:\n",
    "            for k, v in record.items():\n",
    "                if isinstance(v, np.int64):\n",
    "                    record[k] = int(v)\n",
    "                if k in [\"gene_name\", \"DSI\", \"DPI\", \"score\", \"EI\"]:\n",
    "                    doc[k] = v\n",
    "                elif k in [\"YearInitial\", \"YearFinal\"]:\n",
    "                    doc[k] = int(v) if v else v\n",
    "                elif k == \"pubmed\" and v:\n",
    "                    doc[k].add(int(v))\n",
    "        doc[\"pubmed\"] = list(doc[\"pubmed\"])\n",
    "        d[grp[0].replace(\"umls\", \"umls_cui\")].append(doc)\n",
    "    return d"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-03-05T04:02:48.988926Z",
     "start_time": "2021-03-05T03:44:17.592993Z"
    }
   },
   "outputs": [],
   "source": [
    "processed1 = process_gene(allgene_pmid_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-03-05T19:50:38.532170Z",
     "start_time": "2021-03-05T19:50:38.528169Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "7473"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(processed1['C0678222'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## HPO parsing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-20T01:25:51.261921Z",
     "start_time": "2021-04-20T01:25:51.259241Z"
    }
   },
   "outputs": [],
   "source": [
    "## put the path you want to use here\n",
    "HPO_path = pathlib.Path.home().joinpath('Desktop', 'ScrippsJob', 'phenotype.hpoa')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-20T01:25:52.419824Z",
     "start_time": "2021-04-20T01:25:52.395760Z"
    }
   },
   "outputs": [],
   "source": [
    "def process_disease2hp(file_path_disease_hpo):\n",
    "    df_disease_hpo = pd.read_csv(file_path_disease_hpo, sep=\"\\t\", skiprows=4, dtype=str)\n",
    "    df_disease_hpo = df_disease_hpo.rename(\n",
    "        index=str, columns={\"DiseaseName\": \"disease_name\", \"#DatabaseID\": \"disease_id\"}\n",
    "    )\n",
    "    \n",
    "    ## removing qualifier = 'NOT' annotations, because it means the disease does not\n",
    "    ##   have this phenotypic feature. The HPO website doesn't show these 'NOT' annots\n",
    "    df_disease_hpo = df_disease_hpo[df_disease_hpo['Qualifier'] != \"NOT\"]\n",
    "    ## then remove the qualifier\n",
    "    df_disease_hpo.drop(columns = 'Qualifier', inplace = True)\n",
    "    ## make sure all null values are None\n",
    "    df_disease_hpo = df_disease_hpo.where((pd.notnull(df_disease_hpo)), None)\n",
    "    \n",
    "    d = []\n",
    "    for did, subdf in df_disease_hpo.groupby(\"disease_id\"):\n",
    "        did = did.replace(\"ORPHA\", \"ORPHANET\")\n",
    "        records = subdf.to_dict(orient=\"records\")\n",
    "        \n",
    "        pathway_related = []\n",
    "        course = []\n",
    "        modifiers = []\n",
    "        inheritance = []\n",
    "        \n",
    "        for record in records:\n",
    "            record_dict = {}\n",
    "            if record[\"Aspect\"] == \"C\":\n",
    "                course.append(record[\"HPO_ID\"])\n",
    "                continue\n",
    "            elif record[\"Aspect\"] == \"M\":\n",
    "                modifiers.append(record[\"HPO_ID\"])\n",
    "                continue\n",
    "            elif record[\"Aspect\"] == \"I\":\n",
    "                inheritance.append(record[\"HPO_ID\"])\n",
    "                continue      \n",
    "                \n",
    "            for k, v in record.items():\n",
    "                # name the field based on pathway database\n",
    "                if (k == \"Sex\") and v:\n",
    "                    record_dict['sex'] = v.lower()\n",
    "                elif (k == 'Reference') and v: \n",
    "                ## only process if Reference has a value\n",
    "                ## notes: OMIM:194190, OMIM:180849, OMIM:212050 are disease examples with > 1 type of reference\n",
    "                    ## this is a string representing a list\n",
    "                    tempRefs = v.split(\";\")\n",
    "                    ## prepare to iterate through the tempRefs and store the processed data\n",
    "                    tempProperties = {\n",
    "                        'ISBN': [],\n",
    "                        'PMID': [],\n",
    "                        'http': [],\n",
    "                        'DECIPHER': [],\n",
    "                        'OMIM': [],\n",
    "                        'ORPHA': []\n",
    "                    }\n",
    "                    \n",
    "                    ## remove the prefixes or not? currently keeping the prefix\n",
    "                    for i in tempRefs:\n",
    "                        for key in tempProperties.keys():\n",
    "                            if key in i:\n",
    "                                ## replace curie prefix for isbn and orpha\n",
    "                                if key == 'ISBN':\n",
    "                                    tempProperties[key].append('ISBN:' + i.split(\":\")[1])\n",
    "                                elif key == 'ORPHA':\n",
    "                                    tempProperties[key].append('ORPHANET:' + i.split(\":\")[1])                                    \n",
    "                                else:\n",
    "                                    tempProperties[key].append(i)\n",
    "                    ## ONLY add reference keys/values to the record if there are values\n",
    "                    for k,v in tempProperties.items():\n",
    "                        if v:\n",
    "                            if k == 'ISBN':\n",
    "                                record_dict['isbn_refs'] = v\n",
    "                            elif k == 'PMID':\n",
    "                                record_dict['pmid_refs'] = v                    \n",
    "                            elif k == 'http':\n",
    "                                record_dict['website_refs'] = v  \n",
    "                            elif k == 'DECIPHER':\n",
    "                                record_dict['decipher_refs'] = v  \n",
    "                            elif k == 'OMIM':\n",
    "                                record_dict['omim_refs'] = v  \n",
    "                            elif k == 'ORPHA':\n",
    "                                record_dict['orphanet_refs'] = v  \n",
    "                elif (k == 'Frequency') and v:\n",
    "                ## only process if Frequency has a value\n",
    "                    tempDict = {}\n",
    "                    if 'http' in v:  ## catching an error in the data\n",
    "                        continue\n",
    "                    elif 'HP:' in v:\n",
    "                        tempDict['hp_freq'] = v\n",
    "                    elif '%' in v:\n",
    "                        tempFreq = float(v.strip('%')) / 100\n",
    "                        ## only go forward if this is a valid fraction <=1\n",
    "                        if tempFreq <= 1:\n",
    "                            tempDict['numeric_freq'] = tempFreq\n",
    "                    elif '/' in v:\n",
    "                        ## idx 0 is numerator, idx 1 is denominator\n",
    "                        tempL = [int(ele) for ele in v.split(\"/\")]\n",
    "                        ## only go forward if this is a valid fraction <=1\n",
    "                        if (tempL[0] != 0) and (tempL[1] !=0) and (tempL[0] <= tempL[1]):\n",
    "                            tempDict['freq_numerator'] = tempL[0]\n",
    "                            tempDict['freq_denominator'] = tempL[1]\n",
    "                            tempDict['numeric_freq'] = tempL[0] / tempL[1]\n",
    "\n",
    "                    ## ONLY add frequency keys/values to the record if there are values\n",
    "                    if tempDict:\n",
    "                        record_dict.update(tempDict)\n",
    "                        \n",
    "                elif (k == 'Modifier') and v:\n",
    "                ## only process if Modifier has a value\n",
    "                    ## in <20 records, this is a delimited list with repeated values\n",
    "                    ## this behavior matches the unlist behavior used with biothings APIs\n",
    "                    ## https://github.com/kevinxin90/biothings.api/blob/master/biothings/utils/dataload.py\n",
    "                    if \";\" in v:\n",
    "                        ## transform to list -> set->list to remove repeated values\n",
    "                        tempMods = list(set(v.split(\";\")))\n",
    "                        record_dict['modifier'] = tempMods\n",
    "                    else:\n",
    "                        record_dict['modifier'] = v\n",
    "                elif k not in {\"disease_id\", \"disease_name\", \n",
    "                               \"Aspect\", \"Sex\", \n",
    "                               \"Reference\", \"Frequency\", \n",
    "                               \"Modifier\"}:\n",
    "                    record_dict[k.lower()] = v\n",
    "            pathway_related.append(record_dict)\n",
    "        drecord = {\n",
    "            \"_id\": did,\n",
    "            \"hpo\": pathway_related,\n",
    "            \"disease_name\": records[0][\"disease_name\"],\n",
    "            \"course\": course,\n",
    "            \"modifiers\": modifiers,\n",
    "            \"inheritance\": inheritance\n",
    "        }\n",
    "        d.append(drecord)\n",
    " \n",
    "    return {\n",
    "        x[\"_id\"]: [x[\"hpo\"], x[\"disease_name\"], x[\"course\"], x[\"modifiers\"], x[\"inheritance\"]] for x in d\n",
    "    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-20T01:26:09.988656Z",
     "start_time": "2021-04-20T01:25:56.224947Z"
    }
   },
   "outputs": [],
   "source": [
    "processed1 = process_disease2hp(HPO_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-20T01:27:41.383337Z",
     "start_time": "2021-04-20T01:27:41.374723Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "30"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "'CRANIOFACIAL DYSMORPHISM WITH OCULAR COLOBOMA, ABSENT CORPUS CALLOSUM,AND AORTIC DILATATION'"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "25"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "'Temtamy syndrome'"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## these are actually the same disease but obviously two diff sets of results. \n",
    "## see https://www.orpha.net/consor/cgi-bin/OC_Exp.php?lng=EN&Expert=1777 and its id mapping\n",
    "## mydisease resolves these two as one entity as well\n",
    "len(processed1['OMIM:218340'][0])\n",
    "processed1['OMIM:218340'][1]\n",
    "len(processed1['ORPHANET:1777'][0])\n",
    "processed1['ORPHANET:1777'][1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-04-20T01:27:02.572049Z",
     "start_time": "2021-04-20T01:27:02.560800Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "25"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "[[{'hpo_id': 'HP:0000174',\n",
       "   'orphanet_refs': ['ORPHANET:1777'],\n",
       "   'evidence': 'TAS',\n",
       "   'onset': None,\n",
       "   'hp_freq': 'HP:0040283',\n",
       "   'biocuration': 'ORPHA:orphadata[2021-02-08]'},\n",
       "  {'hpo_id': 'HP:0000179',\n",
       "   'orphanet_refs': ['ORPHANET:1777'],\n",
       "   'evidence': 'TAS',\n",
       "   'onset': None,\n",
       "   'hp_freq': 'HP:0040283',\n",
       "   'biocuration': 'ORPHA:orphadata[2021-02-08]'},\n",
       "  {'hpo_id': 'HP:0000256',\n",
       "   'orphanet_refs': ['ORPHANET:1777'],\n",
       "   'evidence': 'TAS',\n",
       "   'onset': None,\n",
       "   'hp_freq': 'HP:0040282',\n",
       "   'biocuration': 'ORPHA:orphadata[2021-02-08]'},\n",
       "  {'hpo_id': 'HP:0000268',\n",
       "   'orphanet_refs': ['ORPHANET:1777'],\n",
       "   'evidence': 'TAS',\n",
       "   'onset': None,\n",
       "   'hp_freq': 'HP:0040282',\n",
       "   'biocuration': 'ORPHA:orphadata[2021-02-08]'},\n",
       "  {'hpo_id': 'HP:0000276',\n",
       "   'orphanet_refs': ['ORPHANET:1777'],\n",
       "   'evidence': 'TAS',\n",
       "   'onset': None,\n",
       "   'hp_freq': 'HP:0040282',\n",
       "   'biocuration': 'ORPHA:orphadata[2021-02-08]'},\n",
       "  {'hpo_id': 'HP:0000280',\n",
       "   'orphanet_refs': ['ORPHANET:1777'],\n",
       "   'evidence': 'TAS',\n",
       "   'onset': None,\n",
       "   'hp_freq': 'HP:0040282',\n",
       "   'biocuration': 'ORPHA:orphadata[2021-02-08]'},\n",
       "  {'hpo_id': 'HP:0000316',\n",
       "   'orphanet_refs': ['ORPHANET:1777'],\n",
       "   'evidence': 'TAS',\n",
       "   'onset': None,\n",
       "   'hp_freq': 'HP:0040281',\n",
       "   'biocuration': 'ORPHA:orphadata[2021-02-08]'},\n",
       "  {'hpo_id': 'HP:0000324',\n",
       "   'orphanet_refs': ['ORPHANET:1777'],\n",
       "   'evidence': 'TAS',\n",
       "   'onset': None,\n",
       "   'hp_freq': 'HP:0040283',\n",
       "   'biocuration': 'ORPHA:orphadata[2021-02-08]'},\n",
       "  {'hpo_id': 'HP:0000347',\n",
       "   'orphanet_refs': ['ORPHANET:1777'],\n",
       "   'evidence': 'TAS',\n",
       "   'onset': None,\n",
       "   'hp_freq': 'HP:0040282',\n",
       "   'biocuration': 'ORPHA:orphadata[2021-02-08]'},\n",
       "  {'hpo_id': 'HP:0000369',\n",
       "   'orphanet_refs': ['ORPHANET:1777'],\n",
       "   'evidence': 'TAS',\n",
       "   'onset': None,\n",
       "   'hp_freq': 'HP:0040282',\n",
       "   'biocuration': 'ORPHA:orphadata[2021-02-08]'},\n",
       "  {'hpo_id': 'HP:0000444',\n",
       "   'orphanet_refs': ['ORPHANET:1777'],\n",
       "   'evidence': 'TAS',\n",
       "   'onset': None,\n",
       "   'hp_freq': 'HP:0040282',\n",
       "   'biocuration': 'ORPHA:orphadata[2021-02-08]'},\n",
       "  {'hpo_id': 'HP:0000506',\n",
       "   'orphanet_refs': ['ORPHANET:1777'],\n",
       "   'evidence': 'TAS',\n",
       "   'onset': None,\n",
       "   'hp_freq': 'HP:0040283',\n",
       "   'biocuration': 'ORPHA:orphadata[2021-02-08]'},\n",
       "  {'hpo_id': 'HP:0000567',\n",
       "   'orphanet_refs': ['ORPHANET:1777'],\n",
       "   'evidence': 'TAS',\n",
       "   'onset': None,\n",
       "   'hp_freq': 'HP:0040281',\n",
       "   'biocuration': 'ORPHA:orphadata[2021-02-08]'},\n",
       "  {'hpo_id': 'HP:0000568',\n",
       "   'orphanet_refs': ['ORPHANET:1777'],\n",
       "   'evidence': 'TAS',\n",
       "   'onset': None,\n",
       "   'hp_freq': 'HP:0040283',\n",
       "   'biocuration': 'ORPHA:orphadata[2021-02-08]'},\n",
       "  {'hpo_id': 'HP:0000612',\n",
       "   'orphanet_refs': ['ORPHANET:1777'],\n",
       "   'evidence': 'TAS',\n",
       "   'onset': None,\n",
       "   'hp_freq': 'HP:0040281',\n",
       "   'biocuration': 'ORPHA:orphadata[2021-02-08]'},\n",
       "  {'hpo_id': 'HP:0001156',\n",
       "   'orphanet_refs': ['ORPHANET:1777'],\n",
       "   'evidence': 'TAS',\n",
       "   'onset': None,\n",
       "   'hp_freq': 'HP:0040281',\n",
       "   'biocuration': 'ORPHA:orphadata[2021-02-08]'},\n",
       "  {'hpo_id': 'HP:0001249',\n",
       "   'orphanet_refs': ['ORPHANET:1777'],\n",
       "   'evidence': 'TAS',\n",
       "   'onset': None,\n",
       "   'hp_freq': 'HP:0040281',\n",
       "   'biocuration': 'ORPHA:orphadata[2021-02-08]'},\n",
       "  {'hpo_id': 'HP:0001263',\n",
       "   'orphanet_refs': ['ORPHANET:1777'],\n",
       "   'evidence': 'TAS',\n",
       "   'onset': None,\n",
       "   'hp_freq': 'HP:0040281',\n",
       "   'biocuration': 'ORPHA:orphadata[2021-02-08]'},\n",
       "  {'hpo_id': 'HP:0004942',\n",
       "   'orphanet_refs': ['ORPHANET:1777'],\n",
       "   'evidence': 'TAS',\n",
       "   'onset': None,\n",
       "   'hp_freq': 'HP:0040282',\n",
       "   'biocuration': 'ORPHA:orphadata[2021-02-08]'},\n",
       "  {'hpo_id': 'HP:0001763',\n",
       "   'orphanet_refs': ['ORPHANET:1777'],\n",
       "   'evidence': 'TAS',\n",
       "   'onset': None,\n",
       "   'hp_freq': 'HP:0040282',\n",
       "   'biocuration': 'ORPHA:orphadata[2021-02-08]'},\n",
       "  {'hpo_id': 'HP:0001831',\n",
       "   'orphanet_refs': ['ORPHANET:1777'],\n",
       "   'evidence': 'TAS',\n",
       "   'onset': None,\n",
       "   'hp_freq': 'HP:0040281',\n",
       "   'biocuration': 'ORPHA:orphadata[2021-02-08]'},\n",
       "  {'hpo_id': 'HP:0002970',\n",
       "   'orphanet_refs': ['ORPHANET:1777'],\n",
       "   'evidence': 'TAS',\n",
       "   'onset': None,\n",
       "   'hp_freq': 'HP:0040282',\n",
       "   'biocuration': 'ORPHA:orphadata[2021-02-08]'},\n",
       "  {'hpo_id': 'HP:0004209',\n",
       "   'orphanet_refs': ['ORPHANET:1777'],\n",
       "   'evidence': 'TAS',\n",
       "   'onset': None,\n",
       "   'hp_freq': 'HP:0040283',\n",
       "   'biocuration': 'ORPHA:orphadata[2021-02-08]'},\n",
       "  {'hpo_id': 'HP:0005692',\n",
       "   'orphanet_refs': ['ORPHANET:1777'],\n",
       "   'evidence': 'TAS',\n",
       "   'onset': None,\n",
       "   'hp_freq': 'HP:0040283',\n",
       "   'biocuration': 'ORPHA:orphadata[2021-02-08]'},\n",
       "  {'hpo_id': 'HP:0007370',\n",
       "   'orphanet_refs': ['ORPHANET:1777'],\n",
       "   'evidence': 'TAS',\n",
       "   'onset': None,\n",
       "   'hp_freq': 'HP:0040281',\n",
       "   'biocuration': 'ORPHA:orphadata[2021-02-08]'}],\n",
       " 'Temtamy syndrome',\n",
       " [],\n",
       " [],\n",
       " []]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-17T22:08:36.994947Z",
     "start_time": "2021-02-17T22:08:36.990184Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[[{'hpo_id': 'HP:0500043',\n",
       "   'pmid_refs': ['PMID:3767680'],\n",
       "   'omim_refs': ['OMIM:151610'],\n",
       "   'evidence': 'IEA',\n",
       "   'onset': None,\n",
       "   'freq_numerator': 20,\n",
       "   'freq_denominator': 20,\n",
       "   'numeric_freq': 1.0,\n",
       "   'biocuration': 'HPO:lccarmody[2018-10-05];HPO:skoehler[2018-10-08]'},\n",
       "  {'hpo_id': 'HP:0000492',\n",
       "   'pmid_refs': ['PMID:3767680'],\n",
       "   'evidence': 'IEA',\n",
       "   'onset': None,\n",
       "   'biocuration': 'HPO:lccarmody[2018-10-05]'},\n",
       "  {'hpo_id': 'HP:0000508',\n",
       "   'pmid_refs': ['PMID:3767680'],\n",
       "   'evidence': 'PCS',\n",
       "   'onset': None,\n",
       "   'biocuration': 'HPO:lccarmody[2018-10-05]'},\n",
       "  {'hpo_id': 'HP:0000478',\n",
       "   'omim_refs': ['OMIM:151610'],\n",
       "   'evidence': 'IEA',\n",
       "   'onset': None,\n",
       "   'biocuration': 'HPO:iea[2009-02-17]'}],\n",
       " 'LEVATOR-MEDIAL RECTUS SYNKINESIS',\n",
       " [],\n",
       " [],\n",
       " ['HP:0000006']]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## CHECKS:\n",
    "## decipher refs: 'DECIPHER:72', orphanet refs: ORPHANET:221 (none of these are in situations with multiple refs)\n",
    "## omim and pmid refs in the same disease -> pheno association: 'OMIM:151610'\n",
    "## combos of omim, isbn, website refs: 'OMIM:194190', OMIM:180849\n",
    "\n",
    "processed1['OMIM:151610']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-12T20:55:46.627425Z",
     "start_time": "2021-02-12T20:55:46.294121Z"
    }
   },
   "outputs": [],
   "source": [
    "rawHPO = pd.read_csv(HPO_path, sep=\"\\t\", skiprows=4, dtype=str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-12T20:57:19.904767Z",
     "start_time": "2021-02-12T20:57:19.867275Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#DatabaseID</th>\n",
       "      <th>DiseaseName</th>\n",
       "      <th>Qualifier</th>\n",
       "      <th>HPO_ID</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Evidence</th>\n",
       "      <th>Onset</th>\n",
       "      <th>Frequency</th>\n",
       "      <th>Sex</th>\n",
       "      <th>Modifier</th>\n",
       "      <th>Aspect</th>\n",
       "      <th>Biocuration</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>41673</th>\n",
       "      <td>OMIM:191900</td>\n",
       "      <td>Muckle-Wells syndrome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0030953</td>\n",
       "      <td>PMID:25766347</td>\n",
       "      <td>PCS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0025303</td>\n",
       "      <td>P</td>\n",
       "      <td>HPO:probinson[2020-11-17]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41674</th>\n",
       "      <td>OMIM:191900</td>\n",
       "      <td>Muckle-Wells syndrome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0003565</td>\n",
       "      <td>PMID:25766347</td>\n",
       "      <td>TAS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0025303</td>\n",
       "      <td>P</td>\n",
       "      <td>HPO:probinson[2009-02-17];HPO:probinson[2020-1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41675</th>\n",
       "      <td>OMIM:191900</td>\n",
       "      <td>Muckle-Wells syndrome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0003621</td>\n",
       "      <td>PMID:25766347</td>\n",
       "      <td>PCS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>HPO:probinson[2020-11-17]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41676</th>\n",
       "      <td>OMIM:191900</td>\n",
       "      <td>Muckle-Wells syndrome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0000365</td>\n",
       "      <td>PMID:14872505</td>\n",
       "      <td>PCS</td>\n",
       "      <td>HP:0011463</td>\n",
       "      <td>3/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P</td>\n",
       "      <td>HPO:skoehler[2010-06-20];HPO:probinson[2020-11...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41677</th>\n",
       "      <td>OMIM:191900</td>\n",
       "      <td>Muckle-Wells syndrome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0005764</td>\n",
       "      <td>PMID:25766347</td>\n",
       "      <td>PCS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0025303</td>\n",
       "      <td>P</td>\n",
       "      <td>HPO:probinson[2020-11-17]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41678</th>\n",
       "      <td>OMIM:191900</td>\n",
       "      <td>Muckle-Wells syndrome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0004322</td>\n",
       "      <td>PMID:14872505</td>\n",
       "      <td>PCS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P</td>\n",
       "      <td>HPO:probinson[2020-11-17]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41679</th>\n",
       "      <td>OMIM:191900</td>\n",
       "      <td>Muckle-Wells syndrome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0000083</td>\n",
       "      <td>PMID:9704852</td>\n",
       "      <td>PCS</td>\n",
       "      <td>HP:0003584</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P</td>\n",
       "      <td>HPO:iea[2009-02-17];HPO:probinson[2020-11-17]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41680</th>\n",
       "      <td>OMIM:191900</td>\n",
       "      <td>Muckle-Wells syndrome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0011107</td>\n",
       "      <td>OMIM:191900</td>\n",
       "      <td>TAS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P</td>\n",
       "      <td>HPO:skoehler[2012-10-17]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41681</th>\n",
       "      <td>OMIM:191900</td>\n",
       "      <td>Muckle-Wells syndrome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0011227</td>\n",
       "      <td>PMID:25766347</td>\n",
       "      <td>PCS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P</td>\n",
       "      <td>HPO:probinson[2020-11-17]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41682</th>\n",
       "      <td>OMIM:191900</td>\n",
       "      <td>Muckle-Wells syndrome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0012432</td>\n",
       "      <td>PMID:14872505</td>\n",
       "      <td>PCS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P</td>\n",
       "      <td>HPO:probinson[2020-11-17]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41683</th>\n",
       "      <td>OMIM:191900</td>\n",
       "      <td>Muckle-Wells syndrome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0002829</td>\n",
       "      <td>PMID:14872505</td>\n",
       "      <td>PCS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0025206;HP:0025303</td>\n",
       "      <td>P</td>\n",
       "      <td>HPO:probinson[2009-02-17];HPO:probinson[2020-1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41684</th>\n",
       "      <td>OMIM:191900</td>\n",
       "      <td>Muckle-Wells syndrome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0000408</td>\n",
       "      <td>PMID:25766347</td>\n",
       "      <td>PCS</td>\n",
       "      <td>HP:0011462</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P</td>\n",
       "      <td>HPO:probinson[2009-02-17];HPO:probinson[2020-1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41685</th>\n",
       "      <td>OMIM:191900</td>\n",
       "      <td>Muckle-Wells syndrome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0011463</td>\n",
       "      <td>PMID:25766347</td>\n",
       "      <td>PCS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>HPO:probinson[2020-11-17]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41686</th>\n",
       "      <td>OMIM:191900</td>\n",
       "      <td>Muckle-Wells syndrome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0100759</td>\n",
       "      <td>PMID:25766347</td>\n",
       "      <td>PCS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P</td>\n",
       "      <td>HPO:probinson[2020-11-17]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41687</th>\n",
       "      <td>OMIM:191900</td>\n",
       "      <td>Muckle-Wells syndrome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0040186</td>\n",
       "      <td>PMID:25766347;PMID:14872505</td>\n",
       "      <td>PCS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5/5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0025303</td>\n",
       "      <td>P</td>\n",
       "      <td>HPO:probinson[2020-11-17]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41688</th>\n",
       "      <td>OMIM:191900</td>\n",
       "      <td>Muckle-Wells syndrome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0000509</td>\n",
       "      <td>PMID:14872505</td>\n",
       "      <td>PCS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0025206;HP:0025303</td>\n",
       "      <td>P</td>\n",
       "      <td>HPO:probinson[2009-02-17];HPO:probinson[2020-1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41689</th>\n",
       "      <td>OMIM:191900</td>\n",
       "      <td>Muckle-Wells syndrome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0001917</td>\n",
       "      <td>PMID:9704852</td>\n",
       "      <td>PCS</td>\n",
       "      <td>HP:0003584</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P</td>\n",
       "      <td>HPO:probinson[2009-02-17];HPO:probinson[2020-1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41690</th>\n",
       "      <td>OMIM:191900</td>\n",
       "      <td>Muckle-Wells syndrome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0001954</td>\n",
       "      <td>PMID:25766347;PMID:14872505</td>\n",
       "      <td>PCS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5/5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0025206;HP:0025303</td>\n",
       "      <td>P</td>\n",
       "      <td>HPO:probinson[2020-11-17];HPO:iea[2009-02-17];...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41691</th>\n",
       "      <td>OMIM:191900</td>\n",
       "      <td>Muckle-Wells syndrome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0000006</td>\n",
       "      <td>PMID:11687797</td>\n",
       "      <td>PCS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>I</td>\n",
       "      <td>HPO:iea[2009-02-17];HPO:probinson[2020-11-17]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41692</th>\n",
       "      <td>OMIM:191900</td>\n",
       "      <td>Muckle-Wells syndrome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0001974</td>\n",
       "      <td>PMID:25766347</td>\n",
       "      <td>PCS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P</td>\n",
       "      <td>HPO:skoehler[2010-06-20];HPO:probinson[2020-11...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41693</th>\n",
       "      <td>OMIM:191900</td>\n",
       "      <td>Muckle-Wells syndrome</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0003326</td>\n",
       "      <td>OMIM:191900</td>\n",
       "      <td>TAS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HP:0025303</td>\n",
       "      <td>P</td>\n",
       "      <td>HPO:probinson[2009-02-17]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       #DatabaseID            DiseaseName Qualifier      HPO_ID  \\\n",
       "41673  OMIM:191900  Muckle-Wells syndrome       NaN  HP:0030953   \n",
       "41674  OMIM:191900  Muckle-Wells syndrome       NaN  HP:0003565   \n",
       "41675  OMIM:191900  Muckle-Wells syndrome       NaN  HP:0003621   \n",
       "41676  OMIM:191900  Muckle-Wells syndrome       NaN  HP:0000365   \n",
       "41677  OMIM:191900  Muckle-Wells syndrome       NaN  HP:0005764   \n",
       "41678  OMIM:191900  Muckle-Wells syndrome       NaN  HP:0004322   \n",
       "41679  OMIM:191900  Muckle-Wells syndrome       NaN  HP:0000083   \n",
       "41680  OMIM:191900  Muckle-Wells syndrome       NaN  HP:0011107   \n",
       "41681  OMIM:191900  Muckle-Wells syndrome       NaN  HP:0011227   \n",
       "41682  OMIM:191900  Muckle-Wells syndrome       NaN  HP:0012432   \n",
       "41683  OMIM:191900  Muckle-Wells syndrome       NaN  HP:0002829   \n",
       "41684  OMIM:191900  Muckle-Wells syndrome       NaN  HP:0000408   \n",
       "41685  OMIM:191900  Muckle-Wells syndrome       NaN  HP:0011463   \n",
       "41686  OMIM:191900  Muckle-Wells syndrome       NaN  HP:0100759   \n",
       "41687  OMIM:191900  Muckle-Wells syndrome       NaN  HP:0040186   \n",
       "41688  OMIM:191900  Muckle-Wells syndrome       NaN  HP:0000509   \n",
       "41689  OMIM:191900  Muckle-Wells syndrome       NaN  HP:0001917   \n",
       "41690  OMIM:191900  Muckle-Wells syndrome       NaN  HP:0001954   \n",
       "41691  OMIM:191900  Muckle-Wells syndrome       NaN  HP:0000006   \n",
       "41692  OMIM:191900  Muckle-Wells syndrome       NaN  HP:0001974   \n",
       "41693  OMIM:191900  Muckle-Wells syndrome       NaN  HP:0003326   \n",
       "\n",
       "                         Reference Evidence       Onset Frequency  Sex  \\\n",
       "41673                PMID:25766347      PCS         NaN       2/2  NaN   \n",
       "41674                PMID:25766347      TAS         NaN       2/2  NaN   \n",
       "41675                PMID:25766347      PCS         NaN       1/2  NaN   \n",
       "41676                PMID:14872505      PCS  HP:0011463       3/3  NaN   \n",
       "41677                PMID:25766347      PCS         NaN       2/2  NaN   \n",
       "41678                PMID:14872505      PCS         NaN       3/3  NaN   \n",
       "41679                 PMID:9704852      PCS  HP:0003584       NaN  NaN   \n",
       "41680                  OMIM:191900      TAS         NaN       NaN  NaN   \n",
       "41681                PMID:25766347      PCS         NaN       2/2  NaN   \n",
       "41682                PMID:14872505      PCS         NaN       3/3  NaN   \n",
       "41683                PMID:14872505      PCS         NaN       3/3  NaN   \n",
       "41684                PMID:25766347      PCS  HP:0011462       2/2  NaN   \n",
       "41685                PMID:25766347      PCS         NaN       NaN  NaN   \n",
       "41686                PMID:25766347      PCS         NaN       2/2  NaN   \n",
       "41687  PMID:25766347;PMID:14872505      PCS         NaN       5/5  NaN   \n",
       "41688                PMID:14872505      PCS         NaN       3/3  NaN   \n",
       "41689                 PMID:9704852      PCS  HP:0003584       NaN  NaN   \n",
       "41690  PMID:25766347;PMID:14872505      PCS         NaN       5/5  NaN   \n",
       "41691                PMID:11687797      PCS         NaN       NaN  NaN   \n",
       "41692                PMID:25766347      PCS         NaN       NaN  NaN   \n",
       "41693                  OMIM:191900      TAS         NaN       NaN  NaN   \n",
       "\n",
       "                    Modifier Aspect  \\\n",
       "41673             HP:0025303      P   \n",
       "41674             HP:0025303      P   \n",
       "41675                    NaN      C   \n",
       "41676                    NaN      P   \n",
       "41677             HP:0025303      P   \n",
       "41678                    NaN      P   \n",
       "41679                    NaN      P   \n",
       "41680                    NaN      P   \n",
       "41681                    NaN      P   \n",
       "41682                    NaN      P   \n",
       "41683  HP:0025206;HP:0025303      P   \n",
       "41684                    NaN      P   \n",
       "41685                    NaN      C   \n",
       "41686                    NaN      P   \n",
       "41687             HP:0025303      P   \n",
       "41688  HP:0025206;HP:0025303      P   \n",
       "41689                    NaN      P   \n",
       "41690  HP:0025206;HP:0025303      P   \n",
       "41691                    NaN      I   \n",
       "41692                    NaN      P   \n",
       "41693             HP:0025303      P   \n",
       "\n",
       "                                             Biocuration  \n",
       "41673                          HPO:probinson[2020-11-17]  \n",
       "41674  HPO:probinson[2009-02-17];HPO:probinson[2020-1...  \n",
       "41675                          HPO:probinson[2020-11-17]  \n",
       "41676  HPO:skoehler[2010-06-20];HPO:probinson[2020-11...  \n",
       "41677                          HPO:probinson[2020-11-17]  \n",
       "41678                          HPO:probinson[2020-11-17]  \n",
       "41679      HPO:iea[2009-02-17];HPO:probinson[2020-11-17]  \n",
       "41680                           HPO:skoehler[2012-10-17]  \n",
       "41681                          HPO:probinson[2020-11-17]  \n",
       "41682                          HPO:probinson[2020-11-17]  \n",
       "41683  HPO:probinson[2009-02-17];HPO:probinson[2020-1...  \n",
       "41684  HPO:probinson[2009-02-17];HPO:probinson[2020-1...  \n",
       "41685                          HPO:probinson[2020-11-17]  \n",
       "41686                          HPO:probinson[2020-11-17]  \n",
       "41687                          HPO:probinson[2020-11-17]  \n",
       "41688  HPO:probinson[2009-02-17];HPO:probinson[2020-1...  \n",
       "41689  HPO:probinson[2009-02-17];HPO:probinson[2020-1...  \n",
       "41690  HPO:probinson[2020-11-17];HPO:iea[2009-02-17];...  \n",
       "41691      HPO:iea[2009-02-17];HPO:probinson[2020-11-17]  \n",
       "41692  HPO:skoehler[2010-06-20];HPO:probinson[2020-11...  \n",
       "41693                          HPO:probinson[2009-02-17]  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rawHPO[rawHPO['#DatabaseID'] == 'OMIM:191900']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## CTD troubleshooting"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "### disease -> chemical"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-16T02:10:50.909051Z",
     "start_time": "2021-02-16T02:10:50.906240Z"
    },
    "hidden": true
   },
   "outputs": [],
   "source": [
    "CTD_path1 = pathlib.Path.home().joinpath('Downloads', 'CTD_chemicals_diseases.csv.gz')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Changed to only keep the direct annotations, and for the chunking to work correctly (not override the entries) and to deal with typing issues for the columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-16T02:16:25.543062Z",
     "start_time": "2021-02-16T02:16:25.534503Z"
    },
    "hidden": true
   },
   "outputs": [],
   "source": [
    "def process_chemical(file_path_chemical):\n",
    "    chunksize = 100000\n",
    "    d = []\n",
    "    for chunk in pd.read_csv(file_path_chemical, chunksize=chunksize, sep=',', comment='#', compression='gzip', \n",
    "                             names=['chemical_name', 'mesh_chemical_id', 'cas_registry_number', 'DiseaseName', 'DiseaseID', 'direct_evidence', 'inference_gene_symbol', 'inference_score', 'omim_id', 'pubmed'],\n",
    "                             dtype=str):\n",
    "        temp_chunk = chunk.copy()\n",
    "        temp_chunk = temp_chunk.where((pd.notnull(temp_chunk)), None)\n",
    "        ## remove all inferred annotations\n",
    "        temp_chunk = temp_chunk[~ temp_chunk['direct_evidence'].isna()]\n",
    "        ## only work with records if the dataframe still has records \n",
    "        if not temp_chunk.empty: \n",
    "            ## make this the correct type\n",
    "            temp_chunk['inference_score'] = temp_chunk['inference_score'].astype(float)\n",
    "            temp_chunk = temp_chunk.where((pd.notnull(temp_chunk)), None)\n",
    "            # add new column called source\n",
    "            temp_chunk['source'] = 'CTD'         \n",
    "            # the record in these fields are separated by '|', need to convert them into list\n",
    "            for field_id in ['omim_id', 'pubmed']:\n",
    "                temp_chunk[field_id] = temp_chunk[field_id].apply(lambda x: x.split('|') if x and '|' in x else x)\n",
    "            for did, subdf in temp_chunk.groupby('DiseaseID'):\n",
    "                records = subdf.to_dict(orient='records')\n",
    "                chemical_related = [{k: v for k, v in record.items() if k not in {'DiseaseName', 'DiseaseID'}} for record in records]            \n",
    "                drecord = {'_id': did, 'chemical': chemical_related}\n",
    "                d.append(drecord)\n",
    "    finalDict = {}\n",
    "    ## For now, I'm not merging records. Current data situation is separate records when relationship is marker/mechanism AND therapeutic\n",
    "    for ele in d:\n",
    "        tempID = ele['_id']\n",
    "        ## if an entry for this disease already exists in the dictionary\n",
    "        if tempID in finalDict.keys():\n",
    "            finalDict[tempID] = finalDict[tempID] + ele['chemical']\n",
    "        else:\n",
    "            finalDict[tempID] = ele['chemical']\n",
    "    return finalDict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-16T02:17:20.392304Z",
     "start_time": "2021-02-16T02:16:29.179548Z"
    },
    "hidden": true
   },
   "outputs": [],
   "source": [
    "trial1 = process_chemical(CTD_path1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-16T02:17:24.041078Z",
     "start_time": "2021-02-16T02:17:24.027998Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'chemical_name': 'Amitriptyline',\n",
       "  'mesh_chemical_id': 'D000639',\n",
       "  'cas_registry_number': '50-48-6',\n",
       "  'direct_evidence': 'marker/mechanism',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': '7123337',\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Angiotensin-Converting Enzyme Inhibitors',\n",
       "  'mesh_chemical_id': 'D000806',\n",
       "  'cas_registry_number': None,\n",
       "  'direct_evidence': 'therapeutic',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': ['17541236', '6168412'],\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Atenolol',\n",
       "  'mesh_chemical_id': 'D001262',\n",
       "  'cas_registry_number': '29122-68-7',\n",
       "  'direct_evidence': 'therapeutic',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': '2493837',\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Atrasentan',\n",
       "  'mesh_chemical_id': 'D000077868',\n",
       "  'cas_registry_number': None,\n",
       "  'direct_evidence': 'therapeutic',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': '19326569',\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Bumetanide',\n",
       "  'mesh_chemical_id': 'D002034',\n",
       "  'cas_registry_number': '28395-03-1',\n",
       "  'direct_evidence': 'therapeutic',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': '6112544',\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Caffeine',\n",
       "  'mesh_chemical_id': 'D002110',\n",
       "  'cas_registry_number': '58-08-2',\n",
       "  'direct_evidence': 'marker/mechanism',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': '3020089',\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Chlorthalidone',\n",
       "  'mesh_chemical_id': 'D002752',\n",
       "  'cas_registry_number': '77-36-1',\n",
       "  'direct_evidence': 'therapeutic',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': '2493837',\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Cocaine',\n",
       "  'mesh_chemical_id': 'D003042',\n",
       "  'cas_registry_number': '50-36-2',\n",
       "  'direct_evidence': 'marker/mechanism',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': ['18081441', '8969619'],\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Contraceptives, Oral',\n",
       "  'mesh_chemical_id': 'D003276',\n",
       "  'cas_registry_number': None,\n",
       "  'direct_evidence': 'marker/mechanism',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': ['1138738', '4559540', '5083217', '6809465', '7866596'],\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Contraceptives, Oral, Hormonal',\n",
       "  'mesh_chemical_id': 'D003278',\n",
       "  'cas_registry_number': None,\n",
       "  'direct_evidence': 'marker/mechanism',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': '306034',\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Cyclopenthiazide',\n",
       "  'mesh_chemical_id': 'D003518',\n",
       "  'cas_registry_number': '742-20-1',\n",
       "  'direct_evidence': 'therapeutic',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': '1139266',\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Desoxycorticosterone',\n",
       "  'mesh_chemical_id': 'D003900',\n",
       "  'cas_registry_number': '64-85-7',\n",
       "  'direct_evidence': 'marker/mechanism',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': '1860718',\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Diazoxide',\n",
       "  'mesh_chemical_id': 'D003981',\n",
       "  'cas_registry_number': '364-98-7',\n",
       "  'direct_evidence': 'therapeutic',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': '4444609',\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Diuretics',\n",
       "  'mesh_chemical_id': 'D004232',\n",
       "  'cas_registry_number': None,\n",
       "  'direct_evidence': 'therapeutic',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': ['2493837', '6156352'],\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Enalapril',\n",
       "  'mesh_chemical_id': 'D004656',\n",
       "  'cas_registry_number': '75847-73-3',\n",
       "  'direct_evidence': 'therapeutic',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': ['2493837', '3020089', '9546244'],\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Enzyme Inhibitors',\n",
       "  'mesh_chemical_id': 'D004791',\n",
       "  'cas_registry_number': None,\n",
       "  'direct_evidence': 'marker/mechanism',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': '19326569',\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Ethinyl Estradiol',\n",
       "  'mesh_chemical_id': 'D004997',\n",
       "  'cas_registry_number': '57-63-6',\n",
       "  'direct_evidence': 'marker/mechanism',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': '1747706',\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Hydralazine',\n",
       "  'mesh_chemical_id': 'D006830',\n",
       "  'cas_registry_number': '86-54-4',\n",
       "  'direct_evidence': 'therapeutic',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': '6112544',\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Lisinopril',\n",
       "  'mesh_chemical_id': 'D017706',\n",
       "  'cas_registry_number': '83915-83-7',\n",
       "  'direct_evidence': 'therapeutic',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': '8723367',\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Losartan',\n",
       "  'mesh_chemical_id': 'D019808',\n",
       "  'cas_registry_number': '114798-26-4',\n",
       "  'direct_evidence': 'therapeutic',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': '19326569',\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Minoxidil',\n",
       "  'mesh_chemical_id': 'D008914',\n",
       "  'cas_registry_number': '38304-91-5',\n",
       "  'direct_evidence': 'therapeutic',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': ['21786', '6156352'],\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Nifedipine',\n",
       "  'mesh_chemical_id': 'D009543',\n",
       "  'cas_registry_number': '21829-25-4',\n",
       "  'direct_evidence': 'therapeutic',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': ['6861634', '8723367'],\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Nitrendipine',\n",
       "  'mesh_chemical_id': 'D009568',\n",
       "  'cas_registry_number': '39562-70-4',\n",
       "  'direct_evidence': 'therapeutic',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': '6085384',\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Norepinephrine',\n",
       "  'mesh_chemical_id': 'D009638',\n",
       "  'cas_registry_number': '51-41-2',\n",
       "  'direct_evidence': 'marker/mechanism',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': '6373596',\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Pindolol',\n",
       "  'mesh_chemical_id': 'D010869',\n",
       "  'cas_registry_number': '13523-86-9',\n",
       "  'direct_evidence': 'therapeutic',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': '1139266',\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Prednisolone',\n",
       "  'mesh_chemical_id': 'D011239',\n",
       "  'cas_registry_number': '50-24-8',\n",
       "  'direct_evidence': 'marker/mechanism',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': '17541236',\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Propranolol',\n",
       "  'mesh_chemical_id': 'D011433',\n",
       "  'cas_registry_number': '525-66-6',\n",
       "  'direct_evidence': 'therapeutic',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': '6112544',\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Sodium Chloride, Dietary',\n",
       "  'mesh_chemical_id': 'D017673',\n",
       "  'cas_registry_number': None,\n",
       "  'direct_evidence': 'marker/mechanism',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': '1860718',\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Spironolactone',\n",
       "  'mesh_chemical_id': 'D013148',\n",
       "  'cas_registry_number': '52-01-7',\n",
       "  'direct_evidence': 'therapeutic',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': '8723367',\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Sympatholytics',\n",
       "  'mesh_chemical_id': 'D013565',\n",
       "  'cas_registry_number': None,\n",
       "  'direct_evidence': 'therapeutic',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': '6156352',\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'trandolapril',\n",
       "  'mesh_chemical_id': 'C052035',\n",
       "  'cas_registry_number': None,\n",
       "  'direct_evidence': 'therapeutic',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': '27428043',\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Trimethaphan',\n",
       "  'mesh_chemical_id': 'D014294',\n",
       "  'cas_registry_number': '7187-66-8',\n",
       "  'direct_evidence': 'therapeutic',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': '727981',\n",
       "  'source': 'CTD'},\n",
       " {'chemical_name': 'Valsartan',\n",
       "  'mesh_chemical_id': 'D000068756',\n",
       "  'cas_registry_number': None,\n",
       "  'direct_evidence': 'therapeutic',\n",
       "  'inference_gene_symbol': None,\n",
       "  'inference_score': None,\n",
       "  'omim_id': None,\n",
       "  'pubmed': '11560862',\n",
       "  'source': 'CTD'}]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trial1['MESH:D006974']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "### disease -> pathway"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-16T02:23:44.449010Z",
     "start_time": "2021-02-16T02:23:44.446380Z"
    },
    "hidden": true
   },
   "outputs": [],
   "source": [
    "CTD_path2 = pathlib.Path.home().joinpath('Downloads', 'CTD_diseases_pathways.csv.gz')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-15T22:00:40.466897Z",
     "start_time": "2021-02-15T22:00:39.707329Z"
    },
    "hidden": true
   },
   "outputs": [],
   "source": [
    "lookat = pd.read_csv(CTD_path2, sep=',', comment='#', compression='gzip', names=['DiseaseName', 'DiseaseID', 'pathway_name', 'pathway_id', 'inference_gene_symbol'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-15T22:47:37.612749Z",
     "start_time": "2021-02-15T22:47:37.605735Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "DiseaseName              object\n",
       "DiseaseID                object\n",
       "pathway_name             object\n",
       "pathway_id               object\n",
       "inference_gene_symbol    object\n",
       "dtype: object"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lookat.dtypes\n",
    "589408 == 572377 + 17031"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-15T22:01:26.325452Z",
     "start_time": "2021-02-15T22:01:25.754067Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False    572377\n",
       "True      17031\n",
       "Name: DiseaseID, dtype: int64"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "text/plain": [
       "True     572377\n",
       "False     17031\n",
       "Name: DiseaseID, dtype: int64"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lookat.DiseaseID.str.contains('OMIM:').value_counts()\n",
    "lookat.DiseaseID.str.contains('MESH:').value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-15T22:50:40.201404Z",
     "start_time": "2021-02-15T22:50:40.100264Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DiseaseName</th>\n",
       "      <th>DiseaseID</th>\n",
       "      <th>pathway_name</th>\n",
       "      <th>pathway_id</th>\n",
       "      <th>inference_gene_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>493395</th>\n",
       "      <td>Prostatic Neoplasms</td>\n",
       "      <td>MESH:D011471</td>\n",
       "      <td>2-Oxocarboxylic acid metabolism</td>\n",
       "      <td>KEGG:hsa01210</td>\n",
       "      <td>IDH1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>493396</th>\n",
       "      <td>Prostatic Neoplasms</td>\n",
       "      <td>MESH:D011471</td>\n",
       "      <td>Abacavir transmembrane transport</td>\n",
       "      <td>REACT:R-HSA-2161517</td>\n",
       "      <td>SLC22A3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>493397</th>\n",
       "      <td>Prostatic Neoplasms</td>\n",
       "      <td>MESH:D011471</td>\n",
       "      <td>Abacavir transport and metabolism</td>\n",
       "      <td>REACT:R-HSA-2161522</td>\n",
       "      <td>SLC22A3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>493398</th>\n",
       "      <td>Prostatic Neoplasms</td>\n",
       "      <td>MESH:D011471</td>\n",
       "      <td>Abasic sugar-phosphate removal via the single-...</td>\n",
       "      <td>REACT:R-HSA-73930</td>\n",
       "      <td>APEX1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>493399</th>\n",
       "      <td>Prostatic Neoplasms</td>\n",
       "      <td>MESH:D011471</td>\n",
       "      <td>ABC-family proteins mediated transport</td>\n",
       "      <td>REACT:R-HSA-382556</td>\n",
       "      <td>ABCC4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504205</th>\n",
       "      <td>Prostatic Neoplasms</td>\n",
       "      <td>MESH:D011471</td>\n",
       "      <td>YAP1- and WWTR1 (TAZ)-stimulated gene expression</td>\n",
       "      <td>REACT:R-HSA-2032785</td>\n",
       "      <td>PPARA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504206</th>\n",
       "      <td>Prostatic Neoplasms</td>\n",
       "      <td>MESH:D011471</td>\n",
       "      <td>YAP1- and WWTR1 (TAZ)-stimulated gene expression</td>\n",
       "      <td>REACT:R-HSA-2032785</td>\n",
       "      <td>RXRA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504207</th>\n",
       "      <td>Prostatic Neoplasms</td>\n",
       "      <td>MESH:D011471</td>\n",
       "      <td>YAP1- and WWTR1 (TAZ)-stimulated gene expression</td>\n",
       "      <td>REACT:R-HSA-2032785</td>\n",
       "      <td>TBL1XR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504208</th>\n",
       "      <td>Prostatic Neoplasms</td>\n",
       "      <td>MESH:D011471</td>\n",
       "      <td>Zinc influx into cells by the SLC39 gene family</td>\n",
       "      <td>REACT:R-HSA-442380</td>\n",
       "      <td>SLC39A1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504209</th>\n",
       "      <td>Prostatic Neoplasms</td>\n",
       "      <td>MESH:D011471</td>\n",
       "      <td>Zinc transporters</td>\n",
       "      <td>REACT:R-HSA-435354</td>\n",
       "      <td>SLC39A1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10815 rows  5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                DiseaseName     DiseaseID  \\\n",
       "493395  Prostatic Neoplasms  MESH:D011471   \n",
       "493396  Prostatic Neoplasms  MESH:D011471   \n",
       "493397  Prostatic Neoplasms  MESH:D011471   \n",
       "493398  Prostatic Neoplasms  MESH:D011471   \n",
       "493399  Prostatic Neoplasms  MESH:D011471   \n",
       "...                     ...           ...   \n",
       "504205  Prostatic Neoplasms  MESH:D011471   \n",
       "504206  Prostatic Neoplasms  MESH:D011471   \n",
       "504207  Prostatic Neoplasms  MESH:D011471   \n",
       "504208  Prostatic Neoplasms  MESH:D011471   \n",
       "504209  Prostatic Neoplasms  MESH:D011471   \n",
       "\n",
       "                                             pathway_name  \\\n",
       "493395                    2-Oxocarboxylic acid metabolism   \n",
       "493396                   Abacavir transmembrane transport   \n",
       "493397                  Abacavir transport and metabolism   \n",
       "493398  Abasic sugar-phosphate removal via the single-...   \n",
       "493399             ABC-family proteins mediated transport   \n",
       "...                                                   ...   \n",
       "504205   YAP1- and WWTR1 (TAZ)-stimulated gene expression   \n",
       "504206   YAP1- and WWTR1 (TAZ)-stimulated gene expression   \n",
       "504207   YAP1- and WWTR1 (TAZ)-stimulated gene expression   \n",
       "504208    Zinc influx into cells by the SLC39 gene family   \n",
       "504209                                  Zinc transporters   \n",
       "\n",
       "                 pathway_id inference_gene_symbol  \n",
       "493395        KEGG:hsa01210                  IDH1  \n",
       "493396  REACT:R-HSA-2161517               SLC22A3  \n",
       "493397  REACT:R-HSA-2161522               SLC22A3  \n",
       "493398    REACT:R-HSA-73930                 APEX1  \n",
       "493399   REACT:R-HSA-382556                 ABCC4  \n",
       "...                     ...                   ...  \n",
       "504205  REACT:R-HSA-2032785                 PPARA  \n",
       "504206  REACT:R-HSA-2032785                  RXRA  \n",
       "504207  REACT:R-HSA-2032785               TBL1XR1  \n",
       "504208   REACT:R-HSA-442380               SLC39A1  \n",
       "504209   REACT:R-HSA-435354               SLC39A1  \n",
       "\n",
       "[10815 rows x 5 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lookat[lookat['DiseaseID'] == 'MESH:D011471']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-16T02:23:49.331580Z",
     "start_time": "2021-02-16T02:23:49.319873Z"
    },
    "hidden": true
   },
   "outputs": [],
   "source": [
    "def process_pathway(file_path_pathway):\n",
    "    # read in the data frame\n",
    "    df_disease_pathway = pd.read_csv(file_path_pathway, sep=',', comment='#', compression='gzip',\n",
    "                                     names=['DiseaseName', 'DiseaseID', 'pathway_name', 'pathway_id', 'inference_gene_symbol'],\n",
    "                                     dtype=str)\n",
    "    d = {}\n",
    "    ## going to merge records with the same disease - pathway (but different inference_gene_symbol values)\n",
    "    for grp, subdf in df_disease_pathway.groupby(['DiseaseID', 'pathway_id']):\n",
    "        ## make a new record        \n",
    "        record_dict = {\n",
    "            'source': 'CTD',\n",
    "            'pathway_name': subdf['pathway_name'].tolist()[0]}\n",
    "        ## name the field based on pathway database\n",
    "        tempPathwayID = grp[1].split(':')\n",
    "        record_dict[tempPathwayID[0].lower() + '_pathway_id'] = tempPathwayID[1] \n",
    "        ## get the inference gene symbol list\n",
    "        tempGeneL = subdf['inference_gene_symbol'].unique().tolist()\n",
    "        if len(tempGeneL) == 1:\n",
    "            record_dict['inference_gene_symbol'] = tempGeneL[0]\n",
    "        else:\n",
    "            record_dict['inference_gene_symbol'] = tempGeneL\n",
    "        ## if disease key already exists in d\n",
    "        if d.get(grp[0]):\n",
    "            d[grp[0]].append(record_dict)\n",
    "        else:\n",
    "            d[grp[0]] = [record_dict]\n",
    "        ## note: <20 diseases have >1000 unique pathways linked to them\n",
    "    return d"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-16T02:24:54.620288Z",
     "start_time": "2021-02-16T02:23:56.205447Z"
    },
    "hidden": true
   },
   "outputs": [],
   "source": [
    "pathwaydf = process_pathway(CTD_path2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-02-16T02:26:58.914516Z",
     "start_time": "2021-02-16T02:26:58.667626Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'source': 'CTD',\n",
       "  'pathway_name': 'Glycolysis / Gluconeogenesis',\n",
       "  'kegg_pathway_id': 'hsa00010',\n",
       "  'inference_gene_symbol': ['ENO1', 'GAPDH', 'PGAM1', 'TPI1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Fructose and mannose metabolism',\n",
       "  'kegg_pathway_id': 'hsa00051',\n",
       "  'inference_gene_symbol': 'TPI1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Steroid hormone biosynthesis',\n",
       "  'kegg_pathway_id': 'hsa00140',\n",
       "  'inference_gene_symbol': ['CYP1A2', 'CYP1B1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Caffeine metabolism',\n",
       "  'kegg_pathway_id': 'hsa00232',\n",
       "  'inference_gene_symbol': 'CYP1A2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Pyrimidine metabolism',\n",
       "  'kegg_pathway_id': 'hsa00240',\n",
       "  'inference_gene_symbol': ['TYMS', 'UMPS']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Glycine, serine and threonine metabolism',\n",
       "  'kegg_pathway_id': 'hsa00260',\n",
       "  'inference_gene_symbol': 'PGAM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Tryptophan metabolism',\n",
       "  'kegg_pathway_id': 'hsa00380',\n",
       "  'inference_gene_symbol': ['CYP1A2', 'CYP1B1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Glutathione metabolism',\n",
       "  'kegg_pathway_id': 'hsa00480',\n",
       "  'inference_gene_symbol': ['GPX1', 'GSTM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Inositol phosphate metabolism',\n",
       "  'kegg_pathway_id': 'hsa00562',\n",
       "  'inference_gene_symbol': ['INPP5A', 'PIK3CA', 'TPI1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Arachidonic acid metabolism',\n",
       "  'kegg_pathway_id': 'hsa00590',\n",
       "  'inference_gene_symbol': ['GPX1', 'PTGS2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Linoleic acid metabolism',\n",
       "  'kegg_pathway_id': 'hsa00591',\n",
       "  'inference_gene_symbol': 'CYP1A2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'One carbon pool by folate',\n",
       "  'kegg_pathway_id': 'hsa00670',\n",
       "  'inference_gene_symbol': 'TYMS'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Retinol metabolism',\n",
       "  'kegg_pathway_id': 'hsa00830',\n",
       "  'inference_gene_symbol': 'CYP1A2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Metabolism of xenobiotics by cytochrome P450',\n",
       "  'kegg_pathway_id': 'hsa00980',\n",
       "  'inference_gene_symbol': ['CYP1A2', 'CYP1B1', 'GSTM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Drug metabolism - cytochrome P450',\n",
       "  'kegg_pathway_id': 'hsa00982',\n",
       "  'inference_gene_symbol': ['CYP1A2', 'GSTM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Drug metabolism - other enzymes',\n",
       "  'kegg_pathway_id': 'hsa00983',\n",
       "  'inference_gene_symbol': 'UMPS'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Metabolic pathways',\n",
       "  'kegg_pathway_id': 'hsa01100',\n",
       "  'inference_gene_symbol': ['CYP1A2',\n",
       "   'ENO1',\n",
       "   'GAPDH',\n",
       "   'INPP5A',\n",
       "   'PGAM1',\n",
       "   'PTGS2',\n",
       "   'TPI1',\n",
       "   'TYMS',\n",
       "   'UMPS']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Carbon metabolism',\n",
       "  'kegg_pathway_id': 'hsa01200',\n",
       "  'inference_gene_symbol': ['ENO1', 'GAPDH', 'PGAM1', 'TPI1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Biosynthesis of amino acids',\n",
       "  'kegg_pathway_id': 'hsa01230',\n",
       "  'inference_gene_symbol': ['ENO1', 'GAPDH', 'PGAM1', 'TPI1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'EGFR tyrosine kinase inhibitor resistance',\n",
       "  'kegg_pathway_id': 'hsa01521',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'IL6',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'STAT3',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Endocrine resistance',\n",
       "  'kegg_pathway_id': 'hsa01522',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CCND1',\n",
       "   'CDKN2A',\n",
       "   'EGFR',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MMP9',\n",
       "   'NCOA3',\n",
       "   'NOTCH1',\n",
       "   'NOTCH2',\n",
       "   'NOTCH3',\n",
       "   'NOTCH4',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Antifolate resistance',\n",
       "  'kegg_pathway_id': 'hsa01523',\n",
       "  'inference_gene_symbol': ['IL6', 'TYMS']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Platinum drug resistance',\n",
       "  'kegg_pathway_id': 'hsa01524',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'ATM',\n",
       "   'BIRC5',\n",
       "   'CDKN2A',\n",
       "   'GSTM1',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Ribosome biogenesis in eukaryotes',\n",
       "  'kegg_pathway_id': 'hsa03008',\n",
       "  'inference_gene_symbol': ['CSNK2A1', 'CSNK2A2', 'CSNK2B']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Ribosome',\n",
       "  'kegg_pathway_id': 'hsa03010',\n",
       "  'inference_gene_symbol': 'MRPL13'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RNA degradation',\n",
       "  'kegg_pathway_id': 'hsa03018',\n",
       "  'inference_gene_symbol': 'ENO1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Basal transcription factors',\n",
       "  'kegg_pathway_id': 'hsa03022',\n",
       "  'inference_gene_symbol': 'ERCC2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Spliceosome',\n",
       "  'kegg_pathway_id': 'hsa03040',\n",
       "  'inference_gene_symbol': 'HSPA8'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Proteasome',\n",
       "  'kegg_pathway_id': 'hsa03050',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Nucleotide excision repair',\n",
       "  'kegg_pathway_id': 'hsa03420',\n",
       "  'inference_gene_symbol': ['ERCC2', 'XPC']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Homologous recombination',\n",
       "  'kegg_pathway_id': 'hsa03440',\n",
       "  'inference_gene_symbol': ['ATM', 'RAD52', 'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Non-homologous end-joining',\n",
       "  'kegg_pathway_id': 'hsa03450',\n",
       "  'inference_gene_symbol': 'PRKDC'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'MAPK signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04010',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HRAS',\n",
       "   'HSPA8',\n",
       "   'HSPB1',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MYC',\n",
       "   'RASA1',\n",
       "   'TGFB1',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'ErbB signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04012',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'AREG',\n",
       "   'EGFR',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MYC',\n",
       "   'PIK3CA',\n",
       "   'PTK2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Ras signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04014',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTPN11',\n",
       "   'RASA1',\n",
       "   'TIAM1',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Rap1 signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04015',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CDH1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'ITGB1',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'TIAM1',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Calcium signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04020',\n",
       "  'inference_gene_symbol': ['EGFR', 'PTGER1', 'VDAC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'cGMP-PKG signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04022',\n",
       "  'inference_gene_symbol': ['AKT1', 'MAPK1', 'MAPK3', 'VDAC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'cAMP signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04024',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CREBBP',\n",
       "   'EP300',\n",
       "   'HCAR2',\n",
       "   'HCAR3',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'TIAM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cytokine-cytokine receptor interaction',\n",
       "  'kegg_pathway_id': 'hsa04060',\n",
       "  'inference_gene_symbol': ['CCL2',\n",
       "   'CXCL8',\n",
       "   'EGFR',\n",
       "   'HGF',\n",
       "   'IFNA2',\n",
       "   'IL6',\n",
       "   'MET',\n",
       "   'TGFB1',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Chemokine signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04062',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CCL2',\n",
       "   'CXCL8',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'STAT3',\n",
       "   'TIAM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'NF-kappa B signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04064',\n",
       "  'inference_gene_symbol': ['ATM',\n",
       "   'CSNK2A1',\n",
       "   'CSNK2A2',\n",
       "   'CSNK2B',\n",
       "   'CXCL8',\n",
       "   'PTGS2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'HIF-1 signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04066',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CREBBP',\n",
       "   'EGFR',\n",
       "   'ENO1',\n",
       "   'EP300',\n",
       "   'GAPDH',\n",
       "   'IL6',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'STAT3',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'FoxO signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04068',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'ATM',\n",
       "   'CCND1',\n",
       "   'CREBBP',\n",
       "   'EGFR',\n",
       "   'EP300',\n",
       "   'HRAS',\n",
       "   'IL6',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'SOD2',\n",
       "   'STAT3',\n",
       "   'STK11',\n",
       "   'TGFB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Phosphatidylinositol signaling system',\n",
       "  'kegg_pathway_id': 'hsa04070',\n",
       "  'inference_gene_symbol': ['INPP5A', 'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Sphingolipid signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04071',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Phospholipase D signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04072',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CXCL8',\n",
       "   'EGFR',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'PTPN11',\n",
       "   'TSC1',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Neuroactive ligand-receptor interaction',\n",
       "  'kegg_pathway_id': 'hsa04080',\n",
       "  'inference_gene_symbol': 'PTGER1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cell cycle',\n",
       "  'kegg_pathway_id': 'hsa04110',\n",
       "  'inference_gene_symbol': ['ATM',\n",
       "   'CCND1',\n",
       "   'CDKN2A',\n",
       "   'CREBBP',\n",
       "   'EP300',\n",
       "   'HDAC1',\n",
       "   'HDAC2',\n",
       "   'MYC',\n",
       "   'PRKDC',\n",
       "   'SFN',\n",
       "   'TGFB1',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Oocyte meiosis',\n",
       "  'kegg_pathway_id': 'hsa04114',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'p53 signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04115',\n",
       "  'inference_gene_symbol': ['ATM', 'CCND1', 'CDKN2A', 'SFN', 'TP53', 'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Ubiquitin mediated proteolysis',\n",
       "  'kegg_pathway_id': 'hsa04120',\n",
       "  'inference_gene_symbol': ['UBA1', 'UBA7']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Mitophagy - animal',\n",
       "  'kegg_pathway_id': 'hsa04137',\n",
       "  'inference_gene_symbol': ['CSNK2A1',\n",
       "   'CSNK2A2',\n",
       "   'CSNK2B',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Autophagy - animal',\n",
       "  'kegg_pathway_id': 'hsa04140',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'STK11',\n",
       "   'TSC1',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Protein processing in endoplasmic reticulum',\n",
       "  'kegg_pathway_id': 'hsa04141',\n",
       "  'inference_gene_symbol': ['CRYAB', 'HSPA8', 'NFE2L2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Endocytosis',\n",
       "  'kegg_pathway_id': 'hsa04144',\n",
       "  'inference_gene_symbol': ['EGFR', 'HRAS', 'HSPA8', 'MET', 'TGFB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Phagosome',\n",
       "  'kegg_pathway_id': 'hsa04145',\n",
       "  'inference_gene_symbol': ['ITGA5', 'ITGB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Peroxisome',\n",
       "  'kegg_pathway_id': 'hsa04146',\n",
       "  'inference_gene_symbol': 'SOD2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'mTOR signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04150',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'STK11',\n",
       "   'TSC1',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'PI3K-Akt signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04151',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CCND1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'IFNA2',\n",
       "   'IL6',\n",
       "   'ITGA5',\n",
       "   'ITGB1',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'MYC',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'STK11',\n",
       "   'TP53',\n",
       "   'TSC1',\n",
       "   'TSC2',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'AMPK signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04152',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CCND1',\n",
       "   'PIK3CA',\n",
       "   'STK11',\n",
       "   'TSC1',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Apoptosis',\n",
       "  'kegg_pathway_id': 'hsa04210',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'ATM',\n",
       "   'BIRC5',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Longevity regulating pathway',\n",
       "  'kegg_pathway_id': 'hsa04211',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'PIK3CA',\n",
       "   'SOD2',\n",
       "   'STK11',\n",
       "   'TP53',\n",
       "   'TSC1',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Longevity regulating pathway - multiple species',\n",
       "  'kegg_pathway_id': 'hsa04213',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CRYAB',\n",
       "   'HDAC1',\n",
       "   'HDAC2',\n",
       "   'HRAS',\n",
       "   'HSPA8',\n",
       "   'KRAS',\n",
       "   'PIK3CA',\n",
       "   'SOD2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Apoptosis - multiple species',\n",
       "  'kegg_pathway_id': 'hsa04215',\n",
       "  'inference_gene_symbol': 'BIRC5'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Ferroptosis',\n",
       "  'kegg_pathway_id': 'hsa04216',\n",
       "  'inference_gene_symbol': ['TP53', 'VDAC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Adrenergic signaling in cardiomyocytes',\n",
       "  'kegg_pathway_id': 'hsa04261',\n",
       "  'inference_gene_symbol': ['AKT1', 'MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Vascular smooth muscle contraction',\n",
       "  'kegg_pathway_id': 'hsa04270',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Wnt signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04310',\n",
       "  'inference_gene_symbol': ['CCND1',\n",
       "   'CREBBP',\n",
       "   'CSNK2A1',\n",
       "   'CSNK2A2',\n",
       "   'CSNK2B',\n",
       "   'EP300',\n",
       "   'MYC',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Dorso-ventral axis formation',\n",
       "  'kegg_pathway_id': 'hsa04320',\n",
       "  'inference_gene_symbol': ['EGFR',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'NOTCH1',\n",
       "   'NOTCH2',\n",
       "   'NOTCH3',\n",
       "   'NOTCH4']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Notch signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04330',\n",
       "  'inference_gene_symbol': ['CREBBP',\n",
       "   'EP300',\n",
       "   'HDAC1',\n",
       "   'HDAC2',\n",
       "   'NOTCH1',\n",
       "   'NOTCH2',\n",
       "   'NOTCH3',\n",
       "   'NOTCH4']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Hedgehog signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04340',\n",
       "  'inference_gene_symbol': 'CCND1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TGF-beta signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04350',\n",
       "  'inference_gene_symbol': ['CREBBP',\n",
       "   'EP300',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MYC',\n",
       "   'TGFB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Axon guidance',\n",
       "  'kegg_pathway_id': 'hsa04360',\n",
       "  'inference_gene_symbol': ['HRAS',\n",
       "   'ITGB1',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'RASA1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'VEGF signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04370',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'HRAS',\n",
       "   'HSPB1',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'PTGS2',\n",
       "   'PTK2',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Apelin signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04371',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CCND1',\n",
       "   'CDH1',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'NOTCH3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Osteoclast differentiation',\n",
       "  'kegg_pathway_id': 'hsa04380',\n",
       "  'inference_gene_symbol': ['AKT1', 'MAPK1', 'MAPK3', 'PIK3CA', 'TGFB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Hippo signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04390',\n",
       "  'inference_gene_symbol': ['AREG',\n",
       "   'BIRC5',\n",
       "   'CCND1',\n",
       "   'CDH1',\n",
       "   'MYC',\n",
       "   'SOX2',\n",
       "   'TGFB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Focal adhesion',\n",
       "  'kegg_pathway_id': 'hsa04510',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CCND1',\n",
       "   'EGFR',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'ITGA5',\n",
       "   'ITGB1',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'ECM-receptor interaction',\n",
       "  'kegg_pathway_id': 'hsa04512',\n",
       "  'inference_gene_symbol': ['ITGA5', 'ITGB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cell adhesion molecules (CAMs)',\n",
       "  'kegg_pathway_id': 'hsa04514',\n",
       "  'inference_gene_symbol': ['CDH1', 'ITGB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Adherens junction',\n",
       "  'kegg_pathway_id': 'hsa04520',\n",
       "  'inference_gene_symbol': ['CDH1',\n",
       "   'CREBBP',\n",
       "   'CSNK2A1',\n",
       "   'CSNK2A2',\n",
       "   'CSNK2B',\n",
       "   'EGFR',\n",
       "   'EP300',\n",
       "   'FGFR1',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Tight junction',\n",
       "  'kegg_pathway_id': 'hsa04530',\n",
       "  'inference_gene_symbol': ['CCND1', 'CTTN', 'ITGB1', 'STK11', 'TIAM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Gap junction',\n",
       "  'kegg_pathway_id': 'hsa04540',\n",
       "  'inference_gene_symbol': ['EGFR', 'HRAS', 'KRAS', 'MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling pathways regulating pluripotency of stem cells',\n",
       "  'kegg_pathway_id': 'hsa04550',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'FGFR1',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MYC',\n",
       "   'PIK3CA',\n",
       "   'SOX2',\n",
       "   'STAT3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Platelet activation',\n",
       "  'kegg_pathway_id': 'hsa04611',\n",
       "  'inference_gene_symbol': ['AKT1', 'ITGB1', 'MAPK1', 'MAPK3', 'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Antigen processing and presentation',\n",
       "  'kegg_pathway_id': 'hsa04612',\n",
       "  'inference_gene_symbol': 'HSPA8'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Toll-like receptor signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04620',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CXCL8',\n",
       "   'IFNA2',\n",
       "   'IL6',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'NOD-like receptor signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04621',\n",
       "  'inference_gene_symbol': ['CARD18',\n",
       "   'CCL2',\n",
       "   'CXCL8',\n",
       "   'IFNA2',\n",
       "   'IL6',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PYCARD',\n",
       "   'VDAC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RIG-I-like receptor signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04622',\n",
       "  'inference_gene_symbol': ['CXCL8', 'IFNA2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cytosolic DNA-sensing pathway',\n",
       "  'kegg_pathway_id': 'hsa04623',\n",
       "  'inference_gene_symbol': ['IFNA2', 'IL6', 'PYCARD']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Jak-STAT signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04630',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CCND1',\n",
       "   'CREBBP',\n",
       "   'EP300',\n",
       "   'HRAS',\n",
       "   'IFNA2',\n",
       "   'IL6',\n",
       "   'MYC',\n",
       "   'PIK3CA',\n",
       "   'PTPN11',\n",
       "   'STAT3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Hematopoietic cell lineage',\n",
       "  'kegg_pathway_id': 'hsa04640',\n",
       "  'inference_gene_symbol': ['IL6', 'ITGA5', 'MS4A1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Natural killer cell mediated cytotoxicity',\n",
       "  'kegg_pathway_id': 'hsa04650',\n",
       "  'inference_gene_symbol': ['HRAS',\n",
       "   'IFNA2',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'IL-17 signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04657',\n",
       "  'inference_gene_symbol': ['CCL2',\n",
       "   'CXCL8',\n",
       "   'IL6',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MMP9',\n",
       "   'PTGS2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Th1 and Th2 cell differentiation',\n",
       "  'kegg_pathway_id': 'hsa04658',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3', 'NOTCH1', 'NOTCH2', 'NOTCH3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Th17 cell differentiation',\n",
       "  'kegg_pathway_id': 'hsa04659',\n",
       "  'inference_gene_symbol': ['IL6', 'MAPK1', 'MAPK3', 'STAT3', 'TGFB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'T cell receptor signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04660',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'B cell receptor signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04662',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Fc epsilon RI signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04664',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Fc gamma R-mediated phagocytosis',\n",
       "  'kegg_pathway_id': 'hsa04666',\n",
       "  'inference_gene_symbol': ['AKT1', 'MAPK1', 'MAPK3', 'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TNF signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04668',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CCL2',\n",
       "   'IL6',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MMP9',\n",
       "   'PIK3CA',\n",
       "   'PTGS2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Leukocyte transendothelial migration',\n",
       "  'kegg_pathway_id': 'hsa04670',\n",
       "  'inference_gene_symbol': ['ITGB1', 'MMP9', 'PIK3CA', 'PTK2', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Intestinal immune network for IgA production',\n",
       "  'kegg_pathway_id': 'hsa04672',\n",
       "  'inference_gene_symbol': ['IL6', 'TGFB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Circadian entrainment',\n",
       "  'kegg_pathway_id': 'hsa04713',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Long-term potentiation',\n",
       "  'kegg_pathway_id': 'hsa04720',\n",
       "  'inference_gene_symbol': ['CREBBP',\n",
       "   'EP300',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Neurotrophin signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04722',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'PTPN11',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Retrograde endocannabinoid signaling',\n",
       "  'kegg_pathway_id': 'hsa04723',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3', 'PTGS2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Glutamatergic synapse',\n",
       "  'kegg_pathway_id': 'hsa04724',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cholinergic synapse',\n",
       "  'kegg_pathway_id': 'hsa04725',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Serotonergic synapse',\n",
       "  'kegg_pathway_id': 'hsa04726',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS', 'MAPK1', 'MAPK3', 'PTGS2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Dopaminergic synapse',\n",
       "  'kegg_pathway_id': 'hsa04728',\n",
       "  'inference_gene_symbol': 'AKT1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Long-term depression',\n",
       "  'kegg_pathway_id': 'hsa04730',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS', 'MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Inflammatory mediator regulation of TRP channels',\n",
       "  'kegg_pathway_id': 'hsa04750',\n",
       "  'inference_gene_symbol': 'PIK3CA'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of actin cytoskeleton',\n",
       "  'kegg_pathway_id': 'hsa04810',\n",
       "  'inference_gene_symbol': ['EGFR',\n",
       "   'FGFR1',\n",
       "   'HRAS',\n",
       "   'ITGA5',\n",
       "   'ITGB1',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'TIAM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Insulin signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04910',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'TSC1',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'GnRH signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04912',\n",
       "  'inference_gene_symbol': ['EGFR', 'HRAS', 'KRAS', 'MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Ovarian steroidogenesis',\n",
       "  'kegg_pathway_id': 'hsa04913',\n",
       "  'inference_gene_symbol': ['CYP1B1', 'PTGS2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Progesterone-mediated oocyte maturation',\n",
       "  'kegg_pathway_id': 'hsa04914',\n",
       "  'inference_gene_symbol': ['AKT1', 'KRAS', 'MAPK1', 'MAPK3', 'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Estrogen signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04915',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'HRAS',\n",
       "   'HSPA8',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MMP9',\n",
       "   'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Melanogenesis',\n",
       "  'kegg_pathway_id': 'hsa04916',\n",
       "  'inference_gene_symbol': ['CREBBP',\n",
       "   'EP300',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Prolactin signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04917',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CCND1',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'STAT3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Thyroid hormone synthesis',\n",
       "  'kegg_pathway_id': 'hsa04918',\n",
       "  'inference_gene_symbol': 'GPX1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Thyroid hormone signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04919',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CCND1',\n",
       "   'CREBBP',\n",
       "   'EP300',\n",
       "   'HDAC1',\n",
       "   'HDAC2',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MYC',\n",
       "   'NCOA3',\n",
       "   'NOTCH1',\n",
       "   'NOTCH2',\n",
       "   'NOTCH3',\n",
       "   'NOTCH4',\n",
       "   'PIK3CA',\n",
       "   'TP53',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Adipocytokine signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04920',\n",
       "  'inference_gene_symbol': ['AKT1', 'PTPN11', 'STAT3', 'STK11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Oxytocin signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04921',\n",
       "  'inference_gene_symbol': ['CCND1',\n",
       "   'EGFR',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PTGS2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Glucagon signaling pathway',\n",
       "  'kegg_pathway_id': 'hsa04922',\n",
       "  'inference_gene_symbol': ['AKT1', 'CREBBP', 'EP300', 'PGAM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of lipolysis in adipocytes',\n",
       "  'kegg_pathway_id': 'hsa04923',\n",
       "  'inference_gene_symbol': ['AKT1', 'PIK3CA', 'PTGS2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Type II diabetes mellitus',\n",
       "  'kegg_pathway_id': 'hsa04930',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3', 'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Insulin resistance',\n",
       "  'kegg_pathway_id': 'hsa04931',\n",
       "  'inference_gene_symbol': ['AKT1', 'IL6', 'PIK3CA', 'PTPN11', 'STAT3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Non-alcoholic fatty liver disease (NAFLD)',\n",
       "  'kegg_pathway_id': 'hsa04932',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CEBPA',\n",
       "   'CXCL8',\n",
       "   'IL6',\n",
       "   'PIK3CA',\n",
       "   'TGFB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'AGE-RAGE signaling pathway in diabetic complications',\n",
       "  'kegg_pathway_id': 'hsa04933',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CCL2',\n",
       "   'CCND1',\n",
       "   'CXCL8',\n",
       "   'HRAS',\n",
       "   'IL6',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'STAT3',\n",
       "   'TGFB1',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Aldosterone-regulated sodium reabsorption',\n",
       "  'kegg_pathway_id': 'hsa04960',\n",
       "  'inference_gene_symbol': ['KRAS', 'MAPK1', 'MAPK3', 'PIK3CA', 'SFN']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Vasopressin-regulated water reabsorption',\n",
       "  'kegg_pathway_id': 'hsa04962',\n",
       "  'inference_gene_symbol': 'AQP3'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Carbohydrate digestion and absorption',\n",
       "  'kegg_pathway_id': 'hsa04973',\n",
       "  'inference_gene_symbol': ['AKT1', 'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Mineral absorption',\n",
       "  'kegg_pathway_id': 'hsa04978',\n",
       "  'inference_gene_symbol': 'MT1A'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': \"Alzheimer's disease\",\n",
       "  'kegg_pathway_id': 'hsa05010',\n",
       "  'inference_gene_symbol': ['GAPDH', 'MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': \"Parkinson's disease\",\n",
       "  'kegg_pathway_id': 'hsa05012',\n",
       "  'inference_gene_symbol': ['UBA1', 'UBA7', 'VDAC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Amyotrophic lateral sclerosis (ALS)',\n",
       "  'kegg_pathway_id': 'hsa05014',\n",
       "  'inference_gene_symbol': ['GPX1', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': \"Huntington's disease\",\n",
       "  'kegg_pathway_id': 'hsa05016',\n",
       "  'inference_gene_symbol': ['CREBBP',\n",
       "   'EP300',\n",
       "   'GPX1',\n",
       "   'HDAC1',\n",
       "   'HDAC2',\n",
       "   'SOD2',\n",
       "   'TP53',\n",
       "   'VDAC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Prion diseases',\n",
       "  'kegg_pathway_id': 'hsa05020',\n",
       "  'inference_gene_symbol': ['IL6', 'MAPK1', 'MAPK3', 'NOTCH1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Amphetamine addiction',\n",
       "  'kegg_pathway_id': 'hsa05031',\n",
       "  'inference_gene_symbol': 'HDAC1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Alcoholism',\n",
       "  'kegg_pathway_id': 'hsa05034',\n",
       "  'inference_gene_symbol': ['HDAC1',\n",
       "   'HDAC2',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Bacterial invasion of epithelial cells',\n",
       "  'kegg_pathway_id': 'hsa05100',\n",
       "  'inference_gene_symbol': ['CDH1',\n",
       "   'CTTN',\n",
       "   'ITGA5',\n",
       "   'ITGB1',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTK2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Epithelial cell signaling in Helicobacter pylori infection',\n",
       "  'kegg_pathway_id': 'hsa05120',\n",
       "  'inference_gene_symbol': ['CXCL8', 'EGFR', 'MET', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Pathogenic Escherichia coli infection',\n",
       "  'kegg_pathway_id': 'hsa05130',\n",
       "  'inference_gene_symbol': ['CDH1', 'CTTN', 'ITGB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Shigellosis',\n",
       "  'kegg_pathway_id': 'hsa05131',\n",
       "  'inference_gene_symbol': ['CTTN',\n",
       "   'CXCL8',\n",
       "   'ITGA5',\n",
       "   'ITGB1',\n",
       "   'MAPK1',\n",
       "   'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Salmonella infection',\n",
       "  'kegg_pathway_id': 'hsa05132',\n",
       "  'inference_gene_symbol': ['CXCL8', 'IL6', 'MAPK1', 'MAPK3', 'PYCARD']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Pertussis',\n",
       "  'kegg_pathway_id': 'hsa05133',\n",
       "  'inference_gene_symbol': ['CXCL8',\n",
       "   'IL6',\n",
       "   'ITGA5',\n",
       "   'ITGB1',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PYCARD']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Legionellosis',\n",
       "  'kegg_pathway_id': 'hsa05134',\n",
       "  'inference_gene_symbol': ['CXCL8', 'HSPA8', 'IL6', 'PYCARD']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Leishmaniasis',\n",
       "  'kegg_pathway_id': 'hsa05140',\n",
       "  'inference_gene_symbol': ['ITGB1', 'MAPK1', 'MAPK3', 'PTGS2', 'TGFB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Chagas disease (American trypanosomiasis)',\n",
       "  'kegg_pathway_id': 'hsa05142',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CCL2',\n",
       "   'CXCL8',\n",
       "   'IL6',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'TGFB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'African trypanosomiasis',\n",
       "  'kegg_pathway_id': 'hsa05143',\n",
       "  'inference_gene_symbol': 'IL6'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Malaria',\n",
       "  'kegg_pathway_id': 'hsa05144',\n",
       "  'inference_gene_symbol': ['CCL2', 'CXCL8', 'HGF', 'IL6', 'MET', 'TGFB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Toxoplasmosis',\n",
       "  'kegg_pathway_id': 'hsa05145',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'HSPA8',\n",
       "   'ITGB1',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'STAT3',\n",
       "   'TGFB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Amoebiasis',\n",
       "  'kegg_pathway_id': 'hsa05146',\n",
       "  'inference_gene_symbol': ['CXCL8',\n",
       "   'HSPB1',\n",
       "   'IL6',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'SERPINB3',\n",
       "   'TGFB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Staphylococcus aureus infection',\n",
       "  'kegg_pathway_id': 'hsa05150',\n",
       "  'inference_gene_symbol': 'KRT10'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Tuberculosis',\n",
       "  'kegg_pathway_id': 'hsa05152',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CREBBP',\n",
       "   'EP300',\n",
       "   'IFNA2',\n",
       "   'IL6',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'TGFB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Hepatitis C',\n",
       "  'kegg_pathway_id': 'hsa05160',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CXCL8',\n",
       "   'EGFR',\n",
       "   'HRAS',\n",
       "   'IFNA2',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'STAT3',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Hepatitis B',\n",
       "  'kegg_pathway_id': 'hsa05161',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'BIRC5',\n",
       "   'CCND1',\n",
       "   'CREBBP',\n",
       "   'CXCL8',\n",
       "   'EP300',\n",
       "   'HRAS',\n",
       "   'IFNA2',\n",
       "   'IL6',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MMP9',\n",
       "   'MYC',\n",
       "   'PIK3CA',\n",
       "   'STAT3',\n",
       "   'TGFB1',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Measles',\n",
       "  'kegg_pathway_id': 'hsa05162',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CCND1',\n",
       "   'CSNK2A1',\n",
       "   'CSNK2A2',\n",
       "   'CSNK2B',\n",
       "   'HSPA8',\n",
       "   'IFNA2',\n",
       "   'IL6',\n",
       "   'PIK3CA',\n",
       "   'RACK1',\n",
       "   'STAT3',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Influenza A',\n",
       "  'kegg_pathway_id': 'hsa05164',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CCL2',\n",
       "   'CREBBP',\n",
       "   'CXCL8',\n",
       "   'EP300',\n",
       "   'HSPA8',\n",
       "   'IFNA2',\n",
       "   'IL6',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'PYCARD']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'HTLV-I infection',\n",
       "  'kegg_pathway_id': 'hsa05166',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'ATM',\n",
       "   'CCND1',\n",
       "   'CDKN2A',\n",
       "   'CREBBP',\n",
       "   'EP300',\n",
       "   'HRAS',\n",
       "   'IL6',\n",
       "   'KRAS',\n",
       "   'MYC',\n",
       "   'PIK3CA',\n",
       "   'TGFB1',\n",
       "   'TP53',\n",
       "   'VDAC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Herpes simplex infection',\n",
       "  'kegg_pathway_id': 'hsa05168',\n",
       "  'inference_gene_symbol': ['CCL2',\n",
       "   'CREBBP',\n",
       "   'CSNK2A1',\n",
       "   'CSNK2A2',\n",
       "   'CSNK2B',\n",
       "   'EP300',\n",
       "   'IFNA2',\n",
       "   'IL6',\n",
       "   'PTPN11',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Epstein-Barr virus infection',\n",
       "  'kegg_pathway_id': 'hsa05169',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CREBBP',\n",
       "   'CSNK2A1',\n",
       "   'CSNK2A2',\n",
       "   'CSNK2B',\n",
       "   'EP300',\n",
       "   'HDAC1',\n",
       "   'HDAC2',\n",
       "   'HSPA8',\n",
       "   'HSPB1',\n",
       "   'MYC',\n",
       "   'PIK3CA',\n",
       "   'SEM1',\n",
       "   'STAT3',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Pathways in cancer',\n",
       "  'kegg_pathway_id': 'hsa05200',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'BIRC5',\n",
       "   'CCND1',\n",
       "   'CDH1',\n",
       "   'CDKN2A',\n",
       "   'CEBPA',\n",
       "   'CREBBP',\n",
       "   'CXCL8',\n",
       "   'EGFR',\n",
       "   'EP300',\n",
       "   'FGFR1',\n",
       "   'HDAC1',\n",
       "   'HDAC2',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'IL6',\n",
       "   'ITGB1',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'MMP9',\n",
       "   'MYC',\n",
       "   'PIK3CA',\n",
       "   'PTGER1',\n",
       "   'PTGS2',\n",
       "   'PTK2',\n",
       "   'RARB',\n",
       "   'STAT3',\n",
       "   'TGFB1',\n",
       "   'TP53',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Transcriptional misregulation in cancer',\n",
       "  'kegg_pathway_id': 'hsa05202',\n",
       "  'inference_gene_symbol': ['ATM',\n",
       "   'CEBPA',\n",
       "   'CXCL8',\n",
       "   'HDAC1',\n",
       "   'HDAC2',\n",
       "   'IL6',\n",
       "   'MET',\n",
       "   'MMP9',\n",
       "   'MYC',\n",
       "   'PTK2',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Viral carcinogenesis',\n",
       "  'kegg_pathway_id': 'hsa05203',\n",
       "  'inference_gene_symbol': ['CCND1',\n",
       "   'CDKN2A',\n",
       "   'CREBBP',\n",
       "   'EP300',\n",
       "   'HDAC1',\n",
       "   'HDAC2',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'STAT3',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Chemical carcinogenesis',\n",
       "  'kegg_pathway_id': 'hsa05204',\n",
       "  'inference_gene_symbol': ['CYP1A2', 'CYP1B1', 'GSTM1', 'PTGS2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Proteoglycans in cancer',\n",
       "  'kegg_pathway_id': 'hsa05205',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CCND1',\n",
       "   'CTTN',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'ITGA5',\n",
       "   'ITGB1',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'MMP9',\n",
       "   'MYC',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'STAT3',\n",
       "   'TGFB1',\n",
       "   'TIAM1',\n",
       "   'TP53',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'MicroRNAs in cancer',\n",
       "  'kegg_pathway_id': 'hsa05206',\n",
       "  'inference_gene_symbol': ['ATM',\n",
       "   'CCND1',\n",
       "   'CDKN2A',\n",
       "   'CREBBP',\n",
       "   'CYP1B1',\n",
       "   'EGFR',\n",
       "   'EP300',\n",
       "   'HDAC1',\n",
       "   'HRAS',\n",
       "   'ITGA5',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MET',\n",
       "   'MIR205',\n",
       "   'MMP9',\n",
       "   'MYC',\n",
       "   'NOTCH1',\n",
       "   'NOTCH2',\n",
       "   'NOTCH3',\n",
       "   'NOTCH4',\n",
       "   'PIK3CA',\n",
       "   'PTGS2',\n",
       "   'STAT3',\n",
       "   'TP53',\n",
       "   'TP63',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Colorectal cancer',\n",
       "  'kegg_pathway_id': 'hsa05210',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'BIRC5',\n",
       "   'CCND1',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MYC',\n",
       "   'PIK3CA',\n",
       "   'TGFB1',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Renal cell carcinoma',\n",
       "  'kegg_pathway_id': 'hsa05211',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CREBBP',\n",
       "   'EP300',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTPN11',\n",
       "   'TGFB1',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Pancreatic cancer',\n",
       "  'kegg_pathway_id': 'hsa05212',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CCND1',\n",
       "   'CDKN2A',\n",
       "   'EGFR',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'STAT3',\n",
       "   'TGFB1',\n",
       "   'TP53',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Endometrial cancer',\n",
       "  'kegg_pathway_id': 'hsa05213',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CCND1',\n",
       "   'CDH1',\n",
       "   'EGFR',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MYC',\n",
       "   'PIK3CA',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Glioma',\n",
       "  'kegg_pathway_id': 'hsa05214',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CCND1',\n",
       "   'CDKN2A',\n",
       "   'EGFR',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Prostate cancer',\n",
       "  'kegg_pathway_id': 'hsa05215',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CCND1',\n",
       "   'CREBBP',\n",
       "   'EGFR',\n",
       "   'EP300',\n",
       "   'FGFR1',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Thyroid cancer',\n",
       "  'kegg_pathway_id': 'hsa05216',\n",
       "  'inference_gene_symbol': ['CCND1',\n",
       "   'CDH1',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MYC',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Basal cell carcinoma',\n",
       "  'kegg_pathway_id': 'hsa05217',\n",
       "  'inference_gene_symbol': 'TP53'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Melanoma',\n",
       "  'kegg_pathway_id': 'hsa05218',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CCND1',\n",
       "   'CDH1',\n",
       "   'CDKN2A',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Bladder cancer',\n",
       "  'kegg_pathway_id': 'hsa05219',\n",
       "  'inference_gene_symbol': ['CCND1',\n",
       "   'CDH1',\n",
       "   'CDKN2A',\n",
       "   'CXCL8',\n",
       "   'EGFR',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MMP9',\n",
       "   'MYC',\n",
       "   'TP53',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Chronic myeloid leukemia',\n",
       "  'kegg_pathway_id': 'hsa05220',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CCND1',\n",
       "   'CDKN2A',\n",
       "   'HDAC1',\n",
       "   'HDAC2',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MYC',\n",
       "   'PIK3CA',\n",
       "   'PTPN11',\n",
       "   'TGFB1',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Acute myeloid leukemia',\n",
       "  'kegg_pathway_id': 'hsa05221',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CCND1',\n",
       "   'CEBPA',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MYC',\n",
       "   'PIK3CA',\n",
       "   'STAT3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Small cell lung cancer',\n",
       "  'kegg_pathway_id': 'hsa05222',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CCND1',\n",
       "   'ITGB1',\n",
       "   'MYC',\n",
       "   'PIK3CA',\n",
       "   'PTGS2',\n",
       "   'PTK2',\n",
       "   'RARB',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Non-small cell lung cancer',\n",
       "  'kegg_pathway_id': 'hsa05223',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CCND1',\n",
       "   'CDKN2A',\n",
       "   'EGFR',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'RARB',\n",
       "   'STAT3',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Breast cancer',\n",
       "  'kegg_pathway_id': 'hsa05224',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CCND1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MYC',\n",
       "   'NCOA3',\n",
       "   'NOTCH1',\n",
       "   'NOTCH2',\n",
       "   'NOTCH3',\n",
       "   'NOTCH4',\n",
       "   'PIK3CA',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Central carbon metabolism in cancer',\n",
       "  'kegg_pathway_id': 'hsa05230',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'MYC',\n",
       "   'PGAM1',\n",
       "   'PIK3CA',\n",
       "   'SIRT3',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Choline metabolism in cancer',\n",
       "  'kegg_pathway_id': 'hsa05231',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'TSC1',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Autoimmune thyroid disease',\n",
       "  'kegg_pathway_id': 'hsa05320',\n",
       "  'inference_gene_symbol': 'IFNA2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Inflammatory bowel disease (IBD)',\n",
       "  'kegg_pathway_id': 'hsa05321',\n",
       "  'inference_gene_symbol': ['IL6', 'STAT3', 'TGFB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Rheumatoid arthritis',\n",
       "  'kegg_pathway_id': 'hsa05323',\n",
       "  'inference_gene_symbol': ['CCL2', 'CXCL8', 'IL6', 'TGFB1', 'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Graft-versus-host disease',\n",
       "  'kegg_pathway_id': 'hsa05332',\n",
       "  'inference_gene_symbol': 'IL6'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Hypertrophic cardiomyopathy (HCM)',\n",
       "  'kegg_pathway_id': 'hsa05410',\n",
       "  'inference_gene_symbol': ['IL6', 'ITGA5', 'ITGB1', 'TGFB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Arrhythmogenic right ventricular cardiomyopathy (ARVC)',\n",
       "  'kegg_pathway_id': 'hsa05412',\n",
       "  'inference_gene_symbol': ['ITGA5', 'ITGB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Dilated cardiomyopathy',\n",
       "  'kegg_pathway_id': 'hsa05414',\n",
       "  'inference_gene_symbol': ['ITGA5', 'ITGB1', 'TGFB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Viral myocarditis',\n",
       "  'kegg_pathway_id': 'hsa05416',\n",
       "  'inference_gene_symbol': 'CCND1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Fluid shear stress and atherosclerosis',\n",
       "  'kegg_pathway_id': 'hsa05418',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CCL2',\n",
       "   'GSTM1',\n",
       "   'MMP9',\n",
       "   'NFE2L2',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'TP53',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Glycolysis (Embden-Meyerhof pathway), glucose => pyruvate',\n",
       "  'kegg_pathway_id': 'hsa_M00001',\n",
       "  'inference_gene_symbol': ['ENO1', 'GAPDH', 'PGAM1', 'TPI1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Glycolysis, core module involving three-carbon compounds',\n",
       "  'kegg_pathway_id': 'hsa_M00002',\n",
       "  'inference_gene_symbol': ['ENO1', 'GAPDH', 'PGAM1', 'TPI1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Gluconeogenesis, oxaloacetate => fructose-6P',\n",
       "  'kegg_pathway_id': 'hsa_M00003',\n",
       "  'inference_gene_symbol': ['ENO1', 'GAPDH', 'PGAM1', 'TPI1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Uridine monophosphate biosynthesis, glutamine (+ PRPP) => UMP',\n",
       "  'kegg_pathway_id': 'hsa_M00051',\n",
       "  'inference_gene_symbol': 'UMPS'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Inositol phosphate metabolism, Ins(1,3,4,5)P4 => Ins(1,3,4)P3 => myo-inositol',\n",
       "  'kegg_pathway_id': 'hsa_M00131',\n",
       "  'inference_gene_symbol': 'INPP5A'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Holo-TFIIH complex',\n",
       "  'kegg_pathway_id': 'hsa_M00290',\n",
       "  'inference_gene_symbol': 'ERCC2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'BRCA1-associated genome surveillance complex (BASC)',\n",
       "  'kegg_pathway_id': 'hsa_M00295',\n",
       "  'inference_gene_symbol': 'ATM'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'DNA-PK complex',\n",
       "  'kegg_pathway_id': 'hsa_M00297',\n",
       "  'inference_gene_symbol': 'PRKDC'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Proteasome, 19S regulatory particle (PA700)',\n",
       "  'kegg_pathway_id': 'hsa_M00341',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Spliceosome, Prp19/CDC5L complex',\n",
       "  'kegg_pathway_id': 'hsa_M00353',\n",
       "  'inference_gene_symbol': 'HSPA8'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Spliceosome, 35S U5-snRNP',\n",
       "  'kegg_pathway_id': 'hsa_M00355',\n",
       "  'inference_gene_symbol': 'HSPA8'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'PI3K-Akt signaling',\n",
       "  'kegg_pathway_id': 'hsa_M00676',\n",
       "  'inference_gene_symbol': ['AKT1', 'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TGF-beta signaling',\n",
       "  'kegg_pathway_id': 'hsa_M00680',\n",
       "  'inference_gene_symbol': 'TGFB1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Notch signaling',\n",
       "  'kegg_pathway_id': 'hsa_M00682',\n",
       "  'inference_gene_symbol': ['NOTCH1', 'NOTCH2', 'NOTCH3', 'NOTCH4']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'JAK-STAT signaling',\n",
       "  'kegg_pathway_id': 'hsa_M00684',\n",
       "  'inference_gene_symbol': 'STAT3'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'MAPK (ERK1/2) signaling',\n",
       "  'kegg_pathway_id': 'hsa_M00687',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'DNA damage-induced cell cycle checkpoints',\n",
       "  'kegg_pathway_id': 'hsa_M00691',\n",
       "  'inference_gene_symbol': 'ATM'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cell cycle - G1/S transition',\n",
       "  'kegg_pathway_id': 'hsa_M00692',\n",
       "  'inference_gene_symbol': 'CCND1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Interleukin-6 signaling',\n",
       "  'react_pathway_id': 'R-HSA-1059683',\n",
       "  'inference_gene_symbol': ['IL6', 'PTPN11', 'STAT3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Apoptosis',\n",
       "  'react_pathway_id': 'R-HSA-109581',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CDH1',\n",
       "   'PTK2',\n",
       "   'SEM1',\n",
       "   'SFN',\n",
       "   'TP53',\n",
       "   'TP63']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Hemostasis',\n",
       "  'react_pathway_id': 'R-HSA-109582',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'ANXA2',\n",
       "   'ANXA5',\n",
       "   'CREBBP',\n",
       "   'EP300',\n",
       "   'HDAC1',\n",
       "   'HDAC2',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'IFNA2',\n",
       "   'ITGA5',\n",
       "   'ITGB1',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'PPIA',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'TGFB1',\n",
       "   'TP53',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Intrinsic Pathway for Apoptosis',\n",
       "  'react_pathway_id': 'R-HSA-109606',\n",
       "  'inference_gene_symbol': ['AKT1', 'SFN', 'TP53', 'TP63']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'PKB-mediated events',\n",
       "  'react_pathway_id': 'R-HSA-109703',\n",
       "  'inference_gene_symbol': ['AKT1', 'STK11', 'TSC1', 'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'PI3K Cascade',\n",
       "  'react_pathway_id': 'R-HSA-109704',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'FGFR1',\n",
       "   'PIK3CA',\n",
       "   'PTPN11',\n",
       "   'STK11',\n",
       "   'TSC1',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'MAPK3 (ERK1) activation',\n",
       "  'react_pathway_id': 'R-HSA-110056',\n",
       "  'inference_gene_symbol': ['IL6', 'MAPK3', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Translesion synthesis by Y family DNA polymerases bypasses lesions on DNA template',\n",
       "  'react_pathway_id': 'R-HSA-110313',\n",
       "  'inference_gene_symbol': 'UBA7'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Activation of BAD and translocation to mitochondria',\n",
       "  'react_pathway_id': 'R-HSA-111447',\n",
       "  'inference_gene_symbol': ['AKT1', 'SFN']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Activation of NOXA and translocation to mitochondria',\n",
       "  'react_pathway_id': 'R-HSA-111448',\n",
       "  'inference_gene_symbol': 'TP53'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Apoptotic cleavage of cellular proteins',\n",
       "  'react_pathway_id': 'R-HSA-111465',\n",
       "  'inference_gene_symbol': ['CDH1', 'PTK2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Opioid Signalling',\n",
       "  'react_pathway_id': 'R-HSA-111885',\n",
       "  'inference_gene_symbol': 'MAPK1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'phospho-PLA2 pathway',\n",
       "  'react_pathway_id': 'R-HSA-111995',\n",
       "  'inference_gene_symbol': 'MAPK1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Ca-dependent events',\n",
       "  'react_pathway_id': 'R-HSA-111996',\n",
       "  'inference_gene_symbol': 'MAPK1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'G-protein mediated events',\n",
       "  'react_pathway_id': 'R-HSA-112040',\n",
       "  'inference_gene_symbol': 'MAPK1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'PLC beta mediated events',\n",
       "  'react_pathway_id': 'R-HSA-112043',\n",
       "  'inference_gene_symbol': 'MAPK1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Neurotransmitter Release Cycle',\n",
       "  'react_pathway_id': 'R-HSA-112310',\n",
       "  'inference_gene_symbol': 'HSPA8'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Neurotransmitter Receptor Binding And Downstream Transmission In The  Postsynaptic Cell',\n",
       "  'react_pathway_id': 'R-HSA-112314',\n",
       "  'inference_gene_symbol': ['HRAS', 'MAPK1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Transmission across Chemical Synapses',\n",
       "  'react_pathway_id': 'R-HSA-112315',\n",
       "  'inference_gene_symbol': ['HRAS', 'HSPA8', 'MAPK1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Neuronal System',\n",
       "  'react_pathway_id': 'R-HSA-112316',\n",
       "  'inference_gene_symbol': ['HRAS', 'HSPA8', 'MAPK1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Formation of RNA Pol II elongation complex',\n",
       "  'react_pathway_id': 'R-HSA-112382',\n",
       "  'inference_gene_symbol': 'ERCC2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'IRS-mediated signalling',\n",
       "  'react_pathway_id': 'R-HSA-112399',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'STK11',\n",
       "   'TSC1',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RAF-independent MAPK1/3 activation',\n",
       "  'react_pathway_id': 'R-HSA-112409',\n",
       "  'inference_gene_symbol': ['IL6', 'MAPK1', 'MAPK3', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'MAPK1 (ERK2) activation',\n",
       "  'react_pathway_id': 'R-HSA-112411',\n",
       "  'inference_gene_symbol': ['IL6', 'MAPK1', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'SOS-mediated signalling',\n",
       "  'react_pathway_id': 'R-HSA-112412',\n",
       "  'inference_gene_symbol': ['EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PTK2',\n",
       "   'RASA1',\n",
       "   'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Formation of the Early Elongation Complex',\n",
       "  'react_pathway_id': 'R-HSA-113418',\n",
       "  'inference_gene_symbol': 'ERCC2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'E2F mediated regulation of DNA replication',\n",
       "  'react_pathway_id': 'R-HSA-113510',\n",
       "  'inference_gene_symbol': 'TYMS'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Activation of BH3-only proteins',\n",
       "  'react_pathway_id': 'R-HSA-114452',\n",
       "  'inference_gene_symbol': ['AKT1', 'SFN', 'TP53', 'TP63']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'GPVI-mediated activation cascade',\n",
       "  'react_pathway_id': 'R-HSA-114604',\n",
       "  'inference_gene_symbol': ['AKT1', 'PIK3CA', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Platelet degranulation',\n",
       "  'react_pathway_id': 'R-HSA-114608',\n",
       "  'inference_gene_symbol': ['ANXA5', 'HGF', 'PPIA', 'TGFB1', 'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Downstream signaling events of B Cell Receptor (BCR)',\n",
       "  'react_pathway_id': 'R-HSA-1168372',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTPN11',\n",
       "   'SEM1',\n",
       "   'TP53',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Activation of NF-kappaB in B cells',\n",
       "  'react_pathway_id': 'R-HSA-1169091',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Activation of RAS in B cells',\n",
       "  'react_pathway_id': 'R-HSA-1169092',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'ISG15 antiviral mechanism',\n",
       "  'react_pathway_id': 'R-HSA-1169408',\n",
       "  'inference_gene_symbol': ['MAPK3', 'UBA7']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Antiviral mechanism by IFN-stimulated genes',\n",
       "  'react_pathway_id': 'R-HSA-1169410',\n",
       "  'inference_gene_symbol': ['MAPK3', 'UBA7']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Prolactin receptor signaling',\n",
       "  'react_pathway_id': 'R-HSA-1170546',\n",
       "  'inference_gene_symbol': 'PTPN11'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by FGFR in disease',\n",
       "  'react_pathway_id': 'R-HSA-1226099',\n",
       "  'inference_gene_symbol': ['FGFR1', 'HRAS', 'KRAS', 'PIK3CA', 'STAT3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by ERBB2',\n",
       "  'react_pathway_id': 'R-HSA-1227986',\n",
       "  'inference_gene_symbol': ['AKT1', 'EGFR', 'HRAS', 'KRAS', 'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of gene expression by Hypoxia-inducible Factor',\n",
       "  'react_pathway_id': 'R-HSA-1234158',\n",
       "  'inference_gene_symbol': ['CREBBP', 'EP300', 'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of Hypoxia-inducible Factor (HIF) by oxygen',\n",
       "  'react_pathway_id': 'R-HSA-1234174',\n",
       "  'inference_gene_symbol': ['CREBBP', 'EP300', 'SEM1', 'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Oxygen-dependent proline hydroxylation of Hypoxia-inducible Factor Alpha',\n",
       "  'react_pathway_id': 'R-HSA-1234176',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Constitutive Signaling by Ligand-Responsive EGFR Cancer Variants',\n",
       "  'react_pathway_id': 'R-HSA-1236382',\n",
       "  'inference_gene_symbol': ['EGFR', 'HRAS', 'KRAS', 'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by ERBB4',\n",
       "  'react_pathway_id': 'R-HSA-1236394',\n",
       "  'inference_gene_symbol': ['EGFR', 'HRAS', 'KRAS', 'PIK3CA', 'WWOX']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'ER-Phagosome pathway',\n",
       "  'react_pathway_id': 'R-HSA-1236974',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Antigen processing-Cross presentation',\n",
       "  'react_pathway_id': 'R-HSA-1236975',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cross-presentation of soluble exogenous antigens (endosomes)',\n",
       "  'react_pathway_id': 'R-HSA-1236978',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'SHC1 events in ERBB2 signaling',\n",
       "  'react_pathway_id': 'R-HSA-1250196',\n",
       "  'inference_gene_symbol': ['EGFR', 'HRAS', 'KRAS']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'PI3K events in ERBB4 signaling',\n",
       "  'react_pathway_id': 'R-HSA-1250342',\n",
       "  'inference_gene_symbol': 'PIK3CA'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'SHC1 events in ERBB4 signaling',\n",
       "  'react_pathway_id': 'R-HSA-1250347',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'PLCG1 events in ERBB2 signaling',\n",
       "  'react_pathway_id': 'R-HSA-1251932',\n",
       "  'inference_gene_symbol': 'EGFR'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Nuclear signaling by ERBB4',\n",
       "  'react_pathway_id': 'R-HSA-1251985',\n",
       "  'inference_gene_symbol': 'WWOX'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'PIP3 activates AKT signaling',\n",
       "  'react_pathway_id': 'R-HSA-1257604',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTPN11',\n",
       "   'TP53',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Interleukin-7 signaling',\n",
       "  'react_pathway_id': 'R-HSA-1266695',\n",
       "  'inference_gene_symbol': 'HGF'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Developmental Biology',\n",
       "  'react_pathway_id': 'R-HSA-1266738',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CEBPA',\n",
       "   'CREBBP',\n",
       "   'CSNK2A1',\n",
       "   'CSNK2A2',\n",
       "   'CSNK2B',\n",
       "   'EGFR',\n",
       "   'EP300',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'HSPA8',\n",
       "   'ITGA5',\n",
       "   'ITGB1',\n",
       "   'IVL',\n",
       "   'KLF5',\n",
       "   'KRAS',\n",
       "   'KRT1',\n",
       "   'KRT10',\n",
       "   'KRT14',\n",
       "   'KRT6B',\n",
       "   'LORICRIN',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'MMP9',\n",
       "   'NCOA3',\n",
       "   'NOTCH1',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'RARB',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'SOX2',\n",
       "   'SPRR1A',\n",
       "   'STAT3',\n",
       "   'TGFB1',\n",
       "   'TIAM1',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cytokine Signaling in Immune system',\n",
       "  'react_pathway_id': 'R-HSA-1280215',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'ANXA1',\n",
       "   'BIRC5',\n",
       "   'CCL2',\n",
       "   'CCND1',\n",
       "   'CXCL8',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'HSPA8',\n",
       "   'IFNA2',\n",
       "   'IL6',\n",
       "   'ITGB1',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'MMP9',\n",
       "   'MYC',\n",
       "   'PIK3CA',\n",
       "   'PTGS2',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'SOX2',\n",
       "   'STAT3',\n",
       "   'TGFB1',\n",
       "   'TP53',\n",
       "   'UBA7',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Adaptive Immune System',\n",
       "  'react_pathway_id': 'R-HSA-1280218',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CDH1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'ITGB1',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTPN11',\n",
       "   'SEM1',\n",
       "   'TP53',\n",
       "   'TSC2',\n",
       "   'UBA1',\n",
       "   'UBA7']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Spry regulation of FGF signaling',\n",
       "  'react_pathway_id': 'R-HSA-1295596',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Downregulation of ERBB2:ERBB3 signaling',\n",
       "  'react_pathway_id': 'R-HSA-1358803',\n",
       "  'inference_gene_symbol': 'AKT1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RORA activates gene expression',\n",
       "  'react_pathway_id': 'R-HSA-1368082',\n",
       "  'inference_gene_symbol': ['CREBBP', 'EP300']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'BMAL1:CLOCK,NPAS2 activates circadian gene expression',\n",
       "  'react_pathway_id': 'R-HSA-1368108',\n",
       "  'inference_gene_symbol': ['CREBBP', 'EP300']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Activation of PUMA and translocation to mitochondria',\n",
       "  'react_pathway_id': 'R-HSA-139915',\n",
       "  'inference_gene_symbol': ['TP53', 'TP63']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Metabolism',\n",
       "  'react_pathway_id': 'R-HSA-1430728',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CREBBP',\n",
       "   'CSNK2A1',\n",
       "   'CSNK2A2',\n",
       "   'CSNK2B',\n",
       "   'CYP1A2',\n",
       "   'CYP1B1',\n",
       "   'ENO1',\n",
       "   'EP300',\n",
       "   'ERCC2',\n",
       "   'GAPDH',\n",
       "   'GPX1',\n",
       "   'GSTM1',\n",
       "   'INPP5A',\n",
       "   'MT1A',\n",
       "   'NCOA3',\n",
       "   'PGAM1',\n",
       "   'PIK3CA',\n",
       "   'PTGS2',\n",
       "   'SEM1',\n",
       "   'SLC45A2',\n",
       "   'STK11',\n",
       "   'TPI1',\n",
       "   'TYMS',\n",
       "   'UMPS']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by SCF-KIT',\n",
       "  'react_pathway_id': 'R-HSA-1433557',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'MMP9',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'STAT3',\n",
       "   'TP53',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Collagen degradation',\n",
       "  'react_pathway_id': 'R-HSA-1442490',\n",
       "  'inference_gene_symbol': 'MMP9'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Translocation of GLUT4 to the plasma membrane',\n",
       "  'react_pathway_id': 'R-HSA-1445148',\n",
       "  'inference_gene_symbol': ['AKT1', 'SFN']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Tetrahydrobiopterin (BH4) synthesis, recycling, salvage and regulation',\n",
       "  'react_pathway_id': 'R-HSA-1474151',\n",
       "  'inference_gene_symbol': 'AKT1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Degradation of the extracellular matrix',\n",
       "  'react_pathway_id': 'R-HSA-1474228',\n",
       "  'inference_gene_symbol': ['CDH1', 'MMP9']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Extracellular matrix organization',\n",
       "  'react_pathway_id': 'R-HSA-1474244',\n",
       "  'inference_gene_symbol': ['CDH1',\n",
       "   'DSPP',\n",
       "   'ITGA5',\n",
       "   'ITGB1',\n",
       "   'MMP9',\n",
       "   'TGFB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Collagen formation',\n",
       "  'react_pathway_id': 'R-HSA-1474290',\n",
       "  'inference_gene_symbol': 'MMP9'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Synthesis of PC',\n",
       "  'react_pathway_id': 'R-HSA-1483191',\n",
       "  'inference_gene_symbol': ['CSNK2A1', 'CSNK2A2', 'CSNK2B']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Glycerophospholipid biosynthesis',\n",
       "  'react_pathway_id': 'R-HSA-1483206',\n",
       "  'inference_gene_symbol': ['CSNK2A1', 'CSNK2A2', 'CSNK2B']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Inositol phosphate metabolism',\n",
       "  'react_pathway_id': 'R-HSA-1483249',\n",
       "  'inference_gene_symbol': 'INPP5A'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'PI Metabolism',\n",
       "  'react_pathway_id': 'R-HSA-1483255',\n",
       "  'inference_gene_symbol': 'PIK3CA'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Phospholipid metabolism',\n",
       "  'react_pathway_id': 'R-HSA-1483257',\n",
       "  'inference_gene_symbol': ['CSNK2A1', 'CSNK2A2', 'CSNK2B', 'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Meiosis',\n",
       "  'react_pathway_id': 'R-HSA-1500620',\n",
       "  'inference_gene_symbol': 'ATM'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cell-Cell communication',\n",
       "  'react_pathway_id': 'R-HSA-1500931',\n",
       "  'inference_gene_symbol': ['CDH1', 'ITGB1', 'PIK3CA', 'PTK2', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'G0 and Early G1',\n",
       "  'react_pathway_id': 'R-HSA-1538133',\n",
       "  'inference_gene_symbol': 'HDAC1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Phase II conjugation',\n",
       "  'react_pathway_id': 'R-HSA-156580',\n",
       "  'inference_gene_symbol': ['CYP1A2', 'GSTM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Methylation',\n",
       "  'react_pathway_id': 'R-HSA-156581',\n",
       "  'inference_gene_symbol': 'CYP1A2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Glutathione conjugation',\n",
       "  'react_pathway_id': 'R-HSA-156590',\n",
       "  'inference_gene_symbol': 'GSTM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Elastic fibre formation',\n",
       "  'react_pathway_id': 'R-HSA-1566948',\n",
       "  'inference_gene_symbol': ['ITGA5', 'ITGB1', 'TGFB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Fibronectin matrix formation',\n",
       "  'react_pathway_id': 'R-HSA-1566977',\n",
       "  'inference_gene_symbol': ['ITGA5', 'ITGB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Polo-like kinase mediated events',\n",
       "  'react_pathway_id': 'R-HSA-156711',\n",
       "  'inference_gene_symbol': 'EP300'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by NOTCH',\n",
       "  'react_pathway_id': 'R-HSA-157118',\n",
       "  'inference_gene_symbol': ['CCND1',\n",
       "   'CREBBP',\n",
       "   'EP300',\n",
       "   'HDAC1',\n",
       "   'HDAC2',\n",
       "   'MYC',\n",
       "   'NOTCH1',\n",
       "   'NOTCH2',\n",
       "   'NOTCH3',\n",
       "   'NOTCH4',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Metabolism of nucleotides',\n",
       "  'react_pathway_id': 'R-HSA-15869',\n",
       "  'inference_gene_symbol': ['GPX1', 'TYMS', 'UMPS']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Mitochondrial biogenesis',\n",
       "  'react_pathway_id': 'R-HSA-1592230',\n",
       "  'inference_gene_symbol': ['CREBBP', 'SIRT3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Activation of Matrix Metalloproteinases',\n",
       "  'react_pathway_id': 'R-HSA-1592389',\n",
       "  'inference_gene_symbol': 'MMP9'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signal Transduction',\n",
       "  'react_pathway_id': 'R-HSA-162582',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'ANXA1',\n",
       "   'BIRC5',\n",
       "   'CCND1',\n",
       "   'CDH1',\n",
       "   'CREBBP',\n",
       "   'CSNK2A1',\n",
       "   'CSNK2A2',\n",
       "   'CSNK2B',\n",
       "   'CTTN',\n",
       "   'CXCL8',\n",
       "   'EGFR',\n",
       "   'EP300',\n",
       "   'FGFR1',\n",
       "   'HCAR2',\n",
       "   'HCAR3',\n",
       "   'HDAC1',\n",
       "   'HDAC2',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'HSPB1',\n",
       "   'IL6',\n",
       "   'ITGB1',\n",
       "   'KRAS',\n",
       "   'LGR6',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'MMP9',\n",
       "   'MYC',\n",
       "   'NCOA3',\n",
       "   'NOTCH1',\n",
       "   'NOTCH2',\n",
       "   'NOTCH3',\n",
       "   'NOTCH4',\n",
       "   'PIK3CA',\n",
       "   'PTGER1',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'RACK1',\n",
       "   'RARB',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'SFN',\n",
       "   'SOX2',\n",
       "   'STAT3',\n",
       "   'STK11',\n",
       "   'TGFB1',\n",
       "   'TIAM1',\n",
       "   'TP53',\n",
       "   'TSC1',\n",
       "   'TSC2',\n",
       "   'VEGFA',\n",
       "   'WWOX']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Uncoating of the HIV Virion',\n",
       "  'react_pathway_id': 'R-HSA-162585',\n",
       "  'inference_gene_symbol': 'PPIA'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'HIV Life Cycle',\n",
       "  'react_pathway_id': 'R-HSA-162587',\n",
       "  'inference_gene_symbol': ['ERCC2', 'PPIA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Budding and maturation of HIV virion',\n",
       "  'react_pathway_id': 'R-HSA-162588',\n",
       "  'inference_gene_symbol': 'PPIA'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Reverse Transcription of HIV RNA',\n",
       "  'react_pathway_id': 'R-HSA-162589',\n",
       "  'inference_gene_symbol': 'PPIA'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Integration of provirus',\n",
       "  'react_pathway_id': 'R-HSA-162592',\n",
       "  'inference_gene_symbol': 'PPIA'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Early Phase of HIV Life Cycle',\n",
       "  'react_pathway_id': 'R-HSA-162594',\n",
       "  'inference_gene_symbol': 'PPIA'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Late Phase of HIV Life Cycle',\n",
       "  'react_pathway_id': 'R-HSA-162599',\n",
       "  'inference_gene_symbol': ['ERCC2', 'PPIA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Golgi Cisternae Pericentriolar Stack Reorganization',\n",
       "  'react_pathway_id': 'R-HSA-162658',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'HIV Infection',\n",
       "  'react_pathway_id': 'R-HSA-162906',\n",
       "  'inference_gene_symbol': ['ERCC2', 'PPIA', 'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Host Interactions of HIV factors',\n",
       "  'react_pathway_id': 'R-HSA-162909',\n",
       "  'inference_gene_symbol': ['PPIA', 'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Macroautophagy',\n",
       "  'react_pathway_id': 'R-HSA-1632852',\n",
       "  'inference_gene_symbol': ['TSC1', 'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'AMPK inhibits chREBP transcriptional activation activity',\n",
       "  'react_pathway_id': 'R-HSA-163680',\n",
       "  'inference_gene_symbol': 'STK11'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Integration of energy metabolism',\n",
       "  'react_pathway_id': 'R-HSA-163685',\n",
       "  'inference_gene_symbol': 'STK11'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cell Cycle',\n",
       "  'react_pathway_id': 'R-HSA-1640170',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'ATM',\n",
       "   'BIRC5',\n",
       "   'CCND1',\n",
       "   'CDKN2A',\n",
       "   'CSNK2A1',\n",
       "   'CSNK2A2',\n",
       "   'CSNK2B',\n",
       "   'EP300',\n",
       "   'HDAC1',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MYC',\n",
       "   'SEM1',\n",
       "   'SFN',\n",
       "   'TP53',\n",
       "   'TYMS']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Disease',\n",
       "  'react_pathway_id': 'R-HSA-1643685',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CDH1',\n",
       "   'CREBBP',\n",
       "   'EGFR',\n",
       "   'EP300',\n",
       "   'ERCC2',\n",
       "   'FGFR1',\n",
       "   'HDAC1',\n",
       "   'HDAC2',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'MYC',\n",
       "   'NOTCH1',\n",
       "   'NOTCH2',\n",
       "   'NOTCH3',\n",
       "   'NOTCH4',\n",
       "   'PIK3CA',\n",
       "   'PPIA',\n",
       "   'PTPN11',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'SOD2',\n",
       "   'STAT3',\n",
       "   'TGFB1',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by EGFR in Cancer',\n",
       "  'react_pathway_id': 'R-HSA-1643713',\n",
       "  'inference_gene_symbol': ['EGFR', 'HRAS', 'KRAS', 'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Minus-strand DNA synthesis',\n",
       "  'react_pathway_id': 'R-HSA-164516',\n",
       "  'inference_gene_symbol': 'PPIA'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Plus-strand DNA synthesis',\n",
       "  'react_pathway_id': 'R-HSA-164525',\n",
       "  'inference_gene_symbol': 'PPIA'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'mTOR signalling',\n",
       "  'react_pathway_id': 'R-HSA-165159',\n",
       "  'inference_gene_symbol': ['AKT1', 'STK11', 'TSC1', 'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Inhibition of TSC complex formation by PKB',\n",
       "  'react_pathway_id': 'R-HSA-165181',\n",
       "  'inference_gene_symbol': ['TSC1', 'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of cholesterol biosynthesis by SREBP (SREBF)',\n",
       "  'react_pathway_id': 'R-HSA-1655829',\n",
       "  'inference_gene_symbol': 'CREBBP'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Toll Like Receptor 4 (TLR4) Cascade',\n",
       "  'react_pathway_id': 'R-HSA-166016',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Synthesis of PIPs at the plasma membrane',\n",
       "  'react_pathway_id': 'R-HSA-1660499',\n",
       "  'inference_gene_symbol': 'PIK3CA'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Activated TLR4 signalling',\n",
       "  'react_pathway_id': 'R-HSA-166054',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'MyD88:Mal cascade initiated on plasma membrane',\n",
       "  'react_pathway_id': 'R-HSA-166058',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'MyD88-independent TLR3/TLR4 cascade',\n",
       "  'react_pathway_id': 'R-HSA-166166',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signalling by NGF',\n",
       "  'react_pathway_id': 'R-HSA-166520',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HDAC1',\n",
       "   'HDAC2',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'STAT3',\n",
       "   'TIAM1',\n",
       "   'TP53',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signalling to RAS',\n",
       "  'react_pathway_id': 'R-HSA-167044',\n",
       "  'inference_gene_symbol': ['EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PTK2',\n",
       "   'RASA1',\n",
       "   'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Formation of HIV elongation complex in the absence of HIV Tat',\n",
       "  'react_pathway_id': 'R-HSA-167152',\n",
       "  'inference_gene_symbol': 'ERCC2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Formation of the HIV-1 Early Elongation Complex',\n",
       "  'react_pathway_id': 'R-HSA-167158',\n",
       "  'inference_gene_symbol': 'ERCC2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RNA Pol II CTD phosphorylation and interaction with CE during HIV infection',\n",
       "  'react_pathway_id': 'R-HSA-167160',\n",
       "  'inference_gene_symbol': 'ERCC2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'HIV Transcription Initiation',\n",
       "  'react_pathway_id': 'R-HSA-167161',\n",
       "  'inference_gene_symbol': 'ERCC2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RNA Polymerase II HIV Promoter Escape',\n",
       "  'react_pathway_id': 'R-HSA-167162',\n",
       "  'inference_gene_symbol': 'ERCC2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'HIV Transcription Elongation',\n",
       "  'react_pathway_id': 'R-HSA-167169',\n",
       "  'inference_gene_symbol': 'ERCC2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Transcription of the HIV genome',\n",
       "  'react_pathway_id': 'R-HSA-167172',\n",
       "  'inference_gene_symbol': 'ERCC2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Formation of HIV-1 elongation complex containing HIV-1 Tat',\n",
       "  'react_pathway_id': 'R-HSA-167200',\n",
       "  'inference_gene_symbol': 'ERCC2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Tat-mediated elongation of the HIV-1 transcript',\n",
       "  'react_pathway_id': 'R-HSA-167246',\n",
       "  'inference_gene_symbol': 'ERCC2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Toll Like Receptor 9 (TLR9) Cascade',\n",
       "  'react_pathway_id': 'R-HSA-168138',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Toll Like Receptor 10 (TLR10) Cascade',\n",
       "  'react_pathway_id': 'R-HSA-168142',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Toll Like Receptor 3 (TLR3) Cascade',\n",
       "  'react_pathway_id': 'R-HSA-168164',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Toll Like Receptor 5 (TLR5) Cascade',\n",
       "  'react_pathway_id': 'R-HSA-168176',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Toll Like Receptor TLR1:TLR2 Cascade',\n",
       "  'react_pathway_id': 'R-HSA-168179',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Toll Like Receptor 7/8 (TLR7/8) Cascade',\n",
       "  'react_pathway_id': 'R-HSA-168181',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Toll Like Receptor TLR6:TLR2 Cascade',\n",
       "  'react_pathway_id': 'R-HSA-168188',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Innate Immune System',\n",
       "  'react_pathway_id': 'R-HSA-168249',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'ANXA2',\n",
       "   'CREBBP',\n",
       "   'CSNK2B',\n",
       "   'EGFR',\n",
       "   'EP300',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'HSPA8',\n",
       "   'IFNA2',\n",
       "   'KRAS',\n",
       "   'KRT1',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'MMP9',\n",
       "   'PGAM1',\n",
       "   'PIK3CA',\n",
       "   'PPIA',\n",
       "   'PRKDC',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'PYCARD',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'SERPINB3',\n",
       "   'TP53',\n",
       "   'TSC2',\n",
       "   'UBA7']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Host Interactions with Influenza Factors',\n",
       "  'react_pathway_id': 'R-HSA-168253',\n",
       "  'inference_gene_symbol': 'TGFB1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Influenza Infection',\n",
       "  'react_pathway_id': 'R-HSA-168254',\n",
       "  'inference_gene_symbol': 'TGFB1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Immune System',\n",
       "  'react_pathway_id': 'R-HSA-168256',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'ANXA1',\n",
       "   'ANXA2',\n",
       "   'BIRC5',\n",
       "   'CCL2',\n",
       "   'CCND1',\n",
       "   'CDH1',\n",
       "   'CREBBP',\n",
       "   'CSNK2B',\n",
       "   'CXCL8',\n",
       "   'EGFR',\n",
       "   'EP300',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'HSPA8',\n",
       "   'IFNA2',\n",
       "   'IL6',\n",
       "   'ITGB1',\n",
       "   'KRAS',\n",
       "   'KRT1',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'MMP9',\n",
       "   'MYC',\n",
       "   'PGAM1',\n",
       "   'PIK3CA',\n",
       "   'PPIA',\n",
       "   'PRKDC',\n",
       "   'PTGS2',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'PYCARD',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'SERPINB3',\n",
       "   'SOX2',\n",
       "   'STAT3',\n",
       "   'TGFB1',\n",
       "   'TP53',\n",
       "   'TSC2',\n",
       "   'UBA1',\n",
       "   'UBA7',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Influenza Virus Induced Apoptosis',\n",
       "  'react_pathway_id': 'R-HSA-168277',\n",
       "  'inference_gene_symbol': 'TGFB1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Nucleotide-binding domain, leucine rich repeat containing receptor (NLR) signaling pathways',\n",
       "  'react_pathway_id': 'R-HSA-168643',\n",
       "  'inference_gene_symbol': 'PYCARD'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Toll-Like Receptors Cascades',\n",
       "  'react_pathway_id': 'R-HSA-168898',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RIG-I/MDA5 mediated induction of IFN-alpha/beta pathways',\n",
       "  'react_pathway_id': 'R-HSA-168928',\n",
       "  'inference_gene_symbol': ['CREBBP', 'EP300', 'IFNA2', 'UBA7']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Prolonged ERK activation events',\n",
       "  'react_pathway_id': 'R-HSA-169893',\n",
       "  'inference_gene_symbol': ['EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PTK2',\n",
       "   'RASA1',\n",
       "   'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of Apoptosis',\n",
       "  'react_pathway_id': 'R-HSA-169911',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by TGF-beta Receptor Complex',\n",
       "  'react_pathway_id': 'R-HSA-170834',\n",
       "  'inference_gene_symbol': ['HDAC1', 'MYC', 'TGFB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Frs2-mediated activation',\n",
       "  'react_pathway_id': 'R-HSA-170968',\n",
       "  'inference_gene_symbol': ['EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PTK2',\n",
       "   'RASA1',\n",
       "   'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'ARMS-mediated activation',\n",
       "  'react_pathway_id': 'R-HSA-170984',\n",
       "  'inference_gene_symbol': ['EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PTK2',\n",
       "   'RASA1',\n",
       "   'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'p38MAPK events',\n",
       "  'react_pathway_id': 'R-HSA-171007',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Binding and entry of HIV virion',\n",
       "  'react_pathway_id': 'R-HSA-173107',\n",
       "  'inference_gene_symbol': 'PPIA'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Autodegradation of Cdh1 by Cdh1:APC/C',\n",
       "  'react_pathway_id': 'R-HSA-174084',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'SCF-beta-TrCP mediated degradation of Emi1',\n",
       "  'react_pathway_id': 'R-HSA-174113',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'APC/C-mediated degradation of cell cycle proteins',\n",
       "  'react_pathway_id': 'R-HSA-174143',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'APC/C:Cdc20 mediated degradation of Securin',\n",
       "  'react_pathway_id': 'R-HSA-174154',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'APC/C:Cdh1 mediated degradation of Cdc20 and other APC/C:Cdh1 targeted proteins in late mitosis/early G1',\n",
       "  'react_pathway_id': 'R-HSA-174178',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cdc20:Phospho-APC/C mediated degradation of Cyclin A',\n",
       "  'react_pathway_id': 'R-HSA-174184',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Assembly Of The HIV Virion',\n",
       "  'react_pathway_id': 'R-HSA-175474',\n",
       "  'inference_gene_symbol': 'PPIA'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of APC/C activators between G1/S and early anaphase',\n",
       "  'react_pathway_id': 'R-HSA-176408',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'APC/C:Cdc20 mediated degradation of mitotic proteins',\n",
       "  'react_pathway_id': 'R-HSA-176409',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Activation of APC/C and APC/C:Cdc20 mediated degradation of mitotic proteins',\n",
       "  'react_pathway_id': 'R-HSA-176814',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by EGFR',\n",
       "  'react_pathway_id': 'R-HSA-177929',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'TP53',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'APC:Cdc20 mediated degradation of cell cycle proteins prior to satisfation of the cell cycle checkpoint',\n",
       "  'react_pathway_id': 'R-HSA-179419',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'GRB2 events in EGFR signaling',\n",
       "  'react_pathway_id': 'R-HSA-179812',\n",
       "  'inference_gene_symbol': ['EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PTK2',\n",
       "   'RASA1',\n",
       "   'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'GAB1 signalosome',\n",
       "  'react_pathway_id': 'R-HSA-180292',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTPN11',\n",
       "   'TP53',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'SHC1 events in EGFR signaling',\n",
       "  'react_pathway_id': 'R-HSA-180336',\n",
       "  'inference_gene_symbol': ['EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PTK2',\n",
       "   'RASA1',\n",
       "   'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Vpu mediated degradation of CD4',\n",
       "  'react_pathway_id': 'R-HSA-180534',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Vif-mediated degradation of APOBEC3G',\n",
       "  'react_pathway_id': 'R-HSA-180585',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'APOBEC3G mediated resistance to HIV-1 infection',\n",
       "  'react_pathway_id': 'R-HSA-180689',\n",
       "  'inference_gene_symbol': 'PPIA'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Toll Like Receptor 2 (TLR2) Cascade',\n",
       "  'react_pathway_id': 'R-HSA-181438',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'EGFR downregulation',\n",
       "  'react_pathway_id': 'R-HSA-182971',\n",
       "  'inference_gene_symbol': 'EGFR'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'STING mediated induction of host immune responses',\n",
       "  'react_pathway_id': 'R-HSA-1834941',\n",
       "  'inference_gene_symbol': 'PRKDC'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cytosolic sensors of pathogen-associated DNA',\n",
       "  'react_pathway_id': 'R-HSA-1834949',\n",
       "  'inference_gene_symbol': ['CREBBP', 'EP300', 'PRKDC']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by cytosolic FGFR1 fusion mutants',\n",
       "  'react_pathway_id': 'R-HSA-1839117',\n",
       "  'inference_gene_symbol': ['PIK3CA', 'STAT3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by FGFR1 amplification mutants',\n",
       "  'react_pathway_id': 'R-HSA-1839120',\n",
       "  'inference_gene_symbol': 'FGFR1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by activated point mutants of FGFR1',\n",
       "  'react_pathway_id': 'R-HSA-1839122',\n",
       "  'inference_gene_symbol': 'FGFR1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'FGFR1 mutant receptor activation',\n",
       "  'react_pathway_id': 'R-HSA-1839124',\n",
       "  'inference_gene_symbol': ['FGFR1', 'PIK3CA', 'STAT3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Organelle biogenesis and maintenance',\n",
       "  'react_pathway_id': 'R-HSA-1852241',\n",
       "  'inference_gene_symbol': ['CREBBP', 'MRPL13', 'SIRT3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Synthesis of IP2, IP, and Ins in the cytosol',\n",
       "  'react_pathway_id': 'R-HSA-1855183',\n",
       "  'inference_gene_symbol': 'INPP5A'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of beta-cell development',\n",
       "  'react_pathway_id': 'R-HSA-186712',\n",
       "  'inference_gene_symbol': ['AKT1', 'NOTCH1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Downstream signal transduction',\n",
       "  'react_pathway_id': 'R-HSA-186763',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'STAT3',\n",
       "   'TP53',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by PDGF',\n",
       "  'react_pathway_id': 'R-HSA-186797',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'STAT3',\n",
       "   'TP53',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'NGF signalling via TRKA from the plasma membrane',\n",
       "  'react_pathway_id': 'R-HSA-187037',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'STAT3',\n",
       "   'TP53',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'SCF(Skp2)-mediated degradation of p27/p21',\n",
       "  'react_pathway_id': 'R-HSA-187577',\n",
       "  'inference_gene_symbol': ['CCND1', 'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signalling to ERKs',\n",
       "  'react_pathway_id': 'R-HSA-187687',\n",
       "  'inference_gene_symbol': ['EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PTK2',\n",
       "   'RASA1',\n",
       "   'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signalling to p38 via RIT and RIN',\n",
       "  'react_pathway_id': 'R-HSA-187706',\n",
       "  'inference_gene_symbol': ['EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PTK2',\n",
       "   'RASA1',\n",
       "   'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by FGFR',\n",
       "  'react_pathway_id': 'R-HSA-190236',\n",
       "  'inference_gene_symbol': ['FGFR1',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'FGFR1 ligand binding and activation',\n",
       "  'react_pathway_id': 'R-HSA-190242',\n",
       "  'inference_gene_symbol': 'FGFR1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'FGFR1b ligand binding and activation',\n",
       "  'react_pathway_id': 'R-HSA-190370',\n",
       "  'inference_gene_symbol': 'FGFR1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'FGFR1c ligand binding and activation',\n",
       "  'react_pathway_id': 'R-HSA-190373',\n",
       "  'inference_gene_symbol': 'FGFR1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'FGFR1c and Klotho ligand binding and activation',\n",
       "  'react_pathway_id': 'R-HSA-190374',\n",
       "  'inference_gene_symbol': 'FGFR1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Pre-NOTCH Processing in the Endoplasmic Reticulum',\n",
       "  'react_pathway_id': 'R-HSA-1912399',\n",
       "  'inference_gene_symbol': ['NOTCH1', 'NOTCH2', 'NOTCH3', 'NOTCH4']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Pre-NOTCH Transcription and Translation',\n",
       "  'react_pathway_id': 'R-HSA-1912408',\n",
       "  'inference_gene_symbol': ['CCND1',\n",
       "   'CREBBP',\n",
       "   'EP300',\n",
       "   'NOTCH1',\n",
       "   'NOTCH2',\n",
       "   'NOTCH3',\n",
       "   'NOTCH4',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Pre-NOTCH Processing in Golgi',\n",
       "  'react_pathway_id': 'R-HSA-1912420',\n",
       "  'inference_gene_symbol': ['NOTCH1', 'NOTCH2', 'NOTCH3', 'NOTCH4']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Pre-NOTCH Expression and Processing',\n",
       "  'react_pathway_id': 'R-HSA-1912422',\n",
       "  'inference_gene_symbol': ['CCND1',\n",
       "   'CREBBP',\n",
       "   'EP300',\n",
       "   'NOTCH1',\n",
       "   'NOTCH2',\n",
       "   'NOTCH3',\n",
       "   'NOTCH4',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'NRAGE signals death through JNK',\n",
       "  'react_pathway_id': 'R-HSA-193648',\n",
       "  'inference_gene_symbol': 'TIAM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'p75NTR negatively regulates cell cycle via SC1',\n",
       "  'react_pathway_id': 'R-HSA-193670',\n",
       "  'inference_gene_symbol': ['HDAC1', 'HDAC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'p75 NTR receptor-mediated signalling',\n",
       "  'react_pathway_id': 'R-HSA-193704',\n",
       "  'inference_gene_symbol': ['HDAC1', 'HDAC2', 'TIAM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by VEGF',\n",
       "  'react_pathway_id': 'R-HSA-194138',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'HSPB1',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'VEGF ligand-receptor interactions',\n",
       "  'react_pathway_id': 'R-HSA-194313',\n",
       "  'inference_gene_symbol': 'VEGFA'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by Rho GTPases',\n",
       "  'react_pathway_id': 'R-HSA-194315',\n",
       "  'inference_gene_symbol': ['BIRC5',\n",
       "   'CDH1',\n",
       "   'CTTN',\n",
       "   'ITGB1',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PTK2',\n",
       "   'SFN',\n",
       "   'TIAM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Rho GTPase cycle',\n",
       "  'react_pathway_id': 'R-HSA-194840',\n",
       "  'inference_gene_symbol': 'TIAM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Degradation of beta-catenin by the destruction complex',\n",
       "  'react_pathway_id': 'R-HSA-195253',\n",
       "  'inference_gene_symbol': ['HDAC1', 'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RHO GTPase Effectors',\n",
       "  'react_pathway_id': 'R-HSA-195258',\n",
       "  'inference_gene_symbol': ['BIRC5',\n",
       "   'CDH1',\n",
       "   'CTTN',\n",
       "   'ITGB1',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PTK2',\n",
       "   'SFN']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'VEGF binds to VEGFR leading to receptor dimerization',\n",
       "  'react_pathway_id': 'R-HSA-195399',\n",
       "  'inference_gene_symbol': 'VEGFA'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by Wnt',\n",
       "  'react_pathway_id': 'R-HSA-195721',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CREBBP',\n",
       "   'CSNK2A1',\n",
       "   'CSNK2A2',\n",
       "   'CSNK2B',\n",
       "   'EP300',\n",
       "   'HDAC1',\n",
       "   'LGR6',\n",
       "   'MYC',\n",
       "   'SEM1',\n",
       "   'SOX2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'GRB2 events in ERBB2 signaling',\n",
       "  'react_pathway_id': 'R-HSA-1963640',\n",
       "  'inference_gene_symbol': ['EGFR', 'HRAS', 'KRAS']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'PI3K events in ERBB2 signaling',\n",
       "  'react_pathway_id': 'R-HSA-1963642',\n",
       "  'inference_gene_symbol': ['EGFR', 'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Nicotinate metabolism',\n",
       "  'react_pathway_id': 'R-HSA-196807',\n",
       "  'inference_gene_symbol': 'PTGS2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Metabolism of water-soluble vitamins and cofactors',\n",
       "  'react_pathway_id': 'R-HSA-196849',\n",
       "  'inference_gene_symbol': 'PTGS2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Metabolism of vitamins and cofactors',\n",
       "  'react_pathway_id': 'R-HSA-196854',\n",
       "  'inference_gene_symbol': 'PTGS2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Nicotinamide salvaging',\n",
       "  'react_pathway_id': 'R-HSA-197264',\n",
       "  'inference_gene_symbol': 'PTGS2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by NOTCH1',\n",
       "  'react_pathway_id': 'R-HSA-1980143',\n",
       "  'inference_gene_symbol': ['CREBBP',\n",
       "   'EP300',\n",
       "   'HDAC1',\n",
       "   'HDAC2',\n",
       "   'MYC',\n",
       "   'NOTCH1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by NOTCH2',\n",
       "  'react_pathway_id': 'R-HSA-1980145',\n",
       "  'inference_gene_symbol': ['EP300', 'NOTCH2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by NOTCH3',\n",
       "  'react_pathway_id': 'R-HSA-1980148',\n",
       "  'inference_gene_symbol': 'NOTCH3'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by NOTCH4',\n",
       "  'react_pathway_id': 'R-HSA-1980150',\n",
       "  'inference_gene_symbol': 'NOTCH4'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'PI3K/AKT activation',\n",
       "  'react_pathway_id': 'R-HSA-198203',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTPN11',\n",
       "   'TP53',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'AKT phosphorylates targets in the cytosol',\n",
       "  'react_pathway_id': 'R-HSA-198323',\n",
       "  'inference_gene_symbol': ['AKT1', 'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'AKT phosphorylates targets in the nucleus',\n",
       "  'react_pathway_id': 'R-HSA-198693',\n",
       "  'inference_gene_symbol': 'AKT1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Nuclear Events (kinase and transcription factor activation)',\n",
       "  'react_pathway_id': 'R-HSA-198725',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signalling to STAT3',\n",
       "  'react_pathway_id': 'R-HSA-198745',\n",
       "  'inference_gene_symbol': 'STAT3'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'ERK/MAPK targets',\n",
       "  'react_pathway_id': 'R-HSA-198753',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Immunoregulatory interactions between a Lymphoid and a non-Lymphoid cell',\n",
       "  'react_pathway_id': 'R-HSA-198933',\n",
       "  'inference_gene_symbol': ['CDH1', 'ITGB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'PPARA activates gene expression',\n",
       "  'react_pathway_id': 'R-HSA-1989781',\n",
       "  'inference_gene_symbol': ['CREBBP', 'EP300', 'NCOA3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Negative regulation of the PI3K/AKT network',\n",
       "  'react_pathway_id': 'R-HSA-199418',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'ER to Golgi Anterograde Transport',\n",
       "  'react_pathway_id': 'R-HSA-199977',\n",
       "  'inference_gene_symbol': 'AREG'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Membrane Trafficking',\n",
       "  'react_pathway_id': 'R-HSA-199991',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'AREG',\n",
       "   'CTTN',\n",
       "   'EGFR',\n",
       "   'HSPA8',\n",
       "   'SFN',\n",
       "   'TSC1',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'trans-Golgi Network Vesicle Budding',\n",
       "  'react_pathway_id': 'R-HSA-199992',\n",
       "  'inference_gene_symbol': 'HSPA8'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TCF dependent signaling in response to WNT',\n",
       "  'react_pathway_id': 'R-HSA-201681',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CREBBP',\n",
       "   'CSNK2A1',\n",
       "   'CSNK2A2',\n",
       "   'CSNK2B',\n",
       "   'EP300',\n",
       "   'HDAC1',\n",
       "   'LGR6',\n",
       "   'MYC',\n",
       "   'SEM1',\n",
       "   'SOX2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'WNT mediated activation of DVL',\n",
       "  'react_pathway_id': 'R-HSA-201688',\n",
       "  'inference_gene_symbol': ['CSNK2A1', 'CSNK2A2', 'CSNK2B']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Formation of the beta-catenin:TCF transactivating complex',\n",
       "  'react_pathway_id': 'R-HSA-201722',\n",
       "  'inference_gene_symbol': ['CREBBP', 'EP300', 'HDAC1', 'MYC']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Metabolism of nitric oxide',\n",
       "  'react_pathway_id': 'R-HSA-202131',\n",
       "  'inference_gene_symbol': 'AKT1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Assembly of collagen fibrils and other multimeric structures',\n",
       "  'react_pathway_id': 'R-HSA-2022090',\n",
       "  'inference_gene_symbol': 'MMP9'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TCR signaling',\n",
       "  'react_pathway_id': 'R-HSA-202403',\n",
       "  'inference_gene_symbol': ['PIK3CA', 'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Downstream TCR signaling',\n",
       "  'react_pathway_id': 'R-HSA-202424',\n",
       "  'inference_gene_symbol': ['PIK3CA', 'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'ERKs are inactivated',\n",
       "  'react_pathway_id': 'R-HSA-202670',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cell surface interactions at the vascular wall',\n",
       "  'react_pathway_id': 'R-HSA-202733',\n",
       "  'inference_gene_symbol': ['HRAS',\n",
       "   'ITGA5',\n",
       "   'ITGB1',\n",
       "   'KRAS',\n",
       "   'PIK3CA',\n",
       "   'PPIA',\n",
       "   'PTPN11',\n",
       "   'TGFB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Fcgamma receptor (FCGR) dependent phagocytosis',\n",
       "  'react_pathway_id': 'R-HSA-2029480',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3', 'PIK3CA', 'PTK2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of actin dynamics for phagocytic cup formation',\n",
       "  'react_pathway_id': 'R-HSA-2029482',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3', 'PTK2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Role of phospholipids in phagocytosis',\n",
       "  'react_pathway_id': 'R-HSA-2029485',\n",
       "  'inference_gene_symbol': 'PIK3CA'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'YAP1- and WWTR1 (TAZ)-stimulated gene expression',\n",
       "  'react_pathway_id': 'R-HSA-2032785',\n",
       "  'inference_gene_symbol': 'CREBBP'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'eNOS activation',\n",
       "  'react_pathway_id': 'R-HSA-203615',\n",
       "  'inference_gene_symbol': 'AKT1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'eNOS activation and regulation',\n",
       "  'react_pathway_id': 'R-HSA-203765',\n",
       "  'inference_gene_symbol': 'AKT1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'COPII (Coat Protein 2) Mediated Vesicle Transport',\n",
       "  'react_pathway_id': 'R-HSA-204005',\n",
       "  'inference_gene_symbol': 'AREG'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cell death signalling via NRAGE, NRIF and NADE',\n",
       "  'react_pathway_id': 'R-HSA-204998',\n",
       "  'inference_gene_symbol': 'TIAM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of gene expression in late stage (branching morphogenesis) pancreatic bud precursor cells',\n",
       "  'react_pathway_id': 'R-HSA-210744',\n",
       "  'inference_gene_symbol': 'NOTCH1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of gene expression in beta cells',\n",
       "  'react_pathway_id': 'R-HSA-210745',\n",
       "  'inference_gene_symbol': 'AKT1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'PECAM1 interactions',\n",
       "  'react_pathway_id': 'R-HSA-210990',\n",
       "  'inference_gene_symbol': 'PTPN11'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Basigin interactions',\n",
       "  'react_pathway_id': 'R-HSA-210991',\n",
       "  'inference_gene_symbol': ['ITGB1', 'PPIA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Tie2 Signaling',\n",
       "  'react_pathway_id': 'R-HSA-210993',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS', 'PIK3CA', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'AKT-mediated inactivation of FOXO1A',\n",
       "  'react_pathway_id': 'R-HSA-211163',\n",
       "  'inference_gene_symbol': 'AKT1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of activated PAK-2p34 by proteasome mediated degradation',\n",
       "  'react_pathway_id': 'R-HSA-211733',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Biological oxidations',\n",
       "  'react_pathway_id': 'R-HSA-211859',\n",
       "  'inference_gene_symbol': ['CYP1A2', 'CYP1B1', 'GSTM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cytochrome P450 - arranged by substrate type',\n",
       "  'react_pathway_id': 'R-HSA-211897',\n",
       "  'inference_gene_symbol': ['CYP1A2', 'CYP1B1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Phase 1 - Functionalization of compounds',\n",
       "  'react_pathway_id': 'R-HSA-211945',\n",
       "  'inference_gene_symbol': ['CYP1A2', 'CYP1B1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Aromatic amines can be N-hydroxylated or N-dealkylated by CYP1A2',\n",
       "  'react_pathway_id': 'R-HSA-211957',\n",
       "  'inference_gene_symbol': 'CYP1A2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Endogenous sterols',\n",
       "  'react_pathway_id': 'R-HSA-211976',\n",
       "  'inference_gene_symbol': 'CYP1B1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Xenobiotics',\n",
       "  'react_pathway_id': 'R-HSA-211981',\n",
       "  'inference_gene_symbol': 'CYP1A2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Epigenetic regulation of gene expression',\n",
       "  'react_pathway_id': 'R-HSA-212165',\n",
       "  'inference_gene_symbol': ['EP300', 'ERCC2', 'HDAC1', 'HDAC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'NOTCH1 Intracellular Domain Regulates Transcription',\n",
       "  'react_pathway_id': 'R-HSA-2122947',\n",
       "  'inference_gene_symbol': ['CREBBP',\n",
       "   'EP300',\n",
       "   'HDAC1',\n",
       "   'HDAC2',\n",
       "   'MYC',\n",
       "   'NOTCH1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Activated NOTCH1 Transmits Signal to the Nucleus',\n",
       "  'react_pathway_id': 'R-HSA-2122948',\n",
       "  'inference_gene_symbol': 'NOTCH1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Generic Transcription Pathway',\n",
       "  'react_pathway_id': 'R-HSA-212436',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'ATM',\n",
       "   'BIRC5',\n",
       "   'CDKN2A',\n",
       "   'CREBBP',\n",
       "   'CSNK2A1',\n",
       "   'CSNK2A2',\n",
       "   'CSNK2B',\n",
       "   'EGFR',\n",
       "   'EP300',\n",
       "   'ERCC2',\n",
       "   'HDAC1',\n",
       "   'HDAC2',\n",
       "   'MYC',\n",
       "   'NOTCH1',\n",
       "   'NOTCH2',\n",
       "   'NOTCH3',\n",
       "   'NOTCH4',\n",
       "   'RARB',\n",
       "   'SFN',\n",
       "   'STK11',\n",
       "   'TP53',\n",
       "   'TP63',\n",
       "   'TSC1',\n",
       "   'TSC2',\n",
       "   'VEGFA',\n",
       "   'WWOX']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'EGFR interacts with phospholipase C-gamma',\n",
       "  'react_pathway_id': 'R-HSA-212718',\n",
       "  'inference_gene_symbol': 'EGFR'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Molecules associated with elastic fibres',\n",
       "  'react_pathway_id': 'R-HSA-2129379',\n",
       "  'inference_gene_symbol': ['ITGB1', 'TGFB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Synthesis of epoxy (EET) and dihydroxyeicosatrienoic acids (DHET)',\n",
       "  'react_pathway_id': 'R-HSA-2142670',\n",
       "  'inference_gene_symbol': ['CYP1A2', 'CYP1B1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Synthesis of 5-eicosatetraenoic acids',\n",
       "  'react_pathway_id': 'R-HSA-2142688',\n",
       "  'inference_gene_symbol': 'GPX1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Synthesis of 12-eicosatetraenoic acid derivatives',\n",
       "  'react_pathway_id': 'R-HSA-2142712',\n",
       "  'inference_gene_symbol': 'GPX1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Arachidonic acid metabolism',\n",
       "  'react_pathway_id': 'R-HSA-2142753',\n",
       "  'inference_gene_symbol': ['CYP1A2', 'CYP1B1', 'GPX1', 'PTGS2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Synthesis of 15-eicosatetraenoic acid derivatives',\n",
       "  'react_pathway_id': 'R-HSA-2142770',\n",
       "  'inference_gene_symbol': ['GPX1', 'PTGS2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Synthesis of (16-20)-hydroxyeicosatetraenoic acids (HETE)',\n",
       "  'react_pathway_id': 'R-HSA-2142816',\n",
       "  'inference_gene_symbol': ['CYP1A2', 'CYP1B1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Transcriptional activation of mitochondrial biogenesis',\n",
       "  'react_pathway_id': 'R-HSA-2151201',\n",
       "  'inference_gene_symbol': ['CREBBP', 'SIRT3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Integrin cell surface interactions',\n",
       "  'react_pathway_id': 'R-HSA-216083',\n",
       "  'inference_gene_symbol': ['CDH1', 'ITGA5', 'ITGB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Synthesis of Prostaglandins (PG) and Thromboxanes (TX)',\n",
       "  'react_pathway_id': 'R-HSA-2162123',\n",
       "  'inference_gene_symbol': 'PTGS2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'DAP12 interactions',\n",
       "  'react_pathway_id': 'R-HSA-2172127',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'TP53',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Downregulation of TGF-beta receptor signaling',\n",
       "  'react_pathway_id': 'R-HSA-2173788',\n",
       "  'inference_gene_symbol': 'TGFB1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TGF-beta receptor signaling activates SMADs',\n",
       "  'react_pathway_id': 'R-HSA-2173789',\n",
       "  'inference_gene_symbol': 'TGFB1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TGF-beta receptor signaling in EMT (epithelial to mesenchymal transition)',\n",
       "  'react_pathway_id': 'R-HSA-2173791',\n",
       "  'inference_gene_symbol': 'TGFB1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Transcriptional activity of SMAD2/SMAD3:SMAD4 heterotrimer',\n",
       "  'react_pathway_id': 'R-HSA-2173793',\n",
       "  'inference_gene_symbol': ['HDAC1', 'MYC']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Downregulation of SMAD2/3:SMAD4 transcriptional activity',\n",
       "  'react_pathway_id': 'R-HSA-2173795',\n",
       "  'inference_gene_symbol': 'HDAC1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'SMAD2/SMAD3:SMAD4 heterotrimer regulates transcription',\n",
       "  'react_pathway_id': 'R-HSA-2173796',\n",
       "  'inference_gene_symbol': ['HDAC1', 'MYC']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'EGFR Transactivation by Gastrin',\n",
       "  'react_pathway_id': 'R-HSA-2179392',\n",
       "  'inference_gene_symbol': ['EGFR', 'HRAS', 'KRAS']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'NOTCH2 intracellular domain regulates transcription',\n",
       "  'react_pathway_id': 'R-HSA-2197563',\n",
       "  'inference_gene_symbol': ['EP300', 'NOTCH2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'PI3K/AKT Signaling in Cancer',\n",
       "  'react_pathway_id': 'R-HSA-2219528',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTPN11',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Constitutive Signaling by Aberrant PI3K in Cancer',\n",
       "  'react_pathway_id': 'R-HSA-2219530',\n",
       "  'inference_gene_symbol': ['EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cellular response to hypoxia',\n",
       "  'react_pathway_id': 'R-HSA-2262749',\n",
       "  'inference_gene_symbol': ['CREBBP', 'EP300', 'SEM1', 'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cellular responses to stress',\n",
       "  'react_pathway_id': 'R-HSA-2262752',\n",
       "  'inference_gene_symbol': ['ATM',\n",
       "   'CDKN2A',\n",
       "   'CREBBP',\n",
       "   'CXCL8',\n",
       "   'EP300',\n",
       "   'GPX1',\n",
       "   'HSPA8',\n",
       "   'IL6',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'SEM1',\n",
       "   'SOD2',\n",
       "   'STAT3',\n",
       "   'TP53',\n",
       "   'TSC1',\n",
       "   'TSC2',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by Type 1 Insulin-like Growth Factor 1 Receptor (IGF1R)',\n",
       "  'react_pathway_id': 'R-HSA-2404192',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'STK11',\n",
       "   'TSC1',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'DAP12 signaling',\n",
       "  'react_pathway_id': 'R-HSA-2424491',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'TP53',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Activation of gene expression by SREBF (SREBP)',\n",
       "  'react_pathway_id': 'R-HSA-2426168',\n",
       "  'inference_gene_symbol': 'CREBBP'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'IGF1R signaling cascade',\n",
       "  'react_pathway_id': 'R-HSA-2428924',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'STK11',\n",
       "   'TSC1',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'IRS-related events triggered by IGF1R',\n",
       "  'react_pathway_id': 'R-HSA-2428928',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'STK11',\n",
       "   'TSC1',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'SHC-related events triggered by IGF1R',\n",
       "  'react_pathway_id': 'R-HSA-2428933',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Fc epsilon receptor (FCERI) signaling',\n",
       "  'react_pathway_id': 'R-HSA-2454202',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'TP53',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Separation of Sister Chromatids',\n",
       "  'react_pathway_id': 'R-HSA-2467813',\n",
       "  'inference_gene_symbol': ['BIRC5', 'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Resolution of Sister Chromatid Cohesion',\n",
       "  'react_pathway_id': 'R-HSA-2500257',\n",
       "  'inference_gene_symbol': 'BIRC5'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Condensation of Prometaphase Chromosomes',\n",
       "  'react_pathway_id': 'R-HSA-2514853',\n",
       "  'inference_gene_symbol': ['CSNK2A1', 'CSNK2A2', 'CSNK2B']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Mitotic Metaphase and Anaphase',\n",
       "  'react_pathway_id': 'R-HSA-2555396',\n",
       "  'inference_gene_symbol': ['BIRC5', 'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Oxidative Stress Induced Senescence',\n",
       "  'react_pathway_id': 'R-HSA-2559580',\n",
       "  'inference_gene_symbol': ['CDKN2A', 'MAPK1', 'MAPK3', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Senescence-Associated Secretory Phenotype (SASP)',\n",
       "  'react_pathway_id': 'R-HSA-2559582',\n",
       "  'inference_gene_symbol': ['CDKN2A',\n",
       "   'CXCL8',\n",
       "   'IL6',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'STAT3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cellular Senescence',\n",
       "  'react_pathway_id': 'R-HSA-2559583',\n",
       "  'inference_gene_symbol': ['ATM',\n",
       "   'CDKN2A',\n",
       "   'CXCL8',\n",
       "   'IL6',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'STAT3',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Formation of Senescence-Associated Heterochromatin Foci (SAHF)',\n",
       "  'react_pathway_id': 'R-HSA-2559584',\n",
       "  'inference_gene_symbol': 'TP53'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Oncogene Induced Senescence',\n",
       "  'react_pathway_id': 'R-HSA-2559585',\n",
       "  'inference_gene_symbol': ['CDKN2A', 'MAPK1', 'MAPK3', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'DNA Damage/Telomere Stress Induced Senescence',\n",
       "  'react_pathway_id': 'R-HSA-2559586',\n",
       "  'inference_gene_symbol': ['ATM', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cytosolic iron-sulfur cluster assembly',\n",
       "  'react_pathway_id': 'R-HSA-2564830',\n",
       "  'inference_gene_symbol': 'ERCC2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by Leptin',\n",
       "  'react_pathway_id': 'R-HSA-2586552',\n",
       "  'inference_gene_symbol': ['EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'STAT3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by NOTCH1 PEST Domain Mutants in Cancer',\n",
       "  'react_pathway_id': 'R-HSA-2644602',\n",
       "  'inference_gene_symbol': ['CREBBP',\n",
       "   'EP300',\n",
       "   'HDAC1',\n",
       "   'HDAC2',\n",
       "   'MYC',\n",
       "   'NOTCH1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by NOTCH1 in Cancer',\n",
       "  'react_pathway_id': 'R-HSA-2644603',\n",
       "  'inference_gene_symbol': ['CREBBP',\n",
       "   'EP300',\n",
       "   'HDAC1',\n",
       "   'HDAC2',\n",
       "   'MYC',\n",
       "   'NOTCH1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'FBXW7 Mutants and NOTCH1 in Cancer',\n",
       "  'react_pathway_id': 'R-HSA-2644605',\n",
       "  'inference_gene_symbol': 'NOTCH1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Constitutive Signaling by NOTCH1 PEST Domain Mutants',\n",
       "  'react_pathway_id': 'R-HSA-2644606',\n",
       "  'inference_gene_symbol': ['CREBBP',\n",
       "   'EP300',\n",
       "   'HDAC1',\n",
       "   'HDAC2',\n",
       "   'MYC',\n",
       "   'NOTCH1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Loss of Function of FBXW7 in Cancer and NOTCH1 Signaling',\n",
       "  'react_pathway_id': 'R-HSA-2644607',\n",
       "  'inference_gene_symbol': 'NOTCH1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by NOTCH1 t(7;9)(NOTCH1:M1580_K2555) Translocation Mutant',\n",
       "  'react_pathway_id': 'R-HSA-2660825',\n",
       "  'inference_gene_symbol': 'NOTCH1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Constitutive Signaling by NOTCH1 t(7;9)(NOTCH1:M1580_K2555) Translocation Mutant',\n",
       "  'react_pathway_id': 'R-HSA-2660826',\n",
       "  'inference_gene_symbol': 'NOTCH1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'EPH-Ephrin signaling',\n",
       "  'react_pathway_id': 'R-HSA-2682334',\n",
       "  'inference_gene_symbol': ['HRAS',\n",
       "   'MMP9',\n",
       "   'PTK2',\n",
       "   'RASA1',\n",
       "   'TIAM1',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by NOTCH1 HD Domain Mutants in Cancer',\n",
       "  'react_pathway_id': 'R-HSA-2691230',\n",
       "  'inference_gene_symbol': 'NOTCH1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Constitutive Signaling by NOTCH1 HD Domain Mutants',\n",
       "  'react_pathway_id': 'R-HSA-2691232',\n",
       "  'inference_gene_symbol': 'NOTCH1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Role of LAT2/NTAL/LAB on calcium mobilization',\n",
       "  'react_pathway_id': 'R-HSA-2730905',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTPN11',\n",
       "   'TP53',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'FCERI mediated MAPK activation',\n",
       "  'react_pathway_id': 'R-HSA-2871796',\n",
       "  'inference_gene_symbol': ['EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PTK2',\n",
       "   'RASA1',\n",
       "   'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'FCERI mediated NF-kB activation',\n",
       "  'react_pathway_id': 'R-HSA-2871837',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'POU5F1 (OCT4), SOX2, NANOG repress genes related to differentiation',\n",
       "  'react_pathway_id': 'R-HSA-2892245',\n",
       "  'inference_gene_symbol': 'SOX2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'POU5F1 (OCT4), SOX2, NANOG activate genes related to proliferation',\n",
       "  'react_pathway_id': 'R-HSA-2892247',\n",
       "  'inference_gene_symbol': ['SOX2', 'STAT3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by NOTCH1 HD+PEST Domain Mutants in Cancer',\n",
       "  'react_pathway_id': 'R-HSA-2894858',\n",
       "  'inference_gene_symbol': ['CREBBP',\n",
       "   'EP300',\n",
       "   'HDAC1',\n",
       "   'HDAC2',\n",
       "   'MYC',\n",
       "   'NOTCH1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Constitutive Signaling by NOTCH1 HD+PEST Domain Mutants',\n",
       "  'react_pathway_id': 'R-HSA-2894862',\n",
       "  'inference_gene_symbol': ['CREBBP',\n",
       "   'EP300',\n",
       "   'HDAC1',\n",
       "   'HDAC2',\n",
       "   'MYC',\n",
       "   'NOTCH1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'NOTCH2 Activation and Transmission of Signal to the Nucleus',\n",
       "  'react_pathway_id': 'R-HSA-2979096',\n",
       "  'inference_gene_symbol': 'NOTCH2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'SUMOylation',\n",
       "  'react_pathway_id': 'R-HSA-2990846',\n",
       "  'inference_gene_symbol': ['BIRC5', 'CDKN2A', 'RAD52', 'TP53', 'XPC']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Laminin interactions',\n",
       "  'react_pathway_id': 'R-HSA-3000157',\n",
       "  'inference_gene_symbol': 'ITGB1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Syndecan interactions',\n",
       "  'react_pathway_id': 'R-HSA-3000170',\n",
       "  'inference_gene_symbol': ['ITGB1', 'TGFB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Non-integrin membrane-ECM interactions',\n",
       "  'react_pathway_id': 'R-HSA-3000171',\n",
       "  'inference_gene_symbol': ['ITGB1', 'TGFB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'ECM proteoglycans',\n",
       "  'react_pathway_id': 'R-HSA-3000178',\n",
       "  'inference_gene_symbol': ['DSPP', 'ITGB1', 'TGFB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'SUMOylation of DNA damage response and repair proteins',\n",
       "  'react_pathway_id': 'R-HSA-3108214',\n",
       "  'inference_gene_symbol': ['CDKN2A', 'RAD52', 'XPC']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'SUMO E3 ligases SUMOylate target proteins',\n",
       "  'react_pathway_id': 'R-HSA-3108232',\n",
       "  'inference_gene_symbol': ['BIRC5', 'CDKN2A', 'RAD52', 'TP53', 'XPC']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'LRR FLII-interacting protein 1 (LRRFIP1) activates type I IFN production',\n",
       "  'react_pathway_id': 'R-HSA-3134973',\n",
       "  'inference_gene_symbol': ['CREBBP', 'EP300']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'HDACs deacetylate histones',\n",
       "  'react_pathway_id': 'R-HSA-3214815',\n",
       "  'inference_gene_symbol': ['HDAC1', 'HDAC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'HATs acetylate histones',\n",
       "  'react_pathway_id': 'R-HSA-3214847',\n",
       "  'inference_gene_symbol': ['CREBBP', 'EP300']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RMTs methylate histone arginines',\n",
       "  'react_pathway_id': 'R-HSA-3214858',\n",
       "  'inference_gene_symbol': 'CCND1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'SUMOylation of transcription factors',\n",
       "  'react_pathway_id': 'R-HSA-3232118',\n",
       "  'inference_gene_symbol': ['CDKN2A', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Chromatin modifying enzymes',\n",
       "  'react_pathway_id': 'R-HSA-3247509',\n",
       "  'inference_gene_symbol': ['CCND1', 'CREBBP', 'EP300', 'HDAC1', 'HDAC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'IRF3-mediated induction of type I IFN',\n",
       "  'react_pathway_id': 'R-HSA-3270619',\n",
       "  'inference_gene_symbol': 'PRKDC'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Hydroxycarboxylic acid-binding receptors',\n",
       "  'react_pathway_id': 'R-HSA-3296197',\n",
       "  'inference_gene_symbol': ['HCAR2', 'HCAR3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Detoxification of Reactive Oxygen Species',\n",
       "  'react_pathway_id': 'R-HSA-3299685',\n",
       "  'inference_gene_symbol': ['GPX1', 'SOD2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Loss of Function of SMAD2/3 in Cancer',\n",
       "  'react_pathway_id': 'R-HSA-3304349',\n",
       "  'inference_gene_symbol': 'TGFB1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by TGF-beta Receptor Complex in Cancer',\n",
       "  'react_pathway_id': 'R-HSA-3304351',\n",
       "  'inference_gene_symbol': 'TGFB1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'SMAD2/3 Phosphorylation Motif Mutants in Cancer',\n",
       "  'react_pathway_id': 'R-HSA-3304356',\n",
       "  'inference_gene_symbol': 'TGFB1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'SMAD2/3 MH2 Domain Mutants in Cancer',\n",
       "  'react_pathway_id': 'R-HSA-3315487',\n",
       "  'inference_gene_symbol': 'TGFB1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of HSF1-mediated heat shock response',\n",
       "  'react_pathway_id': 'R-HSA-3371453',\n",
       "  'inference_gene_symbol': ['ATM', 'HSPA8', 'MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cellular response to heat stress',\n",
       "  'react_pathway_id': 'R-HSA-3371556',\n",
       "  'inference_gene_symbol': ['ATM',\n",
       "   'CREBBP',\n",
       "   'EP300',\n",
       "   'HSPA8',\n",
       "   'MAPK1',\n",
       "   'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Attenuation phase',\n",
       "  'react_pathway_id': 'R-HSA-3371568',\n",
       "  'inference_gene_symbol': ['CREBBP', 'EP300', 'HSPA8']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'HSF1-dependent transactivation',\n",
       "  'react_pathway_id': 'R-HSA-3371571',\n",
       "  'inference_gene_symbol': ['CREBBP', 'EP300', 'HSPA8']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Autodegradation of the E3 ubiquitin ligase COP1',\n",
       "  'react_pathway_id': 'R-HSA-349425',\n",
       "  'inference_gene_symbol': ['ATM', 'SEM1', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Notch-HLH transcription pathway',\n",
       "  'react_pathway_id': 'R-HSA-350054',\n",
       "  'inference_gene_symbol': ['CREBBP', 'NOTCH1', 'NOTCH2', 'NOTCH3', 'NOTCH4']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of ornithine decarboxylase (ODC)',\n",
       "  'react_pathway_id': 'R-HSA-350562',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Metabolism of polyamines',\n",
       "  'react_pathway_id': 'R-HSA-351202',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Apoptotic cleavage of cell adhesion  proteins',\n",
       "  'react_pathway_id': 'R-HSA-351906',\n",
       "  'inference_gene_symbol': 'CDH1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Integrin alphaIIb beta3 signaling',\n",
       "  'react_pathway_id': 'R-HSA-354192',\n",
       "  'inference_gene_symbol': ['AKT1', 'PTK2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'GRB2:SOS provides linkage to MAPK signaling for Integrins',\n",
       "  'react_pathway_id': 'R-HSA-354194',\n",
       "  'inference_gene_symbol': 'PTK2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Loss of Function of TGFBR2 in Cancer',\n",
       "  'react_pathway_id': 'R-HSA-3642278',\n",
       "  'inference_gene_symbol': 'TGFB1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TGFBR2 MSI Frameshift Mutants in Cancer',\n",
       "  'react_pathway_id': 'R-HSA-3642279',\n",
       "  'inference_gene_symbol': 'TGFB1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TGFBR2 Kinase Domain Mutants in Cancer',\n",
       "  'react_pathway_id': 'R-HSA-3645790',\n",
       "  'inference_gene_symbol': 'TGFB1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TGFBR1 KD Mutants in Cancer',\n",
       "  'react_pathway_id': 'R-HSA-3656532',\n",
       "  'inference_gene_symbol': 'TGFB1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Loss of Function of TGFBR1 in Cancer',\n",
       "  'react_pathway_id': 'R-HSA-3656534',\n",
       "  'inference_gene_symbol': 'TGFB1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TGFBR1 LBD Mutants in Cancer',\n",
       "  'react_pathway_id': 'R-HSA-3656535',\n",
       "  'inference_gene_symbol': 'TGFB1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Transcriptional Regulation by TP53',\n",
       "  'react_pathway_id': 'R-HSA-3700989',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'ATM',\n",
       "   'BIRC5',\n",
       "   'CDKN2A',\n",
       "   'CREBBP',\n",
       "   'CSNK2A1',\n",
       "   'CSNK2A2',\n",
       "   'CSNK2B',\n",
       "   'EP300',\n",
       "   'ERCC2',\n",
       "   'HDAC1',\n",
       "   'HDAC2',\n",
       "   'SFN',\n",
       "   'STK11',\n",
       "   'TP53',\n",
       "   'TP63',\n",
       "   'TSC1',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'p130Cas linkage to MAPK signaling for integrins',\n",
       "  'react_pathway_id': 'R-HSA-372708',\n",
       "  'inference_gene_symbol': 'PTK2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by GPCR',\n",
       "  'react_pathway_id': 'R-HSA-372790',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'ANXA1',\n",
       "   'CXCL8',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HCAR2',\n",
       "   'HCAR3',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTGER1',\n",
       "   'PTK2',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'TIAM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Class A/1 (Rhodopsin-like receptors)',\n",
       "  'react_pathway_id': 'R-HSA-373076',\n",
       "  'inference_gene_symbol': ['ANXA1', 'CXCL8', 'HCAR2', 'HCAR3', 'PTGER1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Netrin-1 signaling',\n",
       "  'react_pathway_id': 'R-HSA-373752',\n",
       "  'inference_gene_symbol': ['PTK2', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Nephrin interactions',\n",
       "  'react_pathway_id': 'R-HSA-373753',\n",
       "  'inference_gene_symbol': 'PIK3CA'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Semaphorin interactions',\n",
       "  'react_pathway_id': 'R-HSA-373755',\n",
       "  'inference_gene_symbol': ['ITGB1', 'MET']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'L1CAM interactions',\n",
       "  'react_pathway_id': 'R-HSA-373760',\n",
       "  'inference_gene_symbol': ['CSNK2A1',\n",
       "   'CSNK2A2',\n",
       "   'CSNK2B',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HSPA8',\n",
       "   'ITGA5',\n",
       "   'ITGB1',\n",
       "   'MAPK1',\n",
       "   'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'NCAM signaling for neurite out-growth',\n",
       "  'react_pathway_id': 'R-HSA-375165',\n",
       "  'inference_gene_symbol': ['EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PTK2',\n",
       "   'RASA1',\n",
       "   'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Peptide ligand-binding receptors',\n",
       "  'react_pathway_id': 'R-HSA-375276',\n",
       "  'inference_gene_symbol': ['ANXA1', 'CXCL8']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Deactivation of the beta-catenin transactivating complex',\n",
       "  'react_pathway_id': 'R-HSA-3769402',\n",
       "  'inference_gene_symbol': ['AKT1', 'HDAC1', 'SOX2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Diseases of glycosylation',\n",
       "  'react_pathway_id': 'R-HSA-3781865',\n",
       "  'inference_gene_symbol': ['NOTCH1', 'NOTCH2', 'NOTCH3', 'NOTCH4']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Chemokine receptors bind chemokines',\n",
       "  'react_pathway_id': 'R-HSA-380108',\n",
       "  'inference_gene_symbol': 'CXCL8'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Energy dependent regulation of mTOR by LKB1-AMPK',\n",
       "  'react_pathway_id': 'R-HSA-380972',\n",
       "  'inference_gene_symbol': ['STK11', 'TSC1', 'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'ATF4 activates genes',\n",
       "  'react_pathway_id': 'R-HSA-380994',\n",
       "  'inference_gene_symbol': ['CCL2', 'CXCL8']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'PERK regulates gene expression',\n",
       "  'react_pathway_id': 'R-HSA-381042',\n",
       "  'inference_gene_symbol': ['CCL2', 'CXCL8']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Unfolded Protein Response (UPR)',\n",
       "  'react_pathway_id': 'R-HSA-381119',\n",
       "  'inference_gene_symbol': ['CCL2', 'CXCL8']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Transcriptional regulation of white adipocyte differentiation',\n",
       "  'react_pathway_id': 'R-HSA-381340',\n",
       "  'inference_gene_symbol': ['CEBPA',\n",
       "   'CREBBP',\n",
       "   'EP300',\n",
       "   'KLF5',\n",
       "   'NCOA3',\n",
       "   'TGFB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Transmembrane transport of small molecules',\n",
       "  'react_pathway_id': 'R-HSA-382551',\n",
       "  'inference_gene_symbol': ['AQP3', 'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'ABC-family proteins mediated transport',\n",
       "  'react_pathway_id': 'R-HSA-382556',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Nuclear Receptor transcription pathway',\n",
       "  'react_pathway_id': 'R-HSA-383280',\n",
       "  'inference_gene_symbol': 'RARB'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Beta-catenin independent WNT signaling',\n",
       "  'react_pathway_id': 'R-HSA-3858494',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'GPCR downstream signaling',\n",
       "  'react_pathway_id': 'R-HSA-388396',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'ANXA1',\n",
       "   'CXCL8',\n",
       "   'HCAR2',\n",
       "   'HCAR3',\n",
       "   'PIK3CA',\n",
       "   'PTGER1',\n",
       "   'TIAM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Costimulation by the CD28 family',\n",
       "  'react_pathway_id': 'R-HSA-388841',\n",
       "  'inference_gene_symbol': ['AKT1', 'PIK3CA', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'CD28 co-stimulation',\n",
       "  'react_pathway_id': 'R-HSA-389356',\n",
       "  'inference_gene_symbol': ['AKT1', 'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'CD28 dependent PI3K/Akt signaling',\n",
       "  'react_pathway_id': 'R-HSA-389357',\n",
       "  'inference_gene_symbol': ['AKT1', 'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'CTLA4 inhibitory signaling',\n",
       "  'react_pathway_id': 'R-HSA-389513',\n",
       "  'inference_gene_symbol': ['AKT1', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'PD-1 signaling',\n",
       "  'react_pathway_id': 'R-HSA-389948',\n",
       "  'inference_gene_symbol': 'PTPN11'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Chaperonin-mediated protein folding',\n",
       "  'react_pathway_id': 'R-HSA-390466',\n",
       "  'inference_gene_symbol': ['CSNK2A1', 'CSNK2A2', 'CSNK2B', 'STAT3', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Association of TriC/CCT with target proteins during biosynthesis',\n",
       "  'react_pathway_id': 'R-HSA-390471',\n",
       "  'inference_gene_symbol': ['STAT3', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Diseases associated with O-glycosylation of proteins',\n",
       "  'react_pathway_id': 'R-HSA-3906995',\n",
       "  'inference_gene_symbol': ['NOTCH1', 'NOTCH2', 'NOTCH3', 'NOTCH4']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signal regulatory protein (SIRP) family interactions',\n",
       "  'react_pathway_id': 'R-HSA-391160',\n",
       "  'inference_gene_symbol': ['PTK2', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Protein folding',\n",
       "  'react_pathway_id': 'R-HSA-391251',\n",
       "  'inference_gene_symbol': ['CSNK2A1', 'CSNK2A2', 'CSNK2B', 'STAT3', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Eicosanoid ligand-binding receptors',\n",
       "  'react_pathway_id': 'R-HSA-391903',\n",
       "  'inference_gene_symbol': 'PTGER1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Prostanoid ligand receptors',\n",
       "  'react_pathway_id': 'R-HSA-391908',\n",
       "  'inference_gene_symbol': 'PTGER1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'G beta:gamma signalling through PI3Kgamma',\n",
       "  'react_pathway_id': 'R-HSA-392451',\n",
       "  'inference_gene_symbol': ['AKT1', 'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Metabolism of proteins',\n",
       "  'react_pathway_id': 'R-HSA-392499',\n",
       "  'inference_gene_symbol': ['AREG',\n",
       "   'BIRC5',\n",
       "   'CCL2',\n",
       "   'CDKN2A',\n",
       "   'CSNK2A1',\n",
       "   'CSNK2A2',\n",
       "   'CSNK2B',\n",
       "   'CXCL8',\n",
       "   'EP300',\n",
       "   'HSPA8',\n",
       "   'MYC',\n",
       "   'PRKDC',\n",
       "   'RAD52',\n",
       "   'SEM1',\n",
       "   'STAT3',\n",
       "   'TGFB1',\n",
       "   'TP53',\n",
       "   'UBA1',\n",
       "   'VDAC2',\n",
       "   'XPC']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'EPHB-mediated forward signaling',\n",
       "  'react_pathway_id': 'R-HSA-3928662',\n",
       "  'inference_gene_symbol': ['HRAS', 'PTK2', 'RASA1', 'TIAM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'EPHA-mediated growth cone collapse',\n",
       "  'react_pathway_id': 'R-HSA-3928663',\n",
       "  'inference_gene_symbol': ['PTK2', 'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'EPH-ephrin mediated repulsion of cells',\n",
       "  'react_pathway_id': 'R-HSA-3928665',\n",
       "  'inference_gene_symbol': ['MMP9', 'TIAM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Muscle contraction',\n",
       "  'react_pathway_id': 'R-HSA-397014',\n",
       "  'inference_gene_symbol': ['ANXA1', 'ANXA2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'G-protein beta:gamma signalling',\n",
       "  'react_pathway_id': 'R-HSA-397795',\n",
       "  'inference_gene_symbol': ['AKT1', 'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of lipid metabolism by Peroxisome proliferator-activated receptor alpha (PPARalpha)',\n",
       "  'react_pathway_id': 'R-HSA-400206',\n",
       "  'inference_gene_symbol': ['CREBBP', 'EP300', 'NCOA3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Circadian Clock',\n",
       "  'react_pathway_id': 'R-HSA-400253',\n",
       "  'inference_gene_symbol': ['CREBBP', 'EP300']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Sema4D in semaphorin signaling',\n",
       "  'react_pathway_id': 'R-HSA-400685',\n",
       "  'inference_gene_symbol': 'MET'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'PCP/CE pathway',\n",
       "  'react_pathway_id': 'R-HSA-4086400',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'G alpha (q) signalling events',\n",
       "  'react_pathway_id': 'R-HSA-416476',\n",
       "  'inference_gene_symbol': ['ANXA1', 'PIK3CA', 'PTGER1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'G alpha (12/13) signalling events',\n",
       "  'react_pathway_id': 'R-HSA-416482',\n",
       "  'inference_gene_symbol': ['PIK3CA', 'TIAM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Sema4D mediated inhibition of cell attachment and migration',\n",
       "  'react_pathway_id': 'R-HSA-416550',\n",
       "  'inference_gene_symbol': 'MET'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Other semaphorin interactions',\n",
       "  'react_pathway_id': 'R-HSA-416700',\n",
       "  'inference_gene_symbol': 'ITGB1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Platelet homeostasis',\n",
       "  'react_pathway_id': 'R-HSA-418346',\n",
       "  'inference_gene_symbol': 'PTPN11'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'G alpha (i) signalling events',\n",
       "  'react_pathway_id': 'R-HSA-418594',\n",
       "  'inference_gene_symbol': ['ANXA1', 'CXCL8', 'HCAR2', 'HCAR3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'DCC mediated attractive signaling',\n",
       "  'react_pathway_id': 'R-HSA-418885',\n",
       "  'inference_gene_symbol': 'PTK2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Netrin mediated repulsion signals',\n",
       "  'react_pathway_id': 'R-HSA-418886',\n",
       "  'inference_gene_symbol': ['PTK2', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Adherens junctions interactions',\n",
       "  'react_pathway_id': 'R-HSA-418990',\n",
       "  'inference_gene_symbol': 'CDH1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cell-cell junction organization',\n",
       "  'react_pathway_id': 'R-HSA-421270',\n",
       "  'inference_gene_symbol': 'CDH1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Clathrin derived vesicle budding',\n",
       "  'react_pathway_id': 'R-HSA-421837',\n",
       "  'inference_gene_symbol': 'HSPA8'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Axon guidance',\n",
       "  'react_pathway_id': 'R-HSA-422475',\n",
       "  'inference_gene_symbol': ['CSNK2A1',\n",
       "   'CSNK2A2',\n",
       "   'CSNK2B',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'HSPA8',\n",
       "   'ITGA5',\n",
       "   'ITGB1',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'MMP9',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'TIAM1',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'ERCC6 (CSB) and EHMT2 (G9a) positively regulate rRNA expression',\n",
       "  'react_pathway_id': 'R-HSA-427389',\n",
       "  'inference_gene_symbol': ['HDAC1', 'HDAC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'NoRC negatively regulates rRNA expression',\n",
       "  'react_pathway_id': 'R-HSA-427413',\n",
       "  'inference_gene_symbol': ['ERCC2', 'HDAC1', 'HDAC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Vasopressin regulates renal water homeostasis via Aquaporins',\n",
       "  'react_pathway_id': 'R-HSA-432040',\n",
       "  'inference_gene_symbol': 'AQP3'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Passive transport by Aquaporins',\n",
       "  'react_pathway_id': 'R-HSA-432047',\n",
       "  'inference_gene_symbol': 'AQP3'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Platelet sensitization by LDL',\n",
       "  'react_pathway_id': 'R-HSA-432142',\n",
       "  'inference_gene_symbol': 'PTPN11'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Lysosome Vesicle Biogenesis',\n",
       "  'react_pathway_id': 'R-HSA-432720',\n",
       "  'inference_gene_symbol': 'HSPA8'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Golgi Associated Vesicle Biogenesis',\n",
       "  'react_pathway_id': 'R-HSA-432722',\n",
       "  'inference_gene_symbol': 'HSPA8'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Recycling pathway of L1',\n",
       "  'react_pathway_id': 'R-HSA-437239',\n",
       "  'inference_gene_symbol': 'MAPK1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Post NMDA receptor activation events',\n",
       "  'react_pathway_id': 'R-HSA-438064',\n",
       "  'inference_gene_symbol': ['HRAS', 'MAPK1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Binding of TCF/LEF:CTNNB1 to target gene promoters',\n",
       "  'react_pathway_id': 'R-HSA-4411364',\n",
       "  'inference_gene_symbol': 'MYC'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'VEGFA-VEGFR2 Pathway',\n",
       "  'react_pathway_id': 'R-HSA-4420097',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'HSPB1',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'CREB phosphorylation through the activation of Ras',\n",
       "  'react_pathway_id': 'R-HSA-442742',\n",
       "  'inference_gene_symbol': ['HRAS', 'MAPK1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Activation of NMDA receptor upon glutamate binding and postsynaptic events',\n",
       "  'react_pathway_id': 'R-HSA-442755',\n",
       "  'inference_gene_symbol': ['HRAS', 'MAPK1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Ras activation uopn Ca2+ infux through NMDA receptor',\n",
       "  'react_pathway_id': 'R-HSA-442982',\n",
       "  'inference_gene_symbol': 'HRAS'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RSK activation',\n",
       "  'react_pathway_id': 'R-HSA-444257',\n",
       "  'inference_gene_symbol': 'MAPK1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Formyl peptide receptors bind formyl peptides and many other ligands',\n",
       "  'react_pathway_id': 'R-HSA-444473',\n",
       "  'inference_gene_symbol': 'ANXA1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signal transduction by L1',\n",
       "  'react_pathway_id': 'R-HSA-445144',\n",
       "  'inference_gene_symbol': ['CSNK2A1',\n",
       "   'CSNK2A2',\n",
       "   'CSNK2B',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'ITGA5',\n",
       "   'ITGB1',\n",
       "   'MAPK1',\n",
       "   'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Smooth Muscle Contraction',\n",
       "  'react_pathway_id': 'R-HSA-445355',\n",
       "  'inference_gene_symbol': ['ANXA1', 'ANXA2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Aquaporin-mediated transport',\n",
       "  'react_pathway_id': 'R-HSA-445717',\n",
       "  'inference_gene_symbol': 'AQP3'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Asparagine N-linked glycosylation',\n",
       "  'react_pathway_id': 'R-HSA-446203',\n",
       "  'inference_gene_symbol': 'AREG'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Localization of the PINCH-ILK-PARVIN complex to focal adhesions',\n",
       "  'react_pathway_id': 'R-HSA-446343',\n",
       "  'inference_gene_symbol': 'ITGB1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cell-extracellular matrix interactions',\n",
       "  'react_pathway_id': 'R-HSA-446353',\n",
       "  'inference_gene_symbol': 'ITGB1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cell junction organization',\n",
       "  'react_pathway_id': 'R-HSA-446728',\n",
       "  'inference_gene_symbol': ['CDH1', 'ITGB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'CHL1 interactions',\n",
       "  'react_pathway_id': 'R-HSA-447041',\n",
       "  'inference_gene_symbol': ['HSPA8', 'ITGB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by Interleukins',\n",
       "  'react_pathway_id': 'R-HSA-449147',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'ANXA1',\n",
       "   'BIRC5',\n",
       "   'CCL2',\n",
       "   'CCND1',\n",
       "   'CXCL8',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'HSPA8',\n",
       "   'IL6',\n",
       "   'ITGB1',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'MMP9',\n",
       "   'MYC',\n",
       "   'PIK3CA',\n",
       "   'PTGS2',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'SOX2',\n",
       "   'STAT3',\n",
       "   'TGFB1',\n",
       "   'TP53',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'MAPK targets/ Nuclear events mediated by MAP kinases',\n",
       "  'react_pathway_id': 'R-HSA-450282',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'MAP kinase activation in TLR cascade',\n",
       "  'react_pathway_id': 'R-HSA-450294',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Activation of the AP-1 family of transcription factors',\n",
       "  'react_pathway_id': 'R-HSA-450341',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Butyrate Response Factor 1 (BRF1) binds and destabilizes mRNA',\n",
       "  'react_pathway_id': 'R-HSA-450385',\n",
       "  'inference_gene_symbol': 'AKT1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'AUF1 (hnRNP D0) binds and destabilizes mRNA',\n",
       "  'react_pathway_id': 'R-HSA-450408',\n",
       "  'inference_gene_symbol': ['HSPA8', 'HSPB1', 'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of mRNA stability by proteins that bind AU-rich elements',\n",
       "  'react_pathway_id': 'R-HSA-450531',\n",
       "  'inference_gene_symbol': ['AKT1', 'HSPA8', 'HSPB1', 'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'KSRP (KHSRP) binds and destabilizes mRNA',\n",
       "  'react_pathway_id': 'R-HSA-450604',\n",
       "  'inference_gene_symbol': 'AKT1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Interleukin-2 signaling',\n",
       "  'react_pathway_id': 'R-HSA-451927',\n",
       "  'inference_gene_symbol': ['EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'RASA1',\n",
       "   'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Transcriptional regulation of pluripotent stem cells',\n",
       "  'react_pathway_id': 'R-HSA-452723',\n",
       "  'inference_gene_symbol': ['SOX2', 'STAT3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Mitotic G2-G2/M phases',\n",
       "  'react_pathway_id': 'R-HSA-453274',\n",
       "  'inference_gene_symbol': ['EP300', 'SEM1', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of mitotic cell cycle',\n",
       "  'react_pathway_id': 'R-HSA-453276',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Mitotic G1-G1/S phases',\n",
       "  'react_pathway_id': 'R-HSA-453279',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CCND1',\n",
       "   'CDKN2A',\n",
       "   'HDAC1',\n",
       "   'MYC',\n",
       "   'SEM1',\n",
       "   'TYMS']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Thrombin signalling through proteinase activated receptors (PARs)',\n",
       "  'react_pathway_id': 'R-HSA-456926',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Asymmetric localization of PCP proteins',\n",
       "  'react_pathway_id': 'R-HSA-4608870',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'SUMOylation of DNA replication proteins',\n",
       "  'react_pathway_id': 'R-HSA-4615885',\n",
       "  'inference_gene_symbol': 'BIRC5'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Degradation of AXIN',\n",
       "  'react_pathway_id': 'R-HSA-4641257',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Degradation of DVL',\n",
       "  'react_pathway_id': 'R-HSA-4641258',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of FZD by ubiquitination',\n",
       "  'react_pathway_id': 'R-HSA-4641263',\n",
       "  'inference_gene_symbol': 'LGR6'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Repression of WNT target genes',\n",
       "  'react_pathway_id': 'R-HSA-4641265',\n",
       "  'inference_gene_symbol': 'HDAC1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Chromatin organization',\n",
       "  'react_pathway_id': 'R-HSA-4839726',\n",
       "  'inference_gene_symbol': ['CCND1', 'CREBBP', 'EP300', 'HDAC1', 'HDAC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Pyrimidine biosynthesis',\n",
       "  'react_pathway_id': 'R-HSA-500753',\n",
       "  'inference_gene_symbol': ['TYMS', 'UMPS']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'GPCR ligand binding',\n",
       "  'react_pathway_id': 'R-HSA-500792',\n",
       "  'inference_gene_symbol': ['ANXA1', 'CXCL8', 'HCAR2', 'HCAR3', 'PTGER1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RNA Polymerase I, RNA Polymerase III, and Mitochondrial Transcription',\n",
       "  'react_pathway_id': 'R-HSA-504046',\n",
       "  'inference_gene_symbol': ['ERCC2', 'HDAC1', 'HDAC2', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Defective LFNG causes SCDO3',\n",
       "  'react_pathway_id': 'R-HSA-5083630',\n",
       "  'inference_gene_symbol': ['NOTCH1', 'NOTCH2', 'NOTCH3', 'NOTCH4']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Interleukin-3, 5 and GM-CSF signaling',\n",
       "  'react_pathway_id': 'R-HSA-512988',\n",
       "  'inference_gene_symbol': ['EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'RASA1',\n",
       "   'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'VEGFR2 mediated vascular permeability',\n",
       "  'react_pathway_id': 'R-HSA-5218920',\n",
       "  'inference_gene_symbol': 'AKT1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'VEGFR2 mediated cell proliferation',\n",
       "  'react_pathway_id': 'R-HSA-5218921',\n",
       "  'inference_gene_symbol': ['EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PTK2',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Positive epigenetic regulation of rRNA expression',\n",
       "  'react_pathway_id': 'R-HSA-5250913',\n",
       "  'inference_gene_symbol': ['EP300', 'HDAC1', 'HDAC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'B-WICH complex positively regulates rRNA expression',\n",
       "  'react_pathway_id': 'R-HSA-5250924',\n",
       "  'inference_gene_symbol': 'EP300'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Negative epigenetic regulation of rRNA expression',\n",
       "  'react_pathway_id': 'R-HSA-5250941',\n",
       "  'inference_gene_symbol': ['ERCC2', 'HDAC1', 'HDAC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Fatty acid, triacylglycerol, and ketone body metabolism',\n",
       "  'react_pathway_id': 'R-HSA-535734',\n",
       "  'inference_gene_symbol': ['CREBBP', 'EP300', 'NCOA3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Programmed Cell Death',\n",
       "  'react_pathway_id': 'R-HSA-5357801',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CDH1',\n",
       "   'PTK2',\n",
       "   'SEM1',\n",
       "   'SFN',\n",
       "   'TP53',\n",
       "   'TP63']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of TNFR1 signaling',\n",
       "  'react_pathway_id': 'R-HSA-5357905',\n",
       "  'inference_gene_symbol': 'RACK1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TNFR1-induced NFkappaB signaling pathway',\n",
       "  'react_pathway_id': 'R-HSA-5357956',\n",
       "  'inference_gene_symbol': 'RACK1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Hedgehog ligand biogenesis',\n",
       "  'react_pathway_id': 'R-HSA-5358346',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by Hedgehog',\n",
       "  'react_pathway_id': 'R-HSA-5358351',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by Retinoic Acid',\n",
       "  'react_pathway_id': 'R-HSA-5362517',\n",
       "  'inference_gene_symbol': 'RARB'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': \"Hh mutants that don't undergo autocatalytic processing are degraded by ERAD\",\n",
       "  'react_pathway_id': 'R-HSA-5362768',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Mitochondrial translation initiation',\n",
       "  'react_pathway_id': 'R-HSA-5368286',\n",
       "  'inference_gene_symbol': 'MRPL13'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Mitochondrial translation',\n",
       "  'react_pathway_id': 'R-HSA-5368287',\n",
       "  'inference_gene_symbol': 'MRPL13'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Hh mutants abrogate ligand secretion',\n",
       "  'react_pathway_id': 'R-HSA-5387390',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Mitochondrial translation elongation',\n",
       "  'react_pathway_id': 'R-HSA-5389840',\n",
       "  'inference_gene_symbol': 'MRPL13'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Mitochondrial translation termination',\n",
       "  'react_pathway_id': 'R-HSA-5419276',\n",
       "  'inference_gene_symbol': 'MRPL13'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Aflatoxin activation and detoxification',\n",
       "  'react_pathway_id': 'R-HSA-5423646',\n",
       "  'inference_gene_symbol': 'CYP1A2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Metabolism of lipids and lipoproteins',\n",
       "  'react_pathway_id': 'R-HSA-556833',\n",
       "  'inference_gene_symbol': ['CREBBP',\n",
       "   'CSNK2A1',\n",
       "   'CSNK2A2',\n",
       "   'CSNK2B',\n",
       "   'CYP1A2',\n",
       "   'CYP1B1',\n",
       "   'EP300',\n",
       "   'GPX1',\n",
       "   'NCOA3',\n",
       "   'PIK3CA',\n",
       "   'PTGS2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Dectin-1 mediated noncanonical NF-kB signaling',\n",
       "  'react_pathway_id': 'R-HSA-5607761',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'CLEC7A (Dectin-1) signaling',\n",
       "  'react_pathway_id': 'R-HSA-5607764',\n",
       "  'inference_gene_symbol': ['PYCARD', 'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Degradation of GLI1 by the proteasome',\n",
       "  'react_pathway_id': 'R-HSA-5610780',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Degradation of GLI2 by the proteasome',\n",
       "  'react_pathway_id': 'R-HSA-5610783',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'GLI3 is processed to GLI3R by the proteasome',\n",
       "  'react_pathway_id': 'R-HSA-5610785',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': \"Hedgehog 'off' state\",\n",
       "  'react_pathway_id': 'R-HSA-5610787',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Activation of anterior HOX genes in hindbrain development during early embryogenesis',\n",
       "  'react_pathway_id': 'R-HSA-5617472',\n",
       "  'inference_gene_symbol': ['CREBBP', 'EP300', 'NCOA3', 'RARB']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'ABC transporter disorders',\n",
       "  'react_pathway_id': 'R-HSA-5619084',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Disorders of transmembrane transporters',\n",
       "  'react_pathway_id': 'R-HSA-5619115',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Activation of HOX genes during differentiation',\n",
       "  'react_pathway_id': 'R-HSA-5619507',\n",
       "  'inference_gene_symbol': ['CREBBP', 'EP300', 'NCOA3', 'RARB']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'C-type lectin receptors (CLRs)',\n",
       "  'react_pathway_id': 'R-HSA-5621481',\n",
       "  'inference_gene_symbol': ['CREBBP',\n",
       "   'EP300',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'PYCARD',\n",
       "   'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'CD209 (DC-SIGN) signaling',\n",
       "  'react_pathway_id': 'R-HSA-5621575',\n",
       "  'inference_gene_symbol': ['CREBBP', 'EP300', 'HRAS', 'KRAS']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RHO GTPases activate PKNs',\n",
       "  'react_pathway_id': 'R-HSA-5625740',\n",
       "  'inference_gene_symbol': 'SFN'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RHO GTPases activate IQGAPs',\n",
       "  'react_pathway_id': 'R-HSA-5626467',\n",
       "  'inference_gene_symbol': 'CDH1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TNFR1-mediated ceramide production',\n",
       "  'react_pathway_id': 'R-HSA-5626978',\n",
       "  'inference_gene_symbol': 'RACK1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RHO GTPases activate PAKs',\n",
       "  'react_pathway_id': 'R-HSA-5627123',\n",
       "  'inference_gene_symbol': 'CTTN'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TP53 Regulates Metabolic Genes',\n",
       "  'react_pathway_id': 'R-HSA-5628897',\n",
       "  'inference_gene_symbol': ['AKT1', 'SFN', 'TP53', 'TP63', 'TSC1', 'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': \"Hedgehog 'on' state\",\n",
       "  'react_pathway_id': 'R-HSA-5632684',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of TP53 Activity',\n",
       "  'react_pathway_id': 'R-HSA-5633007',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'ATM',\n",
       "   'CDKN2A',\n",
       "   'CSNK2A1',\n",
       "   'CSNK2A2',\n",
       "   'CSNK2B',\n",
       "   'EP300',\n",
       "   'HDAC1',\n",
       "   'HDAC2',\n",
       "   'STK11',\n",
       "   'TP53',\n",
       "   'TP63']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TP53 Regulates Transcription of Cell Death Genes',\n",
       "  'react_pathway_id': 'R-HSA-5633008',\n",
       "  'inference_gene_symbol': ['ATM', 'BIRC5', 'CREBBP', 'TP53', 'TP63']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Constitutive Signaling by EGFRvIII',\n",
       "  'react_pathway_id': 'R-HSA-5637810',\n",
       "  'inference_gene_symbol': ['EGFR', 'HRAS', 'KRAS', 'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by EGFRvIII in Cancer',\n",
       "  'react_pathway_id': 'R-HSA-5637812',\n",
       "  'inference_gene_symbol': ['EGFR', 'HRAS', 'KRAS', 'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by Ligand-Responsive EGFR Variants in Cancer',\n",
       "  'react_pathway_id': 'R-HSA-5637815',\n",
       "  'inference_gene_symbol': ['EGFR', 'HRAS', 'KRAS', 'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by Overexpressed Wild-Type EGFR in Cancer',\n",
       "  'react_pathway_id': 'R-HSA-5638302',\n",
       "  'inference_gene_symbol': 'EGFR'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Inhibition of Signaling by Overexpressed EGFR',\n",
       "  'react_pathway_id': 'R-HSA-5638303',\n",
       "  'inference_gene_symbol': 'EGFR'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Vesicle-mediated transport',\n",
       "  'react_pathway_id': 'R-HSA-5653656',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'AREG',\n",
       "   'CTTN',\n",
       "   'EGFR',\n",
       "   'HSPA8',\n",
       "   'SFN',\n",
       "   'TSC1',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Phospholipase C-mediated cascade: FGFR1',\n",
       "  'react_pathway_id': 'R-HSA-5654219',\n",
       "  'inference_gene_symbol': 'FGFR1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Downstream signaling of activated FGFR1',\n",
       "  'react_pathway_id': 'R-HSA-5654687',\n",
       "  'inference_gene_symbol': ['FGFR1', 'HRAS', 'KRAS', 'PIK3CA', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'SHC-mediated cascade:FGFR1',\n",
       "  'react_pathway_id': 'R-HSA-5654688',\n",
       "  'inference_gene_symbol': ['FGFR1', 'HRAS', 'KRAS']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'PI-3K cascade:FGFR1',\n",
       "  'react_pathway_id': 'R-HSA-5654689',\n",
       "  'inference_gene_symbol': ['FGFR1', 'PIK3CA', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'FRS-mediated FGFR1 signaling',\n",
       "  'react_pathway_id': 'R-HSA-5654693',\n",
       "  'inference_gene_symbol': ['FGFR1', 'HRAS', 'KRAS', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'PI-3K cascade:FGFR2',\n",
       "  'react_pathway_id': 'R-HSA-5654695',\n",
       "  'inference_gene_symbol': ['PIK3CA', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Downstream signaling of activated FGFR2',\n",
       "  'react_pathway_id': 'R-HSA-5654696',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS', 'PIK3CA', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'SHC-mediated cascade:FGFR2',\n",
       "  'react_pathway_id': 'R-HSA-5654699',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'FRS-mediated FGFR2 signaling',\n",
       "  'react_pathway_id': 'R-HSA-5654700',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'SHC-mediated cascade:FGFR3',\n",
       "  'react_pathway_id': 'R-HSA-5654704',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'FRS-mediated FGFR3 signaling',\n",
       "  'react_pathway_id': 'R-HSA-5654706',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Downstream signaling of activated FGFR3',\n",
       "  'react_pathway_id': 'R-HSA-5654708',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS', 'PIK3CA', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'PI-3K cascade:FGFR3',\n",
       "  'react_pathway_id': 'R-HSA-5654710',\n",
       "  'inference_gene_symbol': ['PIK3CA', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'FRS-mediated FGFR4 signaling',\n",
       "  'react_pathway_id': 'R-HSA-5654712',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Downstream signaling of activated FGFR4',\n",
       "  'react_pathway_id': 'R-HSA-5654716',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS', 'PIK3CA', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'SHC-mediated cascade:FGFR4',\n",
       "  'react_pathway_id': 'R-HSA-5654719',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'PI-3K cascade:FGFR4',\n",
       "  'react_pathway_id': 'R-HSA-5654720',\n",
       "  'inference_gene_symbol': ['PIK3CA', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Negative regulation of FGFR1 signaling',\n",
       "  'react_pathway_id': 'R-HSA-5654726',\n",
       "  'inference_gene_symbol': ['FGFR1', 'MAPK1', 'MAPK3', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Negative regulation of FGFR2 signaling',\n",
       "  'react_pathway_id': 'R-HSA-5654727',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Negative regulation of FGFR3 signaling',\n",
       "  'react_pathway_id': 'R-HSA-5654732',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Negative regulation of FGFR4 signaling',\n",
       "  'react_pathway_id': 'R-HSA-5654733',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by FGFR1',\n",
       "  'react_pathway_id': 'R-HSA-5654736',\n",
       "  'inference_gene_symbol': ['FGFR1',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by FGFR2',\n",
       "  'react_pathway_id': 'R-HSA-5654738',\n",
       "  'inference_gene_symbol': ['HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by FGFR3',\n",
       "  'react_pathway_id': 'R-HSA-5654741',\n",
       "  'inference_gene_symbol': ['HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by FGFR4',\n",
       "  'react_pathway_id': 'R-HSA-5654743',\n",
       "  'inference_gene_symbol': ['HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by FGFR2 in disease',\n",
       "  'react_pathway_id': 'R-HSA-5655253',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS', 'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by FGFR4 in disease',\n",
       "  'react_pathway_id': 'R-HSA-5655291',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS', 'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by FGFR1 in disease',\n",
       "  'react_pathway_id': 'R-HSA-5655302',\n",
       "  'inference_gene_symbol': ['FGFR1', 'HRAS', 'KRAS', 'PIK3CA', 'STAT3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by FGFR3 in disease',\n",
       "  'react_pathway_id': 'R-HSA-5655332',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS', 'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Termination of translesion DNA synthesis',\n",
       "  'react_pathway_id': 'R-HSA-5656169',\n",
       "  'inference_gene_symbol': 'UBA7'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of RAS by GAPs',\n",
       "  'react_pathway_id': 'R-HSA-5658442',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS', 'RASA1', 'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Response to metal ions',\n",
       "  'react_pathway_id': 'R-HSA-5660526',\n",
       "  'inference_gene_symbol': 'MT1A'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'CLEC7A/inflammasome pathway',\n",
       "  'react_pathway_id': 'R-HSA-5660668',\n",
       "  'inference_gene_symbol': 'PYCARD'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Metallothioneins bind metals',\n",
       "  'react_pathway_id': 'R-HSA-5661231',\n",
       "  'inference_gene_symbol': 'MT1A'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Melanin biosynthesis',\n",
       "  'react_pathway_id': 'R-HSA-5662702',\n",
       "  'inference_gene_symbol': 'SLC45A2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Diseases of signal transduction',\n",
       "  'react_pathway_id': 'R-HSA-5663202',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CREBBP',\n",
       "   'EGFR',\n",
       "   'EP300',\n",
       "   'FGFR1',\n",
       "   'HDAC1',\n",
       "   'HDAC2',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'MYC',\n",
       "   'NOTCH1',\n",
       "   'PIK3CA',\n",
       "   'PTPN11',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'STAT3',\n",
       "   'TGFB1',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Infectious disease',\n",
       "  'react_pathway_id': 'R-HSA-5663205',\n",
       "  'inference_gene_symbol': ['CDH1', 'ERCC2', 'MET', 'PPIA', 'SEM1', 'TGFB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RHO GTPases Activate WASPs and WAVEs',\n",
       "  'react_pathway_id': 'R-HSA-5663213',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3', 'PTK2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RHO GTPases Activate Formins',\n",
       "  'react_pathway_id': 'R-HSA-5663220',\n",
       "  'inference_gene_symbol': ['BIRC5', 'ITGB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TNFR2 non-canonical NF-kB pathway',\n",
       "  'react_pathway_id': 'R-HSA-5668541',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RAF activation',\n",
       "  'react_pathway_id': 'R-HSA-5673000',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RAF/MAP kinase cascade',\n",
       "  'react_pathway_id': 'R-HSA-5673001',\n",
       "  'inference_gene_symbol': ['EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PTK2',\n",
       "   'RASA1',\n",
       "   'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'MAP2K and MAPK activation',\n",
       "  'react_pathway_id': 'R-HSA-5674135',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS', 'MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Constitutive Signaling by AKT1 E17K in Cancer',\n",
       "  'react_pathway_id': 'R-HSA-5674400',\n",
       "  'inference_gene_symbol': ['AKT1', 'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Negative feedback regulation of MAPK pathway',\n",
       "  'react_pathway_id': 'R-HSA-5674499',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Negative regulation of MAPK pathway',\n",
       "  'react_pathway_id': 'R-HSA-5675221',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS', 'MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'NIK-->noncanonical NF-kB signaling',\n",
       "  'react_pathway_id': 'R-HSA-5676590',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Defective CFTR causes cystic fibrosis',\n",
       "  'react_pathway_id': 'R-HSA-5678895',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'MAPK family signaling cascades',\n",
       "  'react_pathway_id': 'R-HSA-5683057',\n",
       "  'inference_gene_symbol': ['EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'HSPB1',\n",
       "   'IL6',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'MYC',\n",
       "   'NCOA3',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'RASA1',\n",
       "   'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'MAPK1/MAPK3 signaling',\n",
       "  'react_pathway_id': 'R-HSA-5684996',\n",
       "  'inference_gene_symbol': ['EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'IL6',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'RASA1',\n",
       "   'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'HDR through Single Strand Annealing (SSA)',\n",
       "  'react_pathway_id': 'R-HSA-5685938',\n",
       "  'inference_gene_symbol': ['ATM', 'RAD52']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'HDR through Homologous Recombination (HRR)',\n",
       "  'react_pathway_id': 'R-HSA-5685942',\n",
       "  'inference_gene_symbol': 'ATM'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'MAPK6/MAPK4 signaling',\n",
       "  'react_pathway_id': 'R-HSA-5687128',\n",
       "  'inference_gene_symbol': ['HSPB1', 'MYC', 'NCOA3', 'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Deubiquitination',\n",
       "  'react_pathway_id': 'R-HSA-5688426',\n",
       "  'inference_gene_symbol': ['EP300', 'MYC', 'SEM1', 'TGFB1', 'TP53', 'VDAC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'UCH proteinases',\n",
       "  'react_pathway_id': 'R-HSA-5689603',\n",
       "  'inference_gene_symbol': ['SEM1', 'TGFB1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Ub-specific processing proteases',\n",
       "  'react_pathway_id': 'R-HSA-5689880',\n",
       "  'inference_gene_symbol': ['MYC', 'SEM1', 'TP53', 'VDAC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Ovarian tumor domain proteases',\n",
       "  'react_pathway_id': 'R-HSA-5689896',\n",
       "  'inference_gene_symbol': 'TP53'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Metalloprotease DUBs',\n",
       "  'react_pathway_id': 'R-HSA-5689901',\n",
       "  'inference_gene_symbol': 'EP300'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'DNA Double-Strand Break Repair',\n",
       "  'react_pathway_id': 'R-HSA-5693532',\n",
       "  'inference_gene_symbol': ['ATM', 'PRKDC', 'RAD52', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Resolution of D-Loop Structures',\n",
       "  'react_pathway_id': 'R-HSA-5693537',\n",
       "  'inference_gene_symbol': 'ATM'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Homology Directed Repair',\n",
       "  'react_pathway_id': 'R-HSA-5693538',\n",
       "  'inference_gene_symbol': ['ATM', 'RAD52']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Sensing of DNA Double Strand Breaks',\n",
       "  'react_pathway_id': 'R-HSA-5693548',\n",
       "  'inference_gene_symbol': 'ATM'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Resolution of D-loop Structures through Synthesis-Dependent Strand Annealing (SDSA)',\n",
       "  'react_pathway_id': 'R-HSA-5693554',\n",
       "  'inference_gene_symbol': 'ATM'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Recruitment and ATM-mediated phosphorylation of repair and signaling proteins at DNA double strand breaks',\n",
       "  'react_pathway_id': 'R-HSA-5693565',\n",
       "  'inference_gene_symbol': ['ATM', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'HDR through Homologous Recombination (HR) or Single Strand Annealing (SSA)',\n",
       "  'react_pathway_id': 'R-HSA-5693567',\n",
       "  'inference_gene_symbol': ['ATM', 'RAD52']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Resolution of D-loop Structures through Holliday Junction Intermediates',\n",
       "  'react_pathway_id': 'R-HSA-5693568',\n",
       "  'inference_gene_symbol': 'ATM'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Nonhomologous End-Joining (NHEJ)',\n",
       "  'react_pathway_id': 'R-HSA-5693571',\n",
       "  'inference_gene_symbol': ['ATM', 'PRKDC']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Homologous DNA Pairing and Strand Exchange',\n",
       "  'react_pathway_id': 'R-HSA-5693579',\n",
       "  'inference_gene_symbol': 'ATM'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'DNA Double Strand Break Response',\n",
       "  'react_pathway_id': 'R-HSA-5693606',\n",
       "  'inference_gene_symbol': ['ATM', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Processing of DNA double-strand break ends',\n",
       "  'react_pathway_id': 'R-HSA-5693607',\n",
       "  'inference_gene_symbol': 'ATM'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Presynaptic phase of homologous DNA pairing and strand exchange',\n",
       "  'react_pathway_id': 'R-HSA-5693616',\n",
       "  'inference_gene_symbol': 'ATM'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cargo concentration in the ER',\n",
       "  'react_pathway_id': 'R-HSA-5694530',\n",
       "  'inference_gene_symbol': 'AREG'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'DNA Damage Recognition in GG-NER',\n",
       "  'react_pathway_id': 'R-HSA-5696394',\n",
       "  'inference_gene_symbol': 'XPC'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Formation of Incision Complex in GG-NER',\n",
       "  'react_pathway_id': 'R-HSA-5696395',\n",
       "  'inference_gene_symbol': ['ERCC2', 'XPC']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Nucleotide Excision Repair',\n",
       "  'react_pathway_id': 'R-HSA-5696398',\n",
       "  'inference_gene_symbol': ['EP300', 'ERCC2', 'XPC']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Global Genome Nucleotide Excision Repair (GG-NER)',\n",
       "  'react_pathway_id': 'R-HSA-5696399',\n",
       "  'inference_gene_symbol': ['ERCC2', 'XPC']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Dual Incision in GG-NER',\n",
       "  'react_pathway_id': 'R-HSA-5696400',\n",
       "  'inference_gene_symbol': 'ERCC2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Post-translational protein modification',\n",
       "  'react_pathway_id': 'R-HSA-597592',\n",
       "  'inference_gene_symbol': ['AREG',\n",
       "   'BIRC5',\n",
       "   'CDKN2A',\n",
       "   'EP300',\n",
       "   'HSPA8',\n",
       "   'MYC',\n",
       "   'PRKDC',\n",
       "   'RAD52',\n",
       "   'SEM1',\n",
       "   'TGFB1',\n",
       "   'TP53',\n",
       "   'UBA1',\n",
       "   'VDAC2',\n",
       "   'XPC']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Inflammasomes',\n",
       "  'react_pathway_id': 'R-HSA-622312',\n",
       "  'inference_gene_symbol': 'PYCARD'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RNA Polymerase II Pre-transcription Events',\n",
       "  'react_pathway_id': 'R-HSA-674695',\n",
       "  'inference_gene_symbol': 'ERCC2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Formation of TC-NER Pre-Incision Complex',\n",
       "  'react_pathway_id': 'R-HSA-6781823',\n",
       "  'inference_gene_symbol': ['EP300', 'ERCC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Transcription-Coupled Nucleotide Excision Repair (TC-NER)',\n",
       "  'react_pathway_id': 'R-HSA-6781827',\n",
       "  'inference_gene_symbol': ['EP300', 'ERCC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Dual incision in TC-NER',\n",
       "  'react_pathway_id': 'R-HSA-6782135',\n",
       "  'inference_gene_symbol': ['EP300', 'ERCC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Gap-filling DNA repair synthesis and ligation in TC-NER',\n",
       "  'react_pathway_id': 'R-HSA-6782210',\n",
       "  'inference_gene_symbol': ['EP300', 'ERCC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Interleukin-6 family signaling',\n",
       "  'react_pathway_id': 'R-HSA-6783589',\n",
       "  'inference_gene_symbol': ['IL6', 'PTPN11', 'STAT3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Interleukin-10 signaling',\n",
       "  'react_pathway_id': 'R-HSA-6783783',\n",
       "  'inference_gene_symbol': ['CCL2', 'CXCL8', 'IL6', 'PTGS2', 'STAT3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'ERBB2 Regulates Cell Motility',\n",
       "  'react_pathway_id': 'R-HSA-6785631',\n",
       "  'inference_gene_symbol': 'EGFR'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Interleukin-4 and 13 signaling',\n",
       "  'react_pathway_id': 'R-HSA-6785807',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'ANXA1',\n",
       "   'BIRC5',\n",
       "   'CCL2',\n",
       "   'CCND1',\n",
       "   'CXCL8',\n",
       "   'HGF',\n",
       "   'HSPA8',\n",
       "   'IL6',\n",
       "   'ITGB1',\n",
       "   'MMP9',\n",
       "   'MYC',\n",
       "   'PTGS2',\n",
       "   'SOX2',\n",
       "   'STAT3',\n",
       "   'TGFB1',\n",
       "   'TP53',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TP53 Regulates Transcription of Cell Cycle Genes',\n",
       "  'react_pathway_id': 'R-HSA-6791312',\n",
       "  'inference_gene_symbol': ['EP300', 'SFN', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TP53 Regulates Transcription of DNA Repair Genes',\n",
       "  'react_pathway_id': 'R-HSA-6796648',\n",
       "  'inference_gene_symbol': ['ATM', 'ERCC2', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Neutrophil degranulation',\n",
       "  'react_pathway_id': 'R-HSA-6798695',\n",
       "  'inference_gene_symbol': ['ANXA2',\n",
       "   'CSNK2B',\n",
       "   'HSPA8',\n",
       "   'KRT1',\n",
       "   'MAPK1',\n",
       "   'MMP9',\n",
       "   'PGAM1',\n",
       "   'PPIA',\n",
       "   'PYCARD',\n",
       "   'SERPINB3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by moderate kinase activity BRAF mutants',\n",
       "  'react_pathway_id': 'R-HSA-6802946',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS', 'MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by high-kinase activity BRAF mutants',\n",
       "  'react_pathway_id': 'R-HSA-6802948',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS', 'MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by RAS mutants',\n",
       "  'react_pathway_id': 'R-HSA-6802949',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS', 'MAPK1', 'MAPK3', 'RASA1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by BRAF and RAF fusions',\n",
       "  'react_pathway_id': 'R-HSA-6802952',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS', 'MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RAS signaling downstream of NF1 loss-of-function variants',\n",
       "  'react_pathway_id': 'R-HSA-6802953',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Paradoxical activation of RAF signaling by kinase inactive BRAF',\n",
       "  'react_pathway_id': 'R-HSA-6802955',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS', 'MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Oncogenic MAPK signaling',\n",
       "  'react_pathway_id': 'R-HSA-6802957',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS', 'MAPK1', 'MAPK3', 'RASA1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TP53 Regulates Transcription of Genes Involved in Cytochrome C Release',\n",
       "  'react_pathway_id': 'R-HSA-6803204',\n",
       "  'inference_gene_symbol': ['ATM', 'CREBBP', 'TP53', 'TP63']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TP53 regulates transcription of several additional cell death genes whose specific roles in p53-dependent apoptosis remain uncertain',\n",
       "  'react_pathway_id': 'R-HSA-6803205',\n",
       "  'inference_gene_symbol': ['BIRC5', 'TP53', 'TP63']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TP53 Regulates Transcription of Caspase Activators and Caspases',\n",
       "  'react_pathway_id': 'R-HSA-6803207',\n",
       "  'inference_gene_symbol': ['ATM', 'TP53', 'TP63']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TP53 Regulates Transcription of Death Receptors and Ligands',\n",
       "  'react_pathway_id': 'R-HSA-6803211',\n",
       "  'inference_gene_symbol': ['TP53', 'TP63']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TP53 Regulates Transcription of Genes Involved in G2 Cell Cycle Arrest',\n",
       "  'react_pathway_id': 'R-HSA-6804114',\n",
       "  'inference_gene_symbol': ['EP300', 'SFN', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TP53 regulates transcription of additional cell cycle genes whose exact role in the p53 pathway remain uncertain',\n",
       "  'react_pathway_id': 'R-HSA-6804115',\n",
       "  'inference_gene_symbol': 'TP53'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TP53 Regulates Transcription of Genes Involved in G1 Cell Cycle Arrest',\n",
       "  'react_pathway_id': 'R-HSA-6804116',\n",
       "  'inference_gene_symbol': 'TP53'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of TP53 Expression',\n",
       "  'react_pathway_id': 'R-HSA-6804754',\n",
       "  'inference_gene_symbol': 'TP53'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of TP53 Activity through Phosphorylation',\n",
       "  'react_pathway_id': 'R-HSA-6804756',\n",
       "  'inference_gene_symbol': ['ATM',\n",
       "   'CSNK2A1',\n",
       "   'CSNK2A2',\n",
       "   'CSNK2B',\n",
       "   'STK11',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of TP53 Degradation',\n",
       "  'react_pathway_id': 'R-HSA-6804757',\n",
       "  'inference_gene_symbol': ['AKT1', 'ATM', 'CDKN2A', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of TP53 Activity through Acetylation',\n",
       "  'react_pathway_id': 'R-HSA-6804758',\n",
       "  'inference_gene_symbol': ['AKT1', 'EP300', 'HDAC1', 'HDAC2', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of TP53 Activity through Association with Co-factors',\n",
       "  'react_pathway_id': 'R-HSA-6804759',\n",
       "  'inference_gene_symbol': ['AKT1', 'TP53', 'TP63']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of TP53 Activity through Methylation',\n",
       "  'react_pathway_id': 'R-HSA-6804760',\n",
       "  'inference_gene_symbol': ['ATM', 'EP300', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Keratinization',\n",
       "  'react_pathway_id': 'R-HSA-6805567',\n",
       "  'inference_gene_symbol': ['IVL',\n",
       "   'KRT1',\n",
       "   'KRT10',\n",
       "   'KRT14',\n",
       "   'KRT6B',\n",
       "   'LORICRIN',\n",
       "   'SPRR1A']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of TP53 Expression and Degradation',\n",
       "  'react_pathway_id': 'R-HSA-6806003',\n",
       "  'inference_gene_symbol': ['AKT1', 'ATM', 'CDKN2A', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by MET',\n",
       "  'react_pathway_id': 'R-HSA-6806834',\n",
       "  'inference_gene_symbol': ['HGF',\n",
       "   'HRAS',\n",
       "   'ITGB1',\n",
       "   'KRAS',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'STAT3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'MET Receptor Activation',\n",
       "  'react_pathway_id': 'R-HSA-6806942',\n",
       "  'inference_gene_symbol': ['HGF', 'MET']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Negative regulation of MET activity',\n",
       "  'react_pathway_id': 'R-HSA-6807004',\n",
       "  'inference_gene_symbol': ['HGF', 'MET']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Formation of the cornified envelope',\n",
       "  'react_pathway_id': 'R-HSA-6809371',\n",
       "  'inference_gene_symbol': ['IVL', 'LORICRIN', 'SPRR1A']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'PI5P Regulates TP53 Acetylation',\n",
       "  'react_pathway_id': 'R-HSA-6811555',\n",
       "  'inference_gene_symbol': ['EP300', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'PI5P, PP2A and IER3 Regulate PI3K/AKT Signaling',\n",
       "  'react_pathway_id': 'R-HSA-6811558',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cooperation of PDCL (PhLP1) and TRiC/CCT in G-protein beta folding',\n",
       "  'react_pathway_id': 'R-HSA-6814122',\n",
       "  'inference_gene_symbol': ['CSNK2A1', 'CSNK2A2', 'CSNK2B']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'CDT1 association with the CDC6:ORC:origin complex',\n",
       "  'react_pathway_id': 'R-HSA-68827',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Assembly of the pre-replicative complex',\n",
       "  'react_pathway_id': 'R-HSA-68867',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'M/G1 Transition',\n",
       "  'react_pathway_id': 'R-HSA-68874',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Mitotic Prophase',\n",
       "  'react_pathway_id': 'R-HSA-68875',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Mitotic Prometaphase',\n",
       "  'react_pathway_id': 'R-HSA-68877',\n",
       "  'inference_gene_symbol': ['BIRC5', 'CSNK2A1', 'CSNK2A2', 'CSNK2B']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Mitotic Anaphase',\n",
       "  'react_pathway_id': 'R-HSA-68882',\n",
       "  'inference_gene_symbol': ['BIRC5', 'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'M Phase',\n",
       "  'react_pathway_id': 'R-HSA-68886',\n",
       "  'inference_gene_symbol': ['BIRC5',\n",
       "   'CSNK2A1',\n",
       "   'CSNK2A2',\n",
       "   'CSNK2B',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Orc1 removal from chromatin',\n",
       "  'react_pathway_id': 'R-HSA-68949',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'DNA Replication Pre-Initiation',\n",
       "  'react_pathway_id': 'R-HSA-69002',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'CDK-mediated phosphorylation and removal of Cdc6',\n",
       "  'react_pathway_id': 'R-HSA-69017',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Switching of origins to a post-replicative state',\n",
       "  'react_pathway_id': 'R-HSA-69052',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cyclin E associated events during G1/S transition',\n",
       "  'react_pathway_id': 'R-HSA-69202',\n",
       "  'inference_gene_symbol': ['AKT1', 'CCND1', 'MYC', 'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'G1/S-Specific Transcription',\n",
       "  'react_pathway_id': 'R-HSA-69205',\n",
       "  'inference_gene_symbol': 'TYMS'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'G1/S Transition',\n",
       "  'react_pathway_id': 'R-HSA-69206',\n",
       "  'inference_gene_symbol': ['AKT1', 'CCND1', 'MYC', 'SEM1', 'TYMS']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Ubiquitin-dependent degradation of Cyclin D1',\n",
       "  'react_pathway_id': 'R-HSA-69229',\n",
       "  'inference_gene_symbol': ['CCND1', 'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cyclin D associated events in G1',\n",
       "  'react_pathway_id': 'R-HSA-69231',\n",
       "  'inference_gene_symbol': ['CCND1', 'CDKN2A']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'G1 Phase',\n",
       "  'react_pathway_id': 'R-HSA-69236',\n",
       "  'inference_gene_symbol': ['CCND1', 'CDKN2A']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Synthesis of DNA',\n",
       "  'react_pathway_id': 'R-HSA-69239',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'S Phase',\n",
       "  'react_pathway_id': 'R-HSA-69242',\n",
       "  'inference_gene_symbol': ['CCND1', 'MYC', 'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'G2/M Transition',\n",
       "  'react_pathway_id': 'R-HSA-69275',\n",
       "  'inference_gene_symbol': ['EP300', 'SEM1', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cell Cycle, Mitotic',\n",
       "  'react_pathway_id': 'R-HSA-69278',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'BIRC5',\n",
       "   'CCND1',\n",
       "   'CDKN2A',\n",
       "   'CSNK2A1',\n",
       "   'CSNK2A2',\n",
       "   'CSNK2B',\n",
       "   'EP300',\n",
       "   'HDAC1',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MYC',\n",
       "   'SEM1',\n",
       "   'TP53',\n",
       "   'TYMS']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Removal of licensing factors from origins',\n",
       "  'react_pathway_id': 'R-HSA-69300',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of DNA replication',\n",
       "  'react_pathway_id': 'R-HSA-69304',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'DNA Replication',\n",
       "  'react_pathway_id': 'R-HSA-69306',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'G2/M DNA damage checkpoint',\n",
       "  'react_pathway_id': 'R-HSA-69473',\n",
       "  'inference_gene_symbol': ['ATM', 'SFN', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'G2/M Checkpoints',\n",
       "  'react_pathway_id': 'R-HSA-69481',\n",
       "  'inference_gene_symbol': ['ATM', 'SEM1', 'SFN', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Stabilization of p53',\n",
       "  'react_pathway_id': 'R-HSA-69541',\n",
       "  'inference_gene_symbol': ['ATM', 'SEM1', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Transcriptional activation of p53 responsive genes',\n",
       "  'react_pathway_id': 'R-HSA-69560',\n",
       "  'inference_gene_symbol': 'TP53'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'p53-Dependent G1 DNA Damage Response',\n",
       "  'react_pathway_id': 'R-HSA-69563',\n",
       "  'inference_gene_symbol': ['ATM', 'SEM1', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'p53-Dependent G1/S DNA damage checkpoint',\n",
       "  'react_pathway_id': 'R-HSA-69580',\n",
       "  'inference_gene_symbol': ['ATM', 'SEM1', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Ubiquitin Mediated Degradation of Phosphorylated Cdc25A',\n",
       "  'react_pathway_id': 'R-HSA-69601',\n",
       "  'inference_gene_symbol': ['ATM', 'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'p53-Independent DNA Damage Response',\n",
       "  'react_pathway_id': 'R-HSA-69610',\n",
       "  'inference_gene_symbol': ['ATM', 'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'p53-Independent G1/S DNA damage checkpoint',\n",
       "  'react_pathway_id': 'R-HSA-69613',\n",
       "  'inference_gene_symbol': ['ATM', 'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'G1/S DNA Damage Checkpoints',\n",
       "  'react_pathway_id': 'R-HSA-69615',\n",
       "  'inference_gene_symbol': ['ATM', 'SEM1', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cell Cycle Checkpoints',\n",
       "  'react_pathway_id': 'R-HSA-69620',\n",
       "  'inference_gene_symbol': ['ATM', 'SEM1', 'SFN', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cyclin A:Cdk2-associated events at S phase entry',\n",
       "  'react_pathway_id': 'R-HSA-69656',\n",
       "  'inference_gene_symbol': ['CCND1', 'MYC', 'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Transcriptional  activation of  cell cycle inhibitor p21',\n",
       "  'react_pathway_id': 'R-HSA-69895',\n",
       "  'inference_gene_symbol': 'TP53'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Glycolysis',\n",
       "  'react_pathway_id': 'R-HSA-70171',\n",
       "  'inference_gene_symbol': ['ENO1', 'GAPDH', 'PGAM1', 'TPI1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Gluconeogenesis',\n",
       "  'react_pathway_id': 'R-HSA-70263',\n",
       "  'inference_gene_symbol': ['ENO1', 'GAPDH', 'PGAM1', 'TPI1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Glucose metabolism',\n",
       "  'react_pathway_id': 'R-HSA-70326',\n",
       "  'inference_gene_symbol': ['ENO1', 'GAPDH', 'PGAM1', 'TPI1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Metabolism of amino acids and derivatives',\n",
       "  'react_pathway_id': 'R-HSA-71291',\n",
       "  'inference_gene_symbol': ['SEM1', 'SLC45A2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Metabolism of carbohydrates',\n",
       "  'react_pathway_id': 'R-HSA-71387',\n",
       "  'inference_gene_symbol': ['ENO1', 'GAPDH', 'PGAM1', 'TPI1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'mRNA Capping',\n",
       "  'react_pathway_id': 'R-HSA-72086',\n",
       "  'inference_gene_symbol': 'ERCC2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'mRNA Splicing - Major Pathway',\n",
       "  'react_pathway_id': 'R-HSA-72163',\n",
       "  'inference_gene_symbol': 'HSPA8'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'mRNA Splicing',\n",
       "  'react_pathway_id': 'R-HSA-72172',\n",
       "  'inference_gene_symbol': 'HSPA8'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'mRNA Editing: C to U Conversion',\n",
       "  'react_pathway_id': 'R-HSA-72200',\n",
       "  'inference_gene_symbol': 'APOBEC3B'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Processing of Capped Intron-Containing Pre-mRNA',\n",
       "  'react_pathway_id': 'R-HSA-72203',\n",
       "  'inference_gene_symbol': 'HSPA8'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RNA Polymerase I Promoter Opening',\n",
       "  'react_pathway_id': 'R-HSA-73728',\n",
       "  'inference_gene_symbol': 'MAPK3'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RNA Polymerase I Transcription Initiation',\n",
       "  'react_pathway_id': 'R-HSA-73762',\n",
       "  'inference_gene_symbol': ['ERCC2', 'HDAC1', 'HDAC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RNA Polymerase I Promoter Escape',\n",
       "  'react_pathway_id': 'R-HSA-73772',\n",
       "  'inference_gene_symbol': 'ERCC2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RNA Polymerase II Promoter Escape',\n",
       "  'react_pathway_id': 'R-HSA-73776',\n",
       "  'inference_gene_symbol': 'ERCC2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RNA Polymerase I Chain Elongation',\n",
       "  'react_pathway_id': 'R-HSA-73777',\n",
       "  'inference_gene_symbol': 'ERCC2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RNA Polymerase II Transcription Pre-Initiation And Promoter Opening',\n",
       "  'react_pathway_id': 'R-HSA-73779',\n",
       "  'inference_gene_symbol': 'ERCC2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Purine metabolism',\n",
       "  'react_pathway_id': 'R-HSA-73847',\n",
       "  'inference_gene_symbol': 'GPX1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Pyrimidine metabolism',\n",
       "  'react_pathway_id': 'R-HSA-73848',\n",
       "  'inference_gene_symbol': ['TYMS', 'UMPS']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RNA Polymerase I Promoter Clearance',\n",
       "  'react_pathway_id': 'R-HSA-73854',\n",
       "  'inference_gene_symbol': ['ERCC2', 'HDAC1', 'HDAC2', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RNA Polymerase II Transcription',\n",
       "  'react_pathway_id': 'R-HSA-73857',\n",
       "  'inference_gene_symbol': 'ERCC2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RNA Polymerase I Transcription Termination',\n",
       "  'react_pathway_id': 'R-HSA-73863',\n",
       "  'inference_gene_symbol': 'ERCC2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RNA Polymerase I Transcription',\n",
       "  'react_pathway_id': 'R-HSA-73864',\n",
       "  'inference_gene_symbol': ['ERCC2', 'HDAC1', 'HDAC2', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Death Receptor Signalling',\n",
       "  'react_pathway_id': 'R-HSA-73887',\n",
       "  'inference_gene_symbol': 'RACK1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'DNA Damage Bypass',\n",
       "  'react_pathway_id': 'R-HSA-73893',\n",
       "  'inference_gene_symbol': 'UBA7'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'DNA Repair',\n",
       "  'react_pathway_id': 'R-HSA-73894',\n",
       "  'inference_gene_symbol': ['ATM',\n",
       "   'EP300',\n",
       "   'ERCC2',\n",
       "   'PRKDC',\n",
       "   'RAD52',\n",
       "   'TP53',\n",
       "   'UBA7',\n",
       "   'XPC']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Gene Expression',\n",
       "  'react_pathway_id': 'R-HSA-74160',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'APOBEC3B',\n",
       "   'ATM',\n",
       "   'BIRC5',\n",
       "   'CDKN2A',\n",
       "   'CREBBP',\n",
       "   'CSNK2A1',\n",
       "   'CSNK2A2',\n",
       "   'CSNK2B',\n",
       "   'EGFR',\n",
       "   'EP300',\n",
       "   'ERCC2',\n",
       "   'HDAC1',\n",
       "   'HDAC2',\n",
       "   'HSPA8',\n",
       "   'HSPB1',\n",
       "   'MAPK3',\n",
       "   'MYC',\n",
       "   'NOTCH1',\n",
       "   'NOTCH2',\n",
       "   'NOTCH3',\n",
       "   'NOTCH4',\n",
       "   'RARB',\n",
       "   'SEM1',\n",
       "   'SFN',\n",
       "   'STK11',\n",
       "   'TP53',\n",
       "   'TP63',\n",
       "   'TSC1',\n",
       "   'TSC2',\n",
       "   'VEGFA',\n",
       "   'WWOX']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Purine catabolism',\n",
       "  'react_pathway_id': 'R-HSA-74259',\n",
       "  'inference_gene_symbol': 'GPX1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signal attenuation',\n",
       "  'react_pathway_id': 'R-HSA-74749',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Insulin receptor signalling cascade',\n",
       "  'react_pathway_id': 'R-HSA-74751',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'STK11',\n",
       "   'TSC1',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by Insulin receptor',\n",
       "  'react_pathway_id': 'R-HSA-74752',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'RASA1',\n",
       "   'SEM1',\n",
       "   'STK11',\n",
       "   'TSC1',\n",
       "   'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Chk1/Chk2(Cds1) mediated inactivation of Cyclin B:Cdk1 complex',\n",
       "  'react_pathway_id': 'R-HSA-75035',\n",
       "  'inference_gene_symbol': 'SFN'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'mRNA Editing',\n",
       "  'react_pathway_id': 'R-HSA-75072',\n",
       "  'inference_gene_symbol': 'APOBEC3B'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Formation of the Editosome',\n",
       "  'react_pathway_id': 'R-HSA-75094',\n",
       "  'inference_gene_symbol': 'APOBEC3B'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Apoptotic execution  phase',\n",
       "  'react_pathway_id': 'R-HSA-75153',\n",
       "  'inference_gene_symbol': ['CDH1', 'PTK2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Dissolution of Fibrin Clot',\n",
       "  'react_pathway_id': 'R-HSA-75205',\n",
       "  'inference_gene_symbol': 'ANXA2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Ubiquitin-dependent degradation of Cyclin D',\n",
       "  'react_pathway_id': 'R-HSA-75815',\n",
       "  'inference_gene_symbol': ['CCND1', 'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Platelet Adhesion to exposed collagen',\n",
       "  'react_pathway_id': 'R-HSA-75892',\n",
       "  'inference_gene_symbol': 'ITGB1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TNF signaling',\n",
       "  'react_pathway_id': 'R-HSA-75893',\n",
       "  'inference_gene_symbol': 'RACK1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RNA Polymerase II Transcription Initiation',\n",
       "  'react_pathway_id': 'R-HSA-75953',\n",
       "  'inference_gene_symbol': 'ERCC2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RNA Polymerase II Transcription Elongation',\n",
       "  'react_pathway_id': 'R-HSA-75955',\n",
       "  'inference_gene_symbol': 'ERCC2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Platelet activation, signaling and aggregation',\n",
       "  'react_pathway_id': 'R-HSA-76002',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'ANXA5',\n",
       "   'HGF',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'PIK3CA',\n",
       "   'PPIA',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'TGFB1',\n",
       "   'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Response to elevated platelet cytosolic Ca2+',\n",
       "  'react_pathway_id': 'R-HSA-76005',\n",
       "  'inference_gene_symbol': ['ANXA5', 'HGF', 'PPIA', 'TGFB1', 'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Platelet Aggregation (Plug Formation)',\n",
       "  'react_pathway_id': 'R-HSA-76009',\n",
       "  'inference_gene_symbol': ['AKT1', 'PTK2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RNA Polymerase II Transcription Initiation And Promoter Clearance',\n",
       "  'react_pathway_id': 'R-HSA-76042',\n",
       "  'inference_gene_symbol': 'ERCC2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RNA Pol II CTD phosphorylation and interaction with CE',\n",
       "  'react_pathway_id': 'R-HSA-77075',\n",
       "  'inference_gene_symbol': 'ERCC2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'The NLRP3 inflammasome',\n",
       "  'react_pathway_id': 'R-HSA-844456',\n",
       "  'inference_gene_symbol': 'PYCARD'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'The AIM2 inflammasome',\n",
       "  'react_pathway_id': 'R-HSA-844615',\n",
       "  'inference_gene_symbol': 'PYCARD'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Interferon gamma signaling',\n",
       "  'react_pathway_id': 'R-HSA-877300',\n",
       "  'inference_gene_symbol': 'PTPN11'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of IFNG signaling',\n",
       "  'react_pathway_id': 'R-HSA-877312',\n",
       "  'inference_gene_symbol': 'PTPN11'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Advanced glycosylation endproduct receptor signaling',\n",
       "  'react_pathway_id': 'R-HSA-879415',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Gastrin-CREB signalling pathway via PKC and MAPK',\n",
       "  'react_pathway_id': 'R-HSA-881907',\n",
       "  'inference_gene_symbol': ['ANXA1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTGER1',\n",
       "   'PTK2',\n",
       "   'RASA1',\n",
       "   'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'ERBB2 Activates PTK6 Signaling',\n",
       "  'react_pathway_id': 'R-HSA-8847993',\n",
       "  'inference_gene_symbol': 'EGFR'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by PTK6',\n",
       "  'react_pathway_id': 'R-HSA-8848021',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'CCND1',\n",
       "   'EGFR',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'RASA1',\n",
       "   'STAT3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'PTK6 Regulates RTKs and Their Effectors AKT1 and DOK1',\n",
       "  'react_pathway_id': 'R-HSA-8849469',\n",
       "  'inference_gene_symbol': 'AKT1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'PTK6 Regulates Cell Cycle',\n",
       "  'react_pathway_id': 'R-HSA-8849470',\n",
       "  'inference_gene_symbol': 'CCND1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'PTK6 Regulates RHO GTPases, RAS GTPase and MAP kinases',\n",
       "  'react_pathway_id': 'R-HSA-8849471',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS', 'RASA1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'PTK6 Activates STAT3',\n",
       "  'react_pathway_id': 'R-HSA-8849474',\n",
       "  'inference_gene_symbol': 'STAT3'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'MET activates RAS signaling',\n",
       "  'react_pathway_id': 'R-HSA-8851805',\n",
       "  'inference_gene_symbol': ['HGF', 'HRAS', 'KRAS', 'MET']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'MET activates PI3K/AKT signaling',\n",
       "  'react_pathway_id': 'R-HSA-8851907',\n",
       "  'inference_gene_symbol': ['HGF', 'MET', 'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Protein ubiquitination',\n",
       "  'react_pathway_id': 'R-HSA-8852135',\n",
       "  'inference_gene_symbol': ['PRKDC', 'UBA1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'The role of GTSE1 in G2/M progression after G2 checkpoint',\n",
       "  'react_pathway_id': 'R-HSA-8852276',\n",
       "  'inference_gene_symbol': ['SEM1', 'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by FGFR3 fusions in cancer',\n",
       "  'react_pathway_id': 'R-HSA-8853334',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS', 'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by plasma membrane FGFR1 fusions',\n",
       "  'react_pathway_id': 'R-HSA-8853336',\n",
       "  'inference_gene_symbol': 'FGFR1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by FGFR3 point mutants in cancer',\n",
       "  'react_pathway_id': 'R-HSA-8853338',\n",
       "  'inference_gene_symbol': ['HRAS', 'KRAS', 'PIK3CA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RET signaling',\n",
       "  'react_pathway_id': 'R-HSA-8853659',\n",
       "  'inference_gene_symbol': ['EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'PTPN11',\n",
       "   'RASA1',\n",
       "   'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'FBXL7 down-regulates AURKA during mitotic entry and in early mitosis',\n",
       "  'react_pathway_id': 'R-HSA-8854050',\n",
       "  'inference_gene_symbol': 'SEM1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TBC/RABGAPs',\n",
       "  'react_pathway_id': 'R-HSA-8854214',\n",
       "  'inference_gene_symbol': ['TSC1', 'TSC2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Cargo recognition for clathrin-mediated endocytosis',\n",
       "  'react_pathway_id': 'R-HSA-8856825',\n",
       "  'inference_gene_symbol': 'EGFR'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Clathrin-mediated endocytosis',\n",
       "  'react_pathway_id': 'R-HSA-8856828',\n",
       "  'inference_gene_symbol': ['CTTN', 'EGFR', 'HSPA8']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'PTK6 promotes HIF1A stabilization',\n",
       "  'react_pathway_id': 'R-HSA-8857538',\n",
       "  'inference_gene_symbol': 'EGFR'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': \"Deregulated CDK5 triggers multiple neurodegenerative pathways in Alzheimer's disease models\",\n",
       "  'react_pathway_id': 'R-HSA-8862803',\n",
       "  'inference_gene_symbol': 'SOD2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Neurodegenerative Diseases',\n",
       "  'react_pathway_id': 'R-HSA-8863678',\n",
       "  'inference_gene_symbol': 'SOD2'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Downregulation of ERBB2 signaling',\n",
       "  'react_pathway_id': 'R-HSA-8863795',\n",
       "  'inference_gene_symbol': ['AKT1', 'EGFR']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Transcriptional regulation by the AP-2 (TFAP2) family of transcription factors',\n",
       "  'react_pathway_id': 'R-HSA-8864260',\n",
       "  'inference_gene_symbol': ['CREBBP',\n",
       "   'EGFR',\n",
       "   'EP300',\n",
       "   'MYC',\n",
       "   'VEGFA',\n",
       "   'WWOX']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'MET activates PTPN11',\n",
       "  'react_pathway_id': 'R-HSA-8865999',\n",
       "  'inference_gene_symbol': ['HGF', 'MET', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Synthesis of active ubiquitin: roles of E1 and E2 enzymes',\n",
       "  'react_pathway_id': 'R-HSA-8866652',\n",
       "  'inference_gene_symbol': 'UBA1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'E3 ubiquitin ligases ubiquitinate target proteins',\n",
       "  'react_pathway_id': 'R-HSA-8866654',\n",
       "  'inference_gene_symbol': 'PRKDC'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Negative regulation of activity of TFAP2 (AP-2) family transcription factors',\n",
       "  'react_pathway_id': 'R-HSA-8866904',\n",
       "  'inference_gene_symbol': 'WWOX'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Activation of the TFAP2 (AP-2) family of transcription factors',\n",
       "  'react_pathway_id': 'R-HSA-8866907',\n",
       "  'inference_gene_symbol': ['CREBBP', 'EP300', 'WWOX']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TFAP2 (AP-2) family regulates transcription of growth factors and their receptors',\n",
       "  'react_pathway_id': 'R-HSA-8866910',\n",
       "  'inference_gene_symbol': ['EGFR', 'VEGFA']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TFAP2 (AP-2) family regulates transcription of cell cycle factors',\n",
       "  'react_pathway_id': 'R-HSA-8866911',\n",
       "  'inference_gene_symbol': 'MYC'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'MET activates PTK2 signaling',\n",
       "  'react_pathway_id': 'R-HSA-8874081',\n",
       "  'inference_gene_symbol': ['HGF', 'ITGB1', 'MET', 'PTK2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'InlB-mediated entry of Listeria monocytogenes into host cell',\n",
       "  'react_pathway_id': 'R-HSA-8875360',\n",
       "  'inference_gene_symbol': 'MET'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'MET interacts with TNS proteins',\n",
       "  'react_pathway_id': 'R-HSA-8875513',\n",
       "  'inference_gene_symbol': ['HGF', 'ITGB1', 'MET']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'MET activates RAP1 and RAC1',\n",
       "  'react_pathway_id': 'R-HSA-8875555',\n",
       "  'inference_gene_symbol': ['HGF', 'MET']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'MET receptor recycling',\n",
       "  'react_pathway_id': 'R-HSA-8875656',\n",
       "  'inference_gene_symbol': ['HGF', 'MET']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'MET activates STAT3',\n",
       "  'react_pathway_id': 'R-HSA-8875791',\n",
       "  'inference_gene_symbol': ['HGF', 'MET', 'STAT3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'MET promotes cell motility',\n",
       "  'react_pathway_id': 'R-HSA-8875878',\n",
       "  'inference_gene_symbol': ['HGF', 'ITGB1', 'MET', 'PTK2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'RAB GEFs exchange GTP for GDP on RABs',\n",
       "  'react_pathway_id': 'R-HSA-8876198',\n",
       "  'inference_gene_symbol': 'AKT1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Listeria monocytogenes entry into host cells',\n",
       "  'react_pathway_id': 'R-HSA-8876384',\n",
       "  'inference_gene_symbol': ['CDH1', 'MET']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'InlA-mediated entry of Listeria monocytogenes into host cells',\n",
       "  'react_pathway_id': 'R-HSA-8876493',\n",
       "  'inference_gene_symbol': 'CDH1'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Protein methylation',\n",
       "  'react_pathway_id': 'R-HSA-8876725',\n",
       "  'inference_gene_symbol': 'HSPA8'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'GABA synthesis, release, reuptake and degradation',\n",
       "  'react_pathway_id': 'R-HSA-888590',\n",
       "  'inference_gene_symbol': 'HSPA8'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Interferon alpha/beta signaling',\n",
       "  'react_pathway_id': 'R-HSA-909733',\n",
       "  'inference_gene_symbol': ['IFNA2', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Meiotic recombination',\n",
       "  'react_pathway_id': 'R-HSA-912446',\n",
       "  'inference_gene_symbol': 'ATM'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Interleukin receptor SHC signaling',\n",
       "  'react_pathway_id': 'R-HSA-912526',\n",
       "  'inference_gene_symbol': ['EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTK2',\n",
       "   'RASA1',\n",
       "   'SEM1']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of signaling by CBL',\n",
       "  'react_pathway_id': 'R-HSA-912631',\n",
       "  'inference_gene_symbol': 'PIK3CA'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Regulation of IFNA signaling',\n",
       "  'react_pathway_id': 'R-HSA-912694',\n",
       "  'inference_gene_symbol': ['IFNA2', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Interferon Signaling',\n",
       "  'react_pathway_id': 'R-HSA-913531',\n",
       "  'inference_gene_symbol': ['IFNA2', 'MAPK3', 'PTPN11', 'UBA7']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TRAF3-dependent IRF activation pathway',\n",
       "  'react_pathway_id': 'R-HSA-918233',\n",
       "  'inference_gene_symbol': ['CREBBP', 'EP300']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TRAF6 mediated IRF7 activation',\n",
       "  'react_pathway_id': 'R-HSA-933541',\n",
       "  'inference_gene_symbol': ['CREBBP', 'EP300', 'IFNA2']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Negative regulators of RIG-I/MDA5 signaling',\n",
       "  'react_pathway_id': 'R-HSA-936440',\n",
       "  'inference_gene_symbol': 'UBA7'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Activation of IRF3/IRF7 mediated by TBK1/IKK epsilon',\n",
       "  'react_pathway_id': 'R-HSA-936964',\n",
       "  'inference_gene_symbol': 'PTPN11'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TRIF-mediated TLR3/TLR4 signaling',\n",
       "  'react_pathway_id': 'R-HSA-937061',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3', 'PTPN11']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Transport to the Golgi and subsequent modification',\n",
       "  'react_pathway_id': 'R-HSA-948021',\n",
       "  'inference_gene_symbol': 'AREG'},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'TRAF6 mediated induction of NFkB and MAP kinases upon TLR7/8 or 9 activation',\n",
       "  'react_pathway_id': 'R-HSA-975138',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'MyD88 dependent cascade initiated on endosome',\n",
       "  'react_pathway_id': 'R-HSA-975155',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'MyD88 cascade initiated on plasma membrane',\n",
       "  'react_pathway_id': 'R-HSA-975871',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Growth hormone receptor signaling',\n",
       "  'react_pathway_id': 'R-HSA-982772',\n",
       "  'inference_gene_symbol': ['MAPK1', 'MAPK3', 'STAT3']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Antigen processing: Ubiquitination & Proteasome degradation',\n",
       "  'react_pathway_id': 'R-HSA-983168',\n",
       "  'inference_gene_symbol': ['SEM1', 'UBA1', 'UBA7']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Class I MHC mediated antigen processing & presentation',\n",
       "  'react_pathway_id': 'R-HSA-983169',\n",
       "  'inference_gene_symbol': ['SEM1', 'UBA1', 'UBA7']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Factors involved in megakaryocyte development and platelet production',\n",
       "  'react_pathway_id': 'R-HSA-983231',\n",
       "  'inference_gene_symbol': ['CREBBP',\n",
       "   'EP300',\n",
       "   'HDAC1',\n",
       "   'HDAC2',\n",
       "   'IFNA2',\n",
       "   'TP53']},\n",
       " {'source': 'CTD',\n",
       "  'pathway_name': 'Signaling by the B Cell Receptor (BCR)',\n",
       "  'react_pathway_id': 'R-HSA-983705',\n",
       "  'inference_gene_symbol': ['AKT1',\n",
       "   'EGFR',\n",
       "   'FGFR1',\n",
       "   'HGF',\n",
       "   'HRAS',\n",
       "   'KRAS',\n",
       "   'MAPK1',\n",
       "   'MAPK3',\n",
       "   'MET',\n",
       "   'PIK3CA',\n",
       "   'PTPN11',\n",
       "   'SEM1',\n",
       "   'TP53',\n",
       "   'TSC2']}]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pathwaydf['MESH:D002294']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.4"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": true,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "165px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
